// vitepress-theme-teek@1.5.1/lib/helper/index.js
'use strict';

var is = require('./is.js');
var date = require('./date.js');
var util = require('./util.js');
var color = require('./color.js');
var baiduAnalytics = require('./analytics/baiduAnalytics.js');
var googleAnalytics = require('./analytics/googleAnalytics.js');
var umamiAnalytics = require('./analytics/umamiAnalytics.js');



exports.inBrowser = is.inBrowser;
exports.is = is.is;
exports.isArray = is.isArray;
exports.isAsyncFunction = is.isAsyncFunction;
exports.isBoolean = is.isBoolean;
exports.isClient = is.isClient;
exports.isDate = is.isDate;
exports.isDef = is.isDef;
exports.isElement = is.isElement;
exports.isEmpty = is.isEmpty;
exports.isExternal = is.isExternal;
exports.isFocusable = is.isFocusable;
exports.isFunction = is.isFunction;
exports.isImageDom = is.isImageDom;
exports.isImagePath = is.isImagePath;
exports.isIos = is.isIos;
exports.isNull = is.isNull;
exports.isNullAndUnDef = is.isNullAndUnDef;
exports.isNullOrUnDef = is.isNullOrUnDef;
exports.isNumber = is.isNumber;
exports.isObject = is.isObject;
exports.isPhone = is.isPhone;
exports.isPlainFunction = is.isPlainFunction;
exports.isPromise = is.isPromise;
exports.isServer = is.isServer;
exports.isString = is.isString;
exports.isStringNumber = is.isStringNumber;
exports.isType = is.isType;
exports.isUnDef = is.isUnDef;
exports.isValidURL = is.isValidURL;
exports.formatDate = date.formatDate;
exports.formatDiffDate = date.formatDiffDate;
exports.formatDiffDateToDay = date.formatDiffDateToDay;
exports.getNowDate = date.getNowDate;
exports.addUnit = util.addUnit;
exports.get = util.get;
exports.removeStorageItem = util.removeStorageItem;
exports.removeUnit = util.removeUnit;
exports.upperFirst = util.upperFirst;
exports.withBase = util.withBase;
exports.getDarkColor = color.getDarkColor;
exports.getLightColor = color.getLightColor;
exports.hexToRgb = color.hexToRgb;
exports.rgbToHex = color.rgbToHex;
exports.baiduAnalytics = baiduAnalytics.baiduAnalytics;
exports.trackPageview = baiduAnalytics.trackPageview;
exports.googleAnalytics = googleAnalytics.googleAnalytics;
exports.umamiAnalytics = umamiAnalytics.umamiAnalytics;
