// vitepress-theme-teek/es/components/theme/HomePostList/src/HomePostItem.vue2.mjs
import { defineComponent, computed, createElementBlock, openBlock, normalizeClass, unref, createCommentVNode, createElementVNode, createVNode, toDisplayString, createBlock, resolveDynamicComponent, mergeProps } from 'vue';
import { withBase } from 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../composables/useLocale.mjs';
import '../../../../composables/useZIndex.mjs';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';
import _sfc_main$2 from '../../ArticleInfo/src/index.vue2.mjs';
import _sfc_main$1 from '../../ArticleTitle/src/index.vue2.mjs';

const _hoisted_1 = ["title", "aria-label"];
const _hoisted_2 = { class: "list flx" };
const _hoisted_3 = ["href", "aria-label"];
const _hoisted_4 = ["aria-label"];
const _hoisted_5 = ["innerHTML"];
const _hoisted_6 = ["href", "aria-label"];
const _hoisted_7 = ["aria-label"];
const _hoisted_8 = ["aria-label"];
const _hoisted_9 = ["innerHTML"];
const _hoisted_10 = ["href", "aria-label"];
const _hoisted_11 = ["href", "aria-label"];
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "HomePostItem" },
  __name: "HomePostItem",
  props: {
    post: {},
    coverImgMode: {}
  },
  setup(__props) {
    const ns = useNamespace("post-item");
    const { t } = useLocale();
    const { getTeekConfigRef } = useTeekConfig();
    const postConfig = getTeekConfigRef("post", {
      excerptPosition: "bottom",
      showMore: true,
      moreLabel: t("tk.homePost.moreLabel"),
      showCapture: false,
      splitSeparator: false,
      listStyleTitleTagPosition: "right",
      defaultCoverImg: []
    });
    const articleConfig = getTeekConfigRef("articleAnalyze", {
      showInfo: true
    });
    const postUrl = __props.post.url && withBase(__props.post.url);
    const excerpt = computed(
      () => __props.post.frontmatter.description || __props.post.excerpt || postConfig.value.showCapture && __props.post.capture
    );
    const imgSrcList = computed(() => [__props.post.frontmatter.coverImg || postConfig.value.defaultCoverImg || []].flat());
    const coverImgMap = computed(() => {
      const imgSrc = imgSrcList.value[Math.floor(Math.random() * imgSrcList.value.length)];
      return {
        default: {
          is: "div",
          props: {
            style: `background-image: url(${withBase(imgSrc)});`
          }
        },
        full: {
          is: "img",
          props: {
            src: withBase(imgSrc)
          }
        }
      };
    });
    const isShowInfo = computed(() => {
      const arr = [articleConfig.value.showInfo].flat();
      return arr.includes(true) || arr.includes("post");
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        "div",
        {
          class: normalizeClass(unref(ns).b())
        },
        [
          !!__props.post.frontmatter.sticky ? (openBlock(), createElementBlock("i", {
            key: 0,
            class: "pin",
            title: unref(t)("tk.homePost.pin", { sticky: __props.post.frontmatter.sticky }),
            "aria-label": unref(t)("tk.homePost.pinLabel")
          }, null, 8, _hoisted_1)) : createCommentVNode("v-if", true),
          createElementVNode("div", _hoisted_2, [
            createElementVNode(
              "div",
              {
                class: normalizeClass(unref(ns).e("left"))
              },
              [
                createCommentVNode(" \u6807\u9898 "),
                createElementVNode("a", {
                  class: normalizeClass([unref(ns).e("left__title"), "hover-color", "sle"]),
                  href: unref(postUrl),
                  "aria-label": __props.post.title
                }, [
                  createVNode(unref(_sfc_main$1), {
                    post: __props.post,
                    "title-tag-props": { position: unref(postConfig).listStyleTitleTagPosition }
                  }, null, 8, ["post", "title-tag-props"])
                ], 10, _hoisted_3),
                createCommentVNode(" \u6458\u8981 top "),
                excerpt.value && unref(postConfig).excerptPosition === "top" ? (openBlock(), createElementBlock("div", {
                  key: 0,
                  class: normalizeClass(`${unref(ns).e("left__excerpt")} top`),
                  "aria-label": unref(t)("tk.homePost.excerptLabel")
                }, [
                  createElementVNode("div", {
                    class: "excerpt",
                    innerHTML: excerpt.value
                  }, null, 8, _hoisted_5),
                  unref(postConfig).showMore ? (openBlock(), createElementBlock("a", {
                    key: 0,
                    class: "more",
                    href: unref(postUrl),
                    "aria-label": unref(postConfig).moreLabel
                  }, toDisplayString(unref(postConfig).moreLabel), 9, _hoisted_6)) : createCommentVNode("v-if", true)
                ], 10, _hoisted_4)) : createCommentVNode("v-if", true),
                createCommentVNode(" \u6587\u7AE0\u4FE1\u606F "),
                createElementVNode("div", {
                  class: normalizeClass(unref(ns).e("left__footer")),
                  "aria-label": unref(t)("tk.homePost.infoLabel")
                }, [
                  isShowInfo.value ? (openBlock(), createBlock(unref(_sfc_main$2), {
                    key: 0,
                    post: __props.post,
                    scope: "post",
                    split: unref(postConfig).splitSeparator
                  }, null, 8, ["post", "split"])) : createCommentVNode("v-if", true)
                ], 10, _hoisted_7),
                createCommentVNode(" \u6458\u8981 bottom "),
                excerpt.value && unref(postConfig).excerptPosition === "bottom" ? (openBlock(), createElementBlock("div", {
                  key: 1,
                  class: normalizeClass(`${unref(ns).e("left__excerpt")} bottom`),
                  "aria-label": unref(t)("tk.homePost.excerptLabel")
                }, [
                  createElementVNode("div", {
                    class: "excerpt",
                    innerHTML: excerpt.value
                  }, null, 8, _hoisted_9),
                  unref(postConfig).showMore ? (openBlock(), createElementBlock("a", {
                    key: 0,
                    class: "more",
                    href: unref(postUrl),
                    "aria-label": unref(postConfig).moreLabel
                  }, toDisplayString(unref(postConfig).moreLabel), 9, _hoisted_10)) : createCommentVNode("v-if", true)
                ], 10, _hoisted_8)) : createCommentVNode("v-if", true)
              ],
              2
              /* CLASS */
            ),
            createCommentVNode(" \u53F3\u4FA7\u5C01\u9762\u56FE "),
            createElementVNode(
              "div",
              {
                class: normalizeClass(`${unref(ns).e("right")} flx-align-center`)
              },
              [
                imgSrcList.value.length ? (openBlock(), createElementBlock("a", {
                  key: 0,
                  href: unref(postUrl),
                  class: normalizeClass(`${__props.coverImgMode} cover`),
                  "aria-label": unref(t)("tk.homePost.coverImgLabel")
                }, [
                  (openBlock(), createBlock(
                    resolveDynamicComponent(coverImgMap.value[__props.coverImgMode].is),
                    mergeProps(coverImgMap.value[__props.coverImgMode].props, {
                      "aria-hidden": "true",
                      class: "cover-img"
                    }),
                    null,
                    16
                    /* FULL_PROPS */
                  ))
                ], 10, _hoisted_11)) : createCommentVNode("v-if", true)
              ],
              2
              /* CLASS */
            )
          ])
        ],
        2
        /* CLASS */
      );
    };
  }
});

export { _sfc_main as default };
