// vitepress-theme-teek/es/components/theme/HomePostList/src/index.vue2.mjs
import { defineComponent, ref, useModel, computed, watchEffect, watch, reactive, onMounted, createElementBlock, openBlock, normalizeClass, unref, Fragment, renderSlot, createElementVNode, normalizeProps, guardReactiveProps, createVNode, TransitionGroup, withCtx, renderList, createBlock, createCommentVNode, mergeProps, toDisplayString, nextTick } from 'vue';
import { useData, useRoute } from 'vitepress';
import { isClient, removeUnit } from '../../../../helper/dist/index.mjs';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../composables/useLocale.mjs';
import { useWindowSize } from '../../../../composables/useWindowSize.mjs';
import { useWindowTransition } from '../../../../composables/useWindowTransition.mjs';
import '../../../../composables/useZIndex.mjs';
import emptyIcon from '../../../../static/icons/empty.mjs';
import { useTeekConfig, usePosts, useWindowTransitionConfig } from '../../ConfigProvider/index.mjs';
import _sfc_main$3 from '../../../common/Pagination/src/index.vue2.mjs';
import _sfc_main$4 from '../../../common/Icon/src/index.vue2.mjs';
import '@iconify/vue';
import { pageNumKey } from './homePostList.mjs';
import _sfc_main$2 from './HomePostItem.vue2.mjs';
import _sfc_main$1 from './HomePostItemCard.vue2.mjs';

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["aria-label"];
const targetSize = "small";
const targetLayout = "prev, pager, next";
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "HomePostList" },
  __name: "index",
  props: {
    "modelValue": { default: false },
    "modelModifiers": {}
  },
  emits: ["update:modelValue"],
  setup(__props, { expose: __expose }) {
    const ns = useNamespace("post-list");
    const { t } = useLocale();
    const { getTeekConfigRef } = useTeekConfig();
    const posts = usePosts();
    const { frontmatter } = useData();
    const postConfig = getTeekConfigRef("post", {
      postStyle: "list",
      coverImgMode: "default",
      emptyLabel: t("tk.homePost.emptyLabel"),
      transition: true,
      transitionName: ns.join("slide-fade")
    });
    const pageConfig = getTeekConfigRef("page", {});
    const pageNum = ref(1);
    const total = ref(0);
    const coverImgMode = ref(postConfig.value.coverImgMode);
    const isPaging = useModel(__props, "modelValue");
    const defaultPageSize = computed(() => postConfig.value.postStyle === "list" ? 10 : 15);
    const pageSize = computed(() => pageConfig.value.pageSize || defaultPageSize.value);
    const route = useRoute();
    const currentPosts = ref([]);
    const totalPostsCount = ref(0);
    watchEffect(() => {
      coverImgMode.value = postConfig.value.coverImgMode;
    });
    const updateData = () => {
      if (!isClient) return;
      const { searchParams } = new URL(window.location.href);
      const p = Number(searchParams.get(pageNumKey)) || 1;
      if (p !== pageNum.value) pageNum.value = p;
      isPaging.value = p > 1;
      const postConst = posts.value;
      const frontmatterConst = frontmatter.value;
      let post = postConst.sortPostsByDateAndSticky;
      if (frontmatterConst.categoriesPage) {
        const c = searchParams.get("category");
        post = c ? postConst.groupPosts.categories[c] : post.filter((item) => item.frontmatter.categories);
      } else if (frontmatterConst.tagsPage) {
        const t2 = searchParams.get("tag");
        post = t2 ? postConst.groupPosts.tags[t2] : post.filter((item) => item.frontmatter.tags);
      }
      if (total.value !== post?.length) total.value = post?.length || 0;
      const inHomePosts = post.filter((item) => item.frontmatter.inHomePost !== false);
      totalPostsCount.value = inHomePosts.length;
      currentPosts.value = inHomePosts.slice((pageNum.value - 1) * pageSize.value, pageNum.value * pageSize.value);
    };
    watch(() => route.path, updateData, { immediate: true });
    const handlePagination = () => {
      const { searchParams } = new URL(window.location.href);
      searchParams.delete(pageNumKey);
      searchParams.append(pageNumKey, String(pageNum.value));
      window.history.pushState({}, "", `${window.location.pathname}?${searchParams.toString()}`);
      updateData();
      nextTick(() => {
        const rootStyles = getComputedStyle(document.documentElement);
        const navHeight = removeUnit(rootStyles.getPropertyValue("--vp-c-text-1"));
        document.querySelector("html")?.scrollTo({ top: window.innerHeight - navHeight, behavior: "smooth" });
      });
    };
    const pagePropsRef = reactive({ ...pageConfig.value });
    const { size = "default", layout = "prev, pager, next, jumper, ->, total" } = pageConfig.value;
    useWindowSize((width) => {
      if (width <= 768) {
        if (pagePropsRef.size !== targetSize) pagePropsRef.size = targetSize;
      } else if (pagePropsRef.size !== size) pagePropsRef.size = size;
      if (width <= 960) {
        if (pagePropsRef.layout !== targetLayout) pagePropsRef.layout = targetLayout;
      } else if (pagePropsRef.layout !== layout) pagePropsRef.layout = layout;
      if (width <= 960) {
        if (coverImgMode.value !== "default") coverImgMode.value = "default";
      } else if (coverImgMode.value !== postConfig.value.coverImgMode) coverImgMode.value = postConfig.value.coverImgMode;
    });
    const windowTransition = useWindowTransitionConfig((config) => config.post);
    const postItemListInstance = ref(null);
    const { start } = useWindowTransition(postItemListInstance, false);
    onMounted(() => {
      windowTransition.value && start();
    });
    __expose({ updateData });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        "div",
        {
          class: normalizeClass([unref(ns).b(), unref(ns).is("card", unref(postConfig).postStyle === "card")])
        },
        [
          currentPosts.value ? (openBlock(), createElementBlock(
            Fragment,
            { key: 0 },
            [
              renderSlot(_ctx.$slots, "teek-home-post-list", normalizeProps(guardReactiveProps({ currentPosts: currentPosts.value, transitionName: unref(postConfig).transitionName })), () => [
                createVNode(TransitionGroup, {
                  tag: "ul",
                  name: unref(postConfig).transition ? unref(postConfig).transitionName : "",
                  "aria-label": unref(t)("tk.homePost.label")
                }, {
                  default: withCtx(() => [
                    (openBlock(true), createElementBlock(
                      Fragment,
                      null,
                      renderList(currentPosts.value, (post) => {
                        return openBlock(), createElementBlock(
                          "li",
                          {
                            key: post.url,
                            class: normalizeClass({ "full-img": coverImgMode.value === "full" })
                          },
                          [
                            unref(windowTransition) ? (openBlock(), createElementBlock(
                              "div",
                              {
                                key: 0,
                                ref_for: true,
                                ref_key: "postItemListInstance",
                                ref: postItemListInstance,
                                style: { "width": "100%", "height": "100%" }
                              },
                              [
                                unref(postConfig).postStyle === "card" ? (openBlock(), createBlock(_sfc_main$1, {
                                  key: 0,
                                  post
                                }, null, 8, ["post"])) : (openBlock(), createBlock(_sfc_main$2, {
                                  key: 1,
                                  post,
                                  coverImgMode: coverImgMode.value
                                }, null, 8, ["post", "coverImgMode"]))
                              ],
                              512
                              /* NEED_PATCH */
                            )) : (openBlock(), createElementBlock(
                              Fragment,
                              { key: 1 },
                              [
                                unref(postConfig).postStyle === "card" ? (openBlock(), createBlock(_sfc_main$1, {
                                  key: 0,
                                  post
                                }, null, 8, ["post"])) : (openBlock(), createBlock(_sfc_main$2, {
                                  key: 1,
                                  post,
                                  coverImgMode: coverImgMode.value
                                }, null, 8, ["post", "coverImgMode"]))
                              ],
                              64
                              /* STABLE_FRAGMENT */
                            ))
                          ],
                          2
                          /* CLASS */
                        );
                      }),
                      128
                      /* KEYED_FRAGMENT */
                    ))
                  ]),
                  _: 1
                  /* STABLE */
                }, 8, ["name", "aria-label"])
              ]),
              createElementVNode("div", {
                class: normalizeClass(`${unref(ns).e("pagination")} flx-justify-center`),
                "aria-label": unref(t)("tk.homePost.pageLabel")
              }, [
                totalPostsCount.value >= pageSize.value ? (openBlock(), createBlock(unref(_sfc_main$3), mergeProps({
                  key: 0,
                  "page-size": pageSize.value,
                  "onUpdate:pageSize": _cache[0] || (_cache[0] = ($event) => pageSize.value = $event),
                  "current-page": pageNum.value,
                  "onUpdate:currentPage": _cache[1] || (_cache[1] = ($event) => pageNum.value = $event),
                  total: total.value,
                  background: "",
                  onCurrentChange: handlePagination
                }, { ...pagePropsRef }), null, 16, ["page-size", "current-page", "total"])) : createCommentVNode("v-if", true)
              ], 10, _hoisted_1)
            ],
            64
            /* STABLE_FRAGMENT */
          )) : (openBlock(), createElementBlock("div", {
            key: 1,
            class: normalizeClass([unref(ns).e("empty"), "flx-column-center"]),
            "aria-label": unref(postConfig).emptyLabel
          }, [
            createVNode(unref(_sfc_main$4), {
              icon: unref(emptyIcon),
              size: 160,
              color: "var(--vp-c-text-3)",
              "aria-hidden": "true"
            }, null, 8, ["icon"]),
            createElementVNode(
              "span",
              {
                class: normalizeClass(unref(ns).e("empty__title"))
              },
              toDisplayString(unref(postConfig).emptyLabel),
              3
              /* TEXT, CLASS */
            )
          ], 10, _hoisted_2))
        ],
        2
        /* CLASS */
      );
    };
  }
});

export { _sfc_main as default };
