// vitepress-theme-teek/es/components/theme/ArticleAppreciation/src/DocAfterAppreciationPopper.vue2.mjs
import { defineComponent, ref, computed, createElementBlock, openBlock, unref, normalizeClass, createVNode, withCtx, createElementVNode, createCommentVNode, createBlock } from 'vue';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../composables/useLocale.mjs';
import '../../../../composables/useZIndex.mjs';
import aliPayIcon from '../../../../static/icons/aliPay.mjs';
import weChatPayIcon from '../../../../static/icons/weChatPay.mjs';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';
import _sfc_main$2 from '../../../common/Icon/src/index.vue2.mjs';
import '@iconify/vue';
import _sfc_main$1 from '../../../common/Popover/src/index.vue2.mjs';

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["innerHTML", "aria-expanded", "aria-controls"];
const _hoisted_3 = ["aria-expanded", "aria-controls"];
const _hoisted_4 = ["innerHTML"];
const _hoisted_5 = ["aria-label"];
const _hoisted_6 = ["innerHTML"];
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "DocAfterAppreciationPopper" },
  __name: "DocAfterAppreciationPopper",
  setup(__props) {
    const ns = useNamespace("article-appreciation");
    const { t } = useLocale();
    const { getTeekConfigRef } = useTeekConfig();
    const appreciateConfig = getTeekConfigRef("appreciation", { position: "" });
    const showContent = ref(false);
    const docAfterPopperOptions = computed(
      () => ({ trigger: "click", ...appreciateConfig.value.options })
    );
    const icon = computed(() => {
      const { icon: icon2 } = docAfterPopperOptions.value;
      if (icon2 === "aliPay") return aliPayIcon;
      if (icon2 === "weChatPay") return weChatPayIcon;
      return icon2;
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass(unref(ns).b()),
        "aria-label": unref(t)("tk.articleAppreciation.label")
      }, [
        createVNode(unref(_sfc_main$1), {
          modelValue: showContent.value,
          "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => showContent.value = $event),
          trigger: docAfterPopperOptions.value.trigger
        }, {
          reference: withCtx(() => [
            docAfterPopperOptions.value.buttonHtml ? (openBlock(), createElementBlock("div", {
              key: 0,
              innerHTML: docAfterPopperOptions.value.buttonHtml,
              role: "button",
              "aria-expanded": showContent.value,
              "aria-controls": `${unref(ns).e("content")}`
            }, null, 8, _hoisted_2)) : docAfterPopperOptions.value.title ? (openBlock(), createElementBlock("button", {
              key: 1,
              class: normalizeClass(unref(ns).e("button")),
              "aria-expanded": showContent.value,
              "aria-controls": `${unref(ns).e("content")}`,
              "aria-live": "polite"
            }, [
              icon.value ? (openBlock(), createBlock(unref(_sfc_main$2), {
                key: 0,
                class: normalizeClass(unref(ns).e("button__icon")),
                icon: icon.value,
                size: 16,
                "aria-hidden": "true"
              }, null, 8, ["class", "icon"])) : createCommentVNode("v-if", true),
              docAfterPopperOptions.value.title ? (openBlock(), createElementBlock("span", {
                key: 1,
                innerHTML: docAfterPopperOptions.value.title
              }, null, 8, _hoisted_4)) : createCommentVNode("v-if", true)
            ], 10, _hoisted_3)) : createCommentVNode("v-if", true)
          ]),
          default: withCtx(() => [
            createElementVNode("div", {
              class: normalizeClass([unref(ns).e("content"), unref(ns).m("doc-after-popper")]),
              "aria-label": unref(t)("tk.articleAppreciation.contentLabel")
            }, [
              createElementVNode("div", {
                class: normalizeClass(unref(ns).e("content")),
                innerHTML: docAfterPopperOptions.value.content
              }, null, 10, _hoisted_6)
            ], 10, _hoisted_5)
          ]),
          _: 1
          /* STABLE */
        }, 8, ["modelValue", "trigger"])
      ], 10, _hoisted_1);
    };
  }
});

export { _sfc_main as default };
