// vitepress-theme-teek@1.5.1/lib/components/theme/HomeFriendLinkCard/src/ItemInfo.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');

const _hoisted_1 = ["href", "aria-label"];
const _hoisted_2 = { class: "friend-avatar skeleton-image" };
const _hoisted_3 = ["src", "alt"];
const _hoisted_4 = ["title"];
const _hoisted_5 = ["title"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  __name: "ItemInfo",
  props: {
    item: {},
    ns: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("a", {
        href: __props.item.link && Vue.unref(vitepress.withBase)(__props.item.link),
        target: "_blank",
        class: "hover-color flx-align-center",
        "aria-label": __props.item.name
      }, [
        Vue.createElementVNode("div", _hoisted_2, [
          Vue.createElementVNode("img", {
            src: __props.item.avatar && Vue.unref(vitepress.withBase)(__props.item.avatar),
            class: "avatar",
            alt: __props.item.alt || __props.item.name,
            "aria-hidden": "true",
            onLoad: _cache[0] || (_cache[0] = (e) => e.target?.classList.add("loaded")),
            onError: _cache[1] || (_cache[1] = (e) => e.target?.classList.add("loaded"))
          }, null, 40, _hoisted_3)
        ]),
        Vue.createElementVNode(
          "div",
          {
            class: Vue.normalizeClass(__props.ns.e("list__item__info"))
          },
          [
            Vue.createElementVNode("div", {
              class: "friend-name sle",
              title: __props.item.name
            }, Vue.toDisplayString(__props.item.name), 9, _hoisted_4),
            Vue.createElementVNode("div", {
              class: "friend-desc sle",
              title: __props.item.desc
            }, Vue.toDisplayString(__props.item.desc), 9, _hoisted_5)
          ],
          2
          /* CLASS */
        )
      ], 8, _hoisted_1);
    };
  }
});

exports.default = _sfc_main;
