// vitepress-theme-teek@1.5.1/es/components/theme/ThemeEnhance/src/namespace.mjs
import 'vue';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import '../../../../composables/useZIndex.mjs';

const ns = useNamespace("theme-enhance");
const transitionName = ns.join("theme-enhance-slide");
const pageMaxWidthVar = ns.cssVarName("page-max-width");
const docMaxWidthVar = ns.cssVarName("doc-max-width");
const layoutModeStorageKey = ns.storageKey("layoutMode");
const pageMaxWidthSlideStorageKey = ns.storageKey("pageMaxWidthSlide");
const docMaxWidthSlideStorageKey = ns.storageKey("docMaxWidthSlide");
const themeColorStorageKey = ns.storageKey("themeColor");
const themeBgColorStorageKey = ns.storageKey("themeBgColor");
const spotlightStorageKey = ns.storageKey("spotlight");
const spotlightStyleStorageKey = ns.storageKey("spotlightStyle");

export { docMaxWidthSlideStorageKey, docMaxWidthVar, layoutModeStorageKey, ns, pageMaxWidthSlideStorageKey, pageMaxWidthVar, spotlightStorageKey, spotlightStyleStorageKey, themeBgColorStorageKey, themeColorStorageKey, transitionName };
