// vitepress-theme-teek@1.5.1/es/components/theme/HomeFeature/src/index.vue2.mjs
import { defineComponent, ref, onMounted, createElementBlock, createCommentVNode, unref, openBlock, normalizeClass, Fragment, renderList, createElementVNode, createBlock, resolveDynamicComponent, withCtx } from 'vue';
import { withBase } from 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useWindowTransition } from '../../../../composables/useWindowTransition.mjs';
import '../../../../composables/useZIndex.mjs';
import { useTeekConfig, useWindowTransitionConfig } from '../../ConfigProvider/index.mjs';
import _sfc_main$1 from '../../../common/Icon/src/index.vue2.mjs';
import '@iconify/vue';

const _hoisted_1 = ["src"];
const _hoisted_2 = ["innerHTML"];
const _hoisted_3 = ["innerHTML"];
const _hoisted_4 = {
  key: 0,
  class: "features"
};
const _hoisted_5 = ["innerHTML"];
const _hoisted_6 = ["innerHTML"];
const _hoisted_7 = ["src"];
const _hoisted_8 = ["innerHTML"];
const _hoisted_9 = ["innerHTML"];
const _hoisted_10 = {
  key: 0,
  class: "highlights"
};
const _hoisted_11 = ["innerHTML"];
const _hoisted_12 = ["innerHTML"];
var _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "index",
  setup(__props) {
    const ns = useNamespace("home-features");
    const { getTeekConfigRef } = useTeekConfig();
    const featuresConfig = getTeekConfigRef("features", []);
    const windowTransition = useWindowTransitionConfig((config) => config.feature);
    const textInstance = ref(null);
    const { start } = useWindowTransition(textInstance, false);
    onMounted(() => {
      windowTransition.value && start();
    });
    return (_ctx, _cache) => {
      return unref(featuresConfig)?.length ? (openBlock(), createElementBlock(
        "div",
        {
          key: 0,
          class: normalizeClass([unref(ns).b(), "vp-doc"])
        },
        [
          (openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(unref(featuresConfig), (feature) => {
              return openBlock(), createElementBlock(
                Fragment,
                {
                  key: feature.title
                },
                [
                  feature.features ? (openBlock(), createElementBlock(
                    "div",
                    {
                      key: 0,
                      class: normalizeClass([unref(ns).e("feature-wrapper"), "features-wrapper"])
                    },
                    [
                      createElementVNode(
                        "div",
                        {
                          class: normalizeClass(unref(ns).e("feature__content"))
                        },
                        [
                          createElementVNode("img", {
                            src: feature.image,
                            class: "image"
                          }, null, 8, _hoisted_1),
                          createElementVNode(
                            "div",
                            {
                              class: normalizeClass([unref(ns).e("feature__header"), "header"])
                            },
                            [
                              createElementVNode("h2", {
                                ref_for: true,
                                ref_key: "textInstance",
                                ref: textInstance,
                                innerHTML: feature.title
                              }, null, 8, _hoisted_2),
                              feature.details ? (openBlock(), createElementBlock("p", {
                                key: 0,
                                ref_for: true,
                                ref_key: "textInstance",
                                ref: textInstance,
                                innerHTML: feature.details
                              }, null, 8, _hoisted_3)) : createCommentVNode("v-if", true)
                            ],
                            2
                            /* CLASS */
                          ),
                          feature.features?.length ? (openBlock(), createElementBlock("div", _hoisted_4, [
                            (openBlock(true), createElementBlock(
                              Fragment,
                              null,
                              renderList(feature.features, (item) => {
                                return openBlock(), createBlock(resolveDynamicComponent(item.link ? "a" : "div"), {
                                  key: item.title,
                                  href: unref(withBase)(item.link ?? ""),
                                  class: "features__item item"
                                }, {
                                  default: withCtx(() => [
                                    createElementVNode(
                                      "h3",
                                      {
                                        ref_for: true,
                                        ref_key: "textInstance",
                                        ref: textInstance,
                                        class: "flx-align-center"
                                      },
                                      [
                                        item.icon ? (openBlock(), createBlock(unref(_sfc_main$1), {
                                          key: 0,
                                          icon: item.icon,
                                          class: "feature-icon"
                                        }, null, 8, ["icon"])) : createCommentVNode("v-if", true),
                                        createElementVNode("span", {
                                          innerHTML: item.title
                                        }, null, 8, _hoisted_5)
                                      ],
                                      512
                                      /* NEED_PATCH */
                                    ),
                                    item.details ? (openBlock(), createElementBlock("p", {
                                      key: 0,
                                      ref_for: true,
                                      ref_key: "textInstance",
                                      ref: textInstance,
                                      innerHTML: item.details
                                    }, null, 8, _hoisted_6)) : createCommentVNode("v-if", true)
                                  ]),
                                  _: 2
                                  /* DYNAMIC */
                                }, 1032, ["href"]);
                              }),
                              128
                              /* KEYED_FRAGMENT */
                            ))
                          ])) : createCommentVNode("v-if", true)
                        ],
                        2
                        /* CLASS */
                      )
                    ],
                    2
                    /* CLASS */
                  )) : feature.highlights ? (openBlock(), createElementBlock(
                    "div",
                    {
                      key: 1,
                      class: normalizeClass([unref(ns).e("highlight-wrapper"), "features-wrapper"])
                    },
                    [
                      createElementVNode(
                        "div",
                        {
                          class: normalizeClass(unref(ns).e("highlight"))
                        },
                        [
                          createElementVNode("img", {
                            src: feature.image,
                            class: "image"
                          }, null, 8, _hoisted_7),
                          createElementVNode(
                            "div",
                            {
                              class: normalizeClass(unref(ns).e("highlight__content"))
                            },
                            [
                              createElementVNode(
                                "div",
                                {
                                  class: normalizeClass([unref(ns).e("highlight__header"), "header"])
                                },
                                [
                                  createElementVNode("h2", {
                                    ref_for: true,
                                    ref_key: "textInstance",
                                    ref: textInstance,
                                    innerHTML: feature.title
                                  }, null, 8, _hoisted_8),
                                  feature.details ? (openBlock(), createElementBlock("p", {
                                    key: 0,
                                    ref_for: true,
                                    ref_key: "textInstance",
                                    ref: textInstance,
                                    innerHTML: feature.details
                                  }, null, 8, _hoisted_9)) : createCommentVNode("v-if", true)
                                ],
                                2
                                /* CLASS */
                              ),
                              feature.highlights?.length ? (openBlock(), createElementBlock("div", _hoisted_10, [
                                (openBlock(true), createElementBlock(
                                  Fragment,
                                  null,
                                  renderList(feature.highlights, (item) => {
                                    return openBlock(), createBlock(resolveDynamicComponent(item.link ? "a" : "div"), {
                                      key: item.title,
                                      href: unref(withBase)(item.link ?? ""),
                                      class: normalizeClass([{ "no-details": !item.details }, "highlights__item item"])
                                    }, {
                                      default: withCtx(() => [
                                        createElementVNode(
                                          "h3",
                                          {
                                            ref_for: true,
                                            ref_key: "textInstance",
                                            ref: textInstance
                                          },
                                          [
                                            item.icon ? (openBlock(), createBlock(unref(_sfc_main$1), {
                                              key: 0,
                                              icon: item.icon,
                                              class: "feature-icon"
                                            }, null, 8, ["icon"])) : createCommentVNode("v-if", true),
                                            createElementVNode("span", {
                                              innerHTML: item.title
                                            }, null, 8, _hoisted_11)
                                          ],
                                          512
                                          /* NEED_PATCH */
                                        ),
                                        item.details ? (openBlock(), createElementBlock("p", {
                                          key: 0,
                                          ref_for: true,
                                          ref_key: "textInstance",
                                          ref: textInstance,
                                          innerHTML: item.details
                                        }, null, 8, _hoisted_12)) : createCommentVNode("v-if", true)
                                      ]),
                                      _: 2
                                      /* DYNAMIC */
                                    }, 1032, ["href", "class"]);
                                  }),
                                  128
                                  /* KEYED_FRAGMENT */
                                ))
                              ])) : createCommentVNode("v-if", true)
                            ],
                            2
                            /* CLASS */
                          )
                        ],
                        2
                        /* CLASS */
                      )
                    ],
                    2
                    /* CLASS */
                  )) : createCommentVNode("v-if", true)
                ],
                64
                /* STABLE_FRAGMENT */
              );
            }),
            128
            /* KEYED_FRAGMENT */
          ))
        ],
        2
        /* CLASS */
      )) : createCommentVNode("v-if", true);
    };
  }
});

export { _sfc_main as default };
