// vitepress-theme-teek@1.5.1/es/components/theme/FooterGroup/src/index.vue2.mjs
import { defineComponent, createElementBlock, createCommentVNode, unref, openBlock, normalizeClass, Fragment, renderList, createElementVNode, createBlock, createTextVNode, toDisplayString } from 'vue';
import { isExternal } from '../../../../helper/dist/index.mjs';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import '../../../../composables/useZIndex.mjs';
import externalLinkIcon from '../../../../static/icons/externalLink.mjs';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';
import _sfc_main$1 from '../../../common/Icon/src/index.vue2.mjs';
import '@iconify/vue';

const _hoisted_1 = ["name", "href", "title", "target", "aria-label", "aria-describedby"];
const _hoisted_2 = { class: "sle" };
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "FooterGroup" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace("footer-group");
    const { getTeekConfigRef } = useTeekConfig();
    const footerGroupConfig = getTeekConfigRef("footerGroup", []);
    return (_ctx, _cache) => {
      return unref(footerGroupConfig).length ? (openBlock(), createElementBlock(
        "div",
        {
          key: 0,
          class: normalizeClass(unref(ns).b())
        },
        [
          (openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(unref(footerGroupConfig), (group, index) => {
              return openBlock(), createElementBlock("div", {
                key: (group.title || "") + index
              }, [
                createElementVNode(
                  "div",
                  {
                    class: normalizeClass([unref(ns).e("title"), "flx-center"])
                  },
                  [
                    group.icon ? (openBlock(), createBlock(unref(_sfc_main$1), {
                      key: 0,
                      icon: group.icon,
                      style: { "margin-right": "4px" },
                      "aria-hidden": "true"
                    }, null, 8, ["icon"])) : createCommentVNode("v-if", true),
                    createTextVNode(
                      " " + toDisplayString(group.title),
                      1
                      /* TEXT */
                    )
                  ],
                  2
                  /* CLASS */
                ),
                createElementVNode("ul", null, [
                  (openBlock(true), createElementBlock(
                    Fragment,
                    null,
                    renderList(group.links || [], (link, idx) => {
                      return openBlock(), createElementBlock(
                        "li",
                        {
                          key: (link.name || "") + idx,
                          class: normalizeClass([unref(ns).e("link")])
                        },
                        [
                          link.icon ? (openBlock(), createBlock(unref(_sfc_main$1), {
                            key: 0,
                            icon: link.icon,
                            style: { "margin-right": "4px" },
                            "aria-hidden": "true"
                          }, null, 8, ["icon"])) : createCommentVNode("v-if", true),
                          createElementVNode("a", {
                            name: link.name,
                            href: link.link,
                            title: link.name,
                            target: unref(isExternal)(link.link || "") ? "_blank" : "_self",
                            class: "hover-color",
                            "aria-label": link.name,
                            rel: "noopener noreferrer",
                            "aria-describedby": link.name
                          }, [
                            createElementVNode(
                              "span",
                              _hoisted_2,
                              toDisplayString(link.name),
                              1
                              /* TEXT */
                            ),
                            unref(isExternal)(link.link || "") ? (openBlock(), createBlock(unref(_sfc_main$1), {
                              key: 0,
                              icon: unref(externalLinkIcon),
                              class: normalizeClass(unref(ns).e("link__external-icon")),
                              "aria-hidden": "true"
                            }, null, 8, ["icon", "class"])) : createCommentVNode("v-if", true)
                          ], 8, _hoisted_1)
                        ],
                        2
                        /* CLASS */
                      );
                    }),
                    128
                    /* KEYED_FRAGMENT */
                  ))
                ])
              ]);
            }),
            128
            /* KEYED_FRAGMENT */
          ))
        ],
        2
        /* CLASS */
      )) : createCommentVNode("v-if", true);
    };
  }
});

export { _sfc_main as default };
