// vitepress-theme-teek@1.5.1/es/components/common/Pagination/src/components/next.vue2.mjs
import { defineComponent, computed, createElementBlock, openBlock, unref, createBlock, createCommentVNode, toDisplayString } from 'vue';
import 'vitepress';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../../composables/useLocale.mjs';
import '../../../../../composables/useZIndex.mjs';
import _sfc_main$1 from '../../../Icon/src/index.vue2.mjs';
import '@iconify/vue';

const _hoisted_1 = ["disabled", "aria-label", "aria-disabled"];
const _hoisted_2 = { key: 0 };
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "PaginationNext" },
  __name: "next",
  props: {
    disabled: { type: Boolean },
    currentPage: {},
    pageCount: {},
    nextText: {},
    nextIcon: {}
  },
  emits: ["click"],
  setup(__props) {
    const props = __props;
    const { t } = useLocale();
    const internalDisabled = computed(
      () => props.disabled || props.currentPage === props.pageCount || props.pageCount === 0
    );
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("button", {
        type: "button",
        class: "btn-next",
        disabled: internalDisabled.value,
        "aria-label": __props.nextText || unref(t)("tk.pagination.next"),
        "aria-disabled": internalDisabled.value,
        onClick: _cache[0] || (_cache[0] = ($event) => _ctx.$emit("click", $event))
      }, [
        __props.nextText ? (openBlock(), createElementBlock(
          "span",
          _hoisted_2,
          toDisplayString(__props.nextText),
          1
          /* TEXT */
        )) : __props.nextIcon ? (openBlock(), createBlock(unref(_sfc_main$1), {
          key: 1,
          icon: __props.nextIcon
        }, null, 8, ["icon"])) : createCommentVNode("v-if", true)
      ], 8, _hoisted_1);
    };
  }
});

export { _sfc_main as default };
