// vitepress-theme-teek@1.5.1/lib/components/theme/CataloguePage/src/CatalogueItem.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
require('../../../../composables/useZIndex.js');
var index_vue_vue_type_script_setup_true_name_TitleTag_lang = require('../../../common/TitleTag/src/index.vue2.js');

const _hoisted_1 = ["href", "aria-label"];
const _hoisted_2 = ["innerHTML"];
const _hoisted_3 = ["id", "aria-labelledby"];
const _hoisted_4 = ["href", "aria-label"];
const _hoisted_5 = ["id"];
const _hoisted_6 = ["aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "CatalogueItem" },
  __name: "CatalogueItem",
  props: {
    item: {},
    index: {}
  },
  setup(__props) {
    const nsSub = useNamespace.useNamespace("sub-catalogue");
    const nsItem = useNamespace.useNamespace("catalogue-item");
    return (_ctx, _cache) => {
      const _component_CatalogueItem = Vue.resolveComponent("CatalogueItem", true);
      return Vue.openBlock(), Vue.createElementBlock(
        "li",
        {
          class: Vue.normalizeClass(__props.item.children ? Vue.unref(nsSub).b() : Vue.unref(nsItem).b())
        },
        [
          !__props.item.children ? (Vue.openBlock(), Vue.createElementBlock("a", {
            key: 0,
            href: __props.item.url && Vue.unref(vitepress.withBase)(__props.item.url),
            "aria-label": `${__props.index}. ${__props.item.title}`
          }, [
            Vue.createTextVNode(
              Vue.toDisplayString(__props.index) + ". ",
              1
              /* TEXT */
            ),
            Vue.createElementVNode("span", {
              innerHTML: __props.item.title
            }, null, 8, _hoisted_2),
            __props.item.frontmatter?.titleTag ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_TitleTag_lang.default), {
              key: 0,
              text: __props.item.frontmatter?.titleTag,
              position: "right",
              size: "small",
              "aria-label": __props.item.frontmatter?.titleTag
            }, null, 8, ["text", "aria-label"])) : Vue.createCommentVNode("v-if", true)
          ], 8, _hoisted_1)) : __props.item.children.length ? (Vue.openBlock(), Vue.createElementBlock(
            Vue.Fragment,
            { key: 1 },
            [
              Vue.createElementVNode("div", {
                id: __props.item.title,
                class: Vue.normalizeClass(Vue.unref(nsSub).e("title")),
                role: "group",
                "aria-labelledby": `${__props.item.title}-label`
              }, [
                Vue.createElementVNode("a", {
                  href: `#${__props.item.title}`,
                  class: "anchor",
                  "aria-label": __props.item.title
                }, "#", 8, _hoisted_4),
                Vue.createElementVNode("span", {
                  id: `${__props.item.title}-label`
                }, Vue.toDisplayString(`${__props.index}. ${__props.item.title}`), 9, _hoisted_5)
              ], 10, _hoisted_3),
              __props.item.children ? (Vue.openBlock(), Vue.createElementBlock("ul", {
                key: 0,
                class: Vue.normalizeClass(`${Vue.unref(nsSub).e("inline")} flx-wrap-between`),
                "aria-label": __props.item.title
              }, [
                Vue.createCommentVNode(" \u9012\u5F52\u81EA\u5DF1 "),
                (Vue.openBlock(true), Vue.createElementBlock(
                  Vue.Fragment,
                  null,
                  Vue.renderList(__props.item.children, (item, i) => {
                    return Vue.openBlock(), Vue.createBlock(_component_CatalogueItem, {
                      key: i,
                      item,
                      index: `${__props.index}-${i + 1}`
                    }, null, 8, ["item", "index"]);
                  }),
                  128
                  /* KEYED_FRAGMENT */
                ))
              ], 10, _hoisted_6)) : Vue.createCommentVNode("v-if", true)
            ],
            64
            /* STABLE_FRAGMENT */
          )) : Vue.createCommentVNode("v-if", true)
        ],
        2
        /* CLASS */
      );
    };
  }
});

exports.default = _sfc_main;
