// vitepress-theme-teek/es/components/theme/CataloguePage/src/CatalogueItem.vue2.mjs
import { defineComponent, resolveComponent, createElementBlock, openBlock, normalizeClass, unref, createCommentVNode, createTextVNode, createElementVNode, createBlock, toDisplayString, Fragment, renderList } from 'vue';
import { withBase } from 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import '../../../../composables/useZIndex.mjs';
import _sfc_main$1 from '../../../common/TitleTag/src/index.vue2.mjs';

const _hoisted_1 = ["href", "aria-label"];
const _hoisted_2 = ["innerHTML"];
const _hoisted_3 = ["id", "aria-labelledby"];
const _hoisted_4 = ["href", "aria-label"];
const _hoisted_5 = ["id"];
const _hoisted_6 = ["aria-label"];
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "CatalogueItem" },
  __name: "CatalogueItem",
  props: {
    item: {},
    index: {}
  },
  setup(__props) {
    const nsSub = useNamespace("sub-catalogue");
    const nsItem = useNamespace("catalogue-item");
    return (_ctx, _cache) => {
      const _component_CatalogueItem = resolveComponent("CatalogueItem", true);
      return openBlock(), createElementBlock(
        "li",
        {
          class: normalizeClass(__props.item.children ? unref(nsSub).b() : unref(nsItem).b())
        },
        [
          !__props.item.children ? (openBlock(), createElementBlock("a", {
            key: 0,
            href: __props.item.url && unref(withBase)(__props.item.url),
            "aria-label": `${__props.index}. ${__props.item.title}`
          }, [
            createTextVNode(
              toDisplayString(__props.index) + ". ",
              1
              /* TEXT */
            ),
            createElementVNode("span", {
              innerHTML: __props.item.title
            }, null, 8, _hoisted_2),
            __props.item.frontmatter?.titleTag ? (openBlock(), createBlock(unref(_sfc_main$1), {
              key: 0,
              text: __props.item.frontmatter?.titleTag,
              position: "right",
              size: "small",
              "aria-label": __props.item.frontmatter?.titleTag
            }, null, 8, ["text", "aria-label"])) : createCommentVNode("v-if", true)
          ], 8, _hoisted_1)) : __props.item.children.length ? (openBlock(), createElementBlock(
            Fragment,
            { key: 1 },
            [
              createElementVNode("div", {
                id: __props.item.title,
                class: normalizeClass(unref(nsSub).e("title")),
                role: "group",
                "aria-labelledby": `${__props.item.title}-label`
              }, [
                createElementVNode("a", {
                  href: `#${__props.item.title}`,
                  class: "anchor",
                  "aria-label": __props.item.title
                }, "#", 8, _hoisted_4),
                createElementVNode("span", {
                  id: `${__props.item.title}-label`
                }, toDisplayString(`${__props.index}. ${__props.item.title}`), 9, _hoisted_5)
              ], 10, _hoisted_3),
              __props.item.children ? (openBlock(), createElementBlock("ul", {
                key: 0,
                class: normalizeClass(`${unref(nsSub).e("inline")} flx-wrap-between`),
                "aria-label": __props.item.title
              }, [
                createCommentVNode(" \u9012\u5F52\u81EA\u5DF1 "),
                (openBlock(true), createElementBlock(
                  Fragment,
                  null,
                  renderList(__props.item.children, (item, i) => {
                    return openBlock(), createBlock(_component_CatalogueItem, {
                      key: i,
                      item,
                      index: `${__props.index}-${i + 1}`
                    }, null, 8, ["item", "index"]);
                  }),
                  128
                  /* KEYED_FRAGMENT */
                ))
              ], 10, _hoisted_6)) : createCommentVNode("v-if", true)
            ],
            64
            /* STABLE_FRAGMENT */
          )) : createCommentVNode("v-if", true)
        ],
        2
        /* CLASS */
      );
    };
  }
});

export { _sfc_main as default };
