// vitepress-theme-teek@1.5.1/es/markdown/plugins/video.mjs
import container from 'markdown-it-container';

const type = "video";
const videoPlugin = (md) => {
  md.use(container, type, {});
  md.renderer.rules[`container_${type}_open`] = (tokens, idx) => {
    const containerToken = tokens[idx];
    const info = containerToken.info.trim().slice(type.length).trim();
    const contentToken = tokens[idx + 2];
    const content = contentToken.content.trim() || "";
    const video = info ? videoMap[info] : { src: () => content, title: "Custom video player" };
    contentToken.type = "";
    contentToken.tag = "";
    contentToken.hidden = true;
    return `<div class="${`${type}-container`}">
      <iframe
        class="video-iframe"
        loading="lazy"
        src="${video.src(content)}"
        title="${video.title}"
        allow="accelerometer; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share; fullscreen;"
        allowfullscreen="true"
      />
    `;
  };
};
const videoMap = {
  bilibili: {
    src: (id) => `https://player.bilibili.com/player.html?bvid=${id}&autoplay=0`,
    title: "Bilibili video player"
  },
  tencent: {
    src: (id) => `https://v.qq.com/txp/iframe/player.html?vid=${id}`,
    title: "Tencent Video player"
  },
  youku: {
    src: (id) => `https://player.youku.com/embed/${id}`,
    title: "Youku video player"
  },
  youtube: {
    src: (id) => `https://www.youtube-nocookie.com/embed/${id}`,
    title: "YouTube video player"
  },
  vimeo: {
    src: (id) => `https://player.vimeo.com/video/${id}`,
    title: "Vimeo video player"
  },
  xigua: {
    src: (id) => `https://www.ixigua.com/iframe/${id}`,
    title: "XiGua video player"
  }
};

export { videoPlugin as default };
