// vitepress-theme-teek@1.5.1/lib/components/theme/ArticleOverviewPage/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var index$1 = require('../../../../helper/dist/index.js');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var index = require('../../ConfigProvider/index.js');
var index_vue_vue_type_script_setup_true_name_ArticlePage_lang = require('../../../common/ArticlePage/src/index.vue2.js');
var compile = require('../../ArticleTitle/src/compile.js');

const _hoisted_1 = { key: 0 };
const _hoisted_2 = ["href", "aria-label"];
const _hoisted_3 = ["id"];
const _hoisted_4 = ["href", "aria-label"];
const _hoisted_5 = ["href"];
const _hoisted_6 = { "aria-describedby": `overview-title` };
const _hoisted_7 = ["href", "aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "ArticleOverviewPage" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace.useNamespace("article-overview");
    const { t } = useLocale.useLocale();
    const posts = index.usePosts();
    const { localeIndex, site, theme, frontmatter } = vitepress.useData();
    const { getTeekConfigRef } = index.useTeekConfig();
    const categoryConfig = getTeekConfigRef("category", {
      path: "/categories"
    });
    const categories = Vue.computed(() => posts.value.groupPosts.categories);
    const eachFileWords = Vue.computed(() => theme.value.docAnalysisInfo?.eachFileWords || []);
    const categoriesPageLink = Vue.computed(() => {
      const localeIndexConst = localeIndex.value;
      const localeName = localeIndexConst !== "root" ? `/${localeIndexConst}` : "";
      return `${localeName}${categoryConfig.value.path}${site.value.cleanUrls ? "" : ".html"}`;
    });
    const getFileWords = (url) => {
      return eachFileWords.value.filter((item) => {
        const path = "/" + item.fileInfo.relativePath.replace(".md", "");
        return [path, `${path}.html`].includes(url);
      })[0];
    };
    const enhancedCategories = Vue.computed(() => {
      return Object.entries(categories.value).map(([key, items]) => ({
        name: key,
        data: items.map((item) => {
          const wordsInfo = getFileWords(item.url);
          return {
            ...item,
            wordCount: wordsInfo?.wordCount || "-",
            readingTime: wordsInfo?.readingTime || "-"
          };
        }) || []
      })).sort(
        (a, b) => new Date(b.data[b.data.length - 1].date).getTime() - new Date(a.data[a.data.length - 1].date).getTime()
      );
    });
    const formatPublishDate = (date) => {
      const publishDateFormat = frontmatter.value.publishDateFormat;
      if (!publishDateFormat) return date;
      return index$1.formatDate(date || /* @__PURE__ */ new Date(), publishDateFormat);
    };
    return (_ctx, _cache) => {
      const _component_Content = Vue.resolveComponent("Content");
      return Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticlePage_lang.default), {
        doc: "",
        aside: "",
        class: Vue.normalizeClass(Vue.unref(ns).b())
      }, {
        default: Vue.withCtx(() => [
          Vue.unref(frontmatter).title ? (Vue.openBlock(), Vue.createElementBlock("h1", _hoisted_1, [
            Vue.createTextVNode(
              Vue.toDisplayString(Vue.unref(frontmatter).title) + " ",
              1
              /* TEXT */
            ),
            Vue.createElementVNode("a", {
              class: "header-anchor",
              href: `#${Vue.unref(frontmatter).title}`,
              "aria-label": `Permalink to '${Vue.unref(frontmatter).title}'`
            }, null, 8, _hoisted_2)
          ])) : Vue.createCommentVNode("v-if", true),
          Vue.createVNode(_component_Content),
          (Vue.openBlock(true), Vue.createElementBlock(
            Vue.Fragment,
            null,
            Vue.renderList(enhancedCategories.value, (item) => {
              return Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                {
                  key: item.name
                },
                [
                  Vue.createElementVNode("h2", {
                    id: `${item.name}-${Vue.unref(t)("tk.articleOverview.overview")}`
                  }, [
                    Vue.createTextVNode(
                      Vue.toDisplayString(item.name) + " " + Vue.toDisplayString(Vue.unref(t)("tk.articleOverview.overview")) + " ",
                      1
                      /* TEXT */
                    ),
                    Vue.createElementVNode("a", {
                      class: "header-anchor",
                      href: `#${item.name}-${Vue.unref(t)("tk.articleOverview.overview")}`,
                      "aria-label": `Permalink to '${item.name}-${Vue.unref(t)("tk.articleOverview.overview")}'`
                    }, null, 8, _hoisted_4)
                  ], 8, _hoisted_3),
                  Vue.createElementVNode("a", {
                    href: `${categoriesPageLink.value}?category=${item.name}`,
                    "aria-describedby": `overview-title`
                  }, Vue.toDisplayString(item.name) + " " + Vue.toDisplayString(Vue.unref(t)("tk.articleOverview.category")), 9, _hoisted_5),
                  Vue.createElementVNode("table", _hoisted_6, [
                    Vue.createElementVNode("thead", null, [
                      Vue.createElementVNode("tr", null, [
                        Vue.createElementVNode(
                          "th",
                          null,
                          Vue.toDisplayString(Vue.unref(t)("tk.articleOverview.name")),
                          1
                          /* TEXT */
                        ),
                        Vue.createElementVNode(
                          "th",
                          null,
                          Vue.toDisplayString(Vue.unref(t)("tk.articleOverview.title")),
                          1
                          /* TEXT */
                        ),
                        Vue.createElementVNode(
                          "th",
                          null,
                          Vue.toDisplayString(Vue.unref(t)("tk.articleOverview.date")),
                          1
                          /* TEXT */
                        ),
                        Vue.createElementVNode(
                          "th",
                          null,
                          Vue.toDisplayString(Vue.unref(t)("tk.articleOverview.wordCount")),
                          1
                          /* TEXT */
                        ),
                        Vue.createElementVNode(
                          "th",
                          null,
                          Vue.toDisplayString(Vue.unref(t)("tk.articleOverview.readingTime")),
                          1
                          /* TEXT */
                        )
                      ])
                    ]),
                    Vue.createElementVNode("tbody", null, [
                      (Vue.openBlock(true), Vue.createElementBlock(
                        Vue.Fragment,
                        null,
                        Vue.renderList(item.data, (data) => {
                          return Vue.openBlock(), Vue.createElementBlock("tr", {
                            key: data.url
                          }, [
                            Vue.createElementVNode(
                              "td",
                              null,
                              Vue.toDisplayString(item.name),
                              1
                              /* TEXT */
                            ),
                            Vue.createElementVNode("td", null, [
                              Vue.createElementVNode("a", {
                                href: data.url && Vue.unref(vitepress.withBase)(data.url),
                                "aria-label": data.title
                              }, [
                                data.title ? (Vue.openBlock(), Vue.createBlock(Vue.resolveDynamicComponent(Vue.unref(compile.createDynamicComponent)(data.title)), { key: 0 })) : Vue.createCommentVNode("v-if", true)
                              ], 8, _hoisted_7)
                            ]),
                            Vue.createElementVNode(
                              "td",
                              null,
                              Vue.toDisplayString(formatPublishDate(data.date)),
                              1
                              /* TEXT */
                            ),
                            Vue.createElementVNode(
                              "td",
                              null,
                              Vue.toDisplayString(data.wordCount),
                              1
                              /* TEXT */
                            ),
                            Vue.createElementVNode(
                              "td",
                              null,
                              Vue.toDisplayString(data.readingTime),
                              1
                              /* TEXT */
                            )
                          ]);
                        }),
                        128
                        /* KEYED_FRAGMENT */
                      ))
                    ])
                  ])
                ],
                64
                /* STABLE_FRAGMENT */
              );
            }),
            128
            /* KEYED_FRAGMENT */
          ))
        ]),
        _: 1
        /* STABLE */
      }, 8, ["class"]);
    };
  }
});

exports.default = _sfc_main;
