// vitepress-theme-teek@1.5.1/es/components/theme/BodyBgImage/src/index.vue2.mjs
import { defineComponent, computed, onMounted, createElementBlock, openBlock, normalizeStyle, normalizeClass, unref, createCommentVNode } from 'vue';
import { withBase } from 'vitepress';
import { isFunction, isString } from '../../../../helper/dist/index.mjs';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useSwitchData } from '../../../../composables/useSwitchData.mjs';
import '../../../../composables/useZIndex.mjs';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';

const _hoisted_1 = {
  key: 0,
  class: "mask"
};
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "BodyBgImage" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace("body-bg-image");
    const { getTeekConfigRef } = useTeekConfig();
    const bodyBgImgConfig = getTeekConfigRef("bodyBgImg", {
      imgSrc: void 0,
      imgOpacity: 1,
      imgInterval: 15e3,
      imgShuffle: false,
      mask: false,
      maskBg: "rgba(0, 0, 0, 0.2)"
    });
    const dataArray = computed(() => {
      const imgSrc = bodyBgImgConfig.value.imgSrc;
      return [isFunction(imgSrc) ? imgSrc() : imgSrc || []].flat().map((item) => item && withBase(item));
    });
    const {
      data: imageSrc,
      start,
      index
    } = useSwitchData(dataArray, {
      timeout: bodyBgImgConfig.value.imgInterval,
      shuffle: bodyBgImgConfig.value.imgShuffle,
      onAfterUpdate: () => {
        const nextIndex = (index.value + 1) % dataArray.value.length;
        const newValue = dataArray.value[nextIndex];
        if (newValue) {
          const img = new Image();
          img.src = newValue;
        }
      }
    });
    onMounted(() => {
      start();
    });
    const getStyle = () => {
      const { imgSrc, imgOpacity, maskBg } = bodyBgImgConfig.value;
      const imgBgVar = ns.cssVarName("body-bg-img");
      const imgBgOpacityVar = ns.cssVarName("body-bg-img-opacity");
      const maskBgColorVar = ns.cssVarName("body-mask-bg-color");
      if (!imgSrc?.length) return { [imgBgVar]: ns.cssVar("bg-img-default") };
      return {
        [imgBgVar]: `url(${imageSrc.value}) center center / cover no-repeat`,
        [imgBgOpacityVar]: imgOpacity,
        [maskBgColorVar]: isString(maskBg) ? maskBg : `rgba(0, 0, 0, ${maskBg})`
      };
    };
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        "div",
        {
          class: normalizeClass(unref(ns).b()),
          style: normalizeStyle(getStyle())
        },
        [
          unref(bodyBgImgConfig).mask ? (openBlock(), createElementBlock("div", _hoisted_1)) : createCommentVNode("v-if", true)
        ],
        6
        /* CLASS, STYLE */
      );
    };
  }
});

export { _sfc_main as default };
