// vitepress-theme-teek/es/config/vitePlugins.mjs
import Sidebar from 'vitepress-plugin-sidebar-resolve';
import Permalink from 'vitepress-plugin-permalink';
import MdH1 from 'vitepress-plugin-md-h1';
import Catalogue from 'vitepress-plugin-catalogue';
import DocAnalysis from 'vitepress-plugin-doc-analysis';
import FileContentLoader from 'vitepress-plugin-file-content-loader';
import AutoFrontmatter from 'vitepress-plugin-auto-frontmatter';
import { createSimplePermalink, createComplexPermalink, createCategory, createCoverImg } from './autoFrontmatter/addFrontmatter.mjs';
import { transformRaw, transformData } from './post/index.mjs';

const registerPluginAndGet = (vitePlugins = {}, teekTheme = true) => {
  const plugins = [];
  const ignoreDir = {
    autoFrontmatter: ["**/@pages/**", "**/.scripts/**"],
    sidebar: ["@pages", "@fragment", "examples", ".scripts"],
    mdH1: ["@pages", ".scripts"],
    docAnalysis: ["@pages", /目录页/, ".scripts"],
    fileContentLoader: ["**/components/**", "**/.vitepress/**", "**/public/**", "**/*\u76EE\u5F55\u9875*/**", "**/.scripts/**"]
  };
  plugins.push(...registerLoosePlugins(vitePlugins, ignoreDir));
  if (teekTheme !== false) plugins.push(...registerTightPlugins(vitePlugins, ignoreDir));
  return plugins;
};
const registerLoosePlugins = (vitePlugins, ignoreDir) => {
  const plugins = [];
  const {
    sidebar = true,
    sidebarOption = {},
    permalink = true,
    permalinkOption = {},
    mdH1 = true,
    mdH1Option = {},
    docAnalysis = true,
    docAnalysisOption = {},
    autoFrontmatter = false,
    autoFrontmatterOption = {}
  } = vitePlugins || {};
  if (autoFrontmatter) {
    const {
      pattern,
      globOptions = {},
      transform,
      categories = true,
      permalink: permalink2 = true,
      coverImg = false,
      forceCoverImg = false,
      coverImgList = [],
      permalinkType = sidebarOption.resolveRule === "rewrites" ? "rules" : "simple",
      permalinkPrefix = "pages",
      permalinkRules = []
    } = autoFrontmatterOption;
    if (!pattern) autoFrontmatterOption.pattern = "**/*.md";
    autoFrontmatterOption.globOptions = {
      ...autoFrontmatterOption.globOptions,
      ignore: [...ignoreDir.autoFrontmatter, ...globOptions.ignore || []]
    };
    autoFrontmatterOption.transform = (frontmatter, fileInfo) => {
      let transformResult = {};
      if (permalink2 && !frontmatter.permalink?.trim()) {
        const permalinkObj = permalinkType === "simple" || !fileInfo.relativePath.includes("/") ? createSimplePermalink(permalinkPrefix) : createComplexPermalink(frontmatter.permalink, fileInfo, permalinkRules);
        transformResult = { ...transformResult, ...permalinkObj };
      }
      if (categories && !frontmatter.categories) {
        transformResult = { ...transformResult, ...createCategory(fileInfo, ["@fragment"]) };
      }
      if (coverImg && coverImgList.length) {
        if (!frontmatter.coverImg) {
          transformResult = { ...transformResult, ...createCoverImg(coverImgList) };
        } else if (frontmatter.coverImg && forceCoverImg) {
          transformResult = { ...transformResult, ...createCoverImg(coverImgList) };
        }
      }
      transformResult = transform?.({ ...frontmatter, ...transformResult }, fileInfo) || transformResult;
      return Object.keys(transformResult).length ? { ...frontmatter, ...transformResult } : void 0;
    };
    plugins.push(AutoFrontmatter(autoFrontmatterOption));
  }
  if (sidebar) {
    sidebarOption.ignoreList = [...sidebarOption?.ignoreList || [], ...ignoreDir.sidebar];
    plugins.push(Sidebar(sidebarOption));
  }
  if (permalink) {
    plugins.push(...Permalink(permalinkOption));
  }
  if (mdH1) {
    const selfBeforeInject = mdH1Option.beforeInject;
    mdH1Option.beforeInject = (frontmatter, id, title) => {
      if (["cataloguePage", "TkCataloguePage"].includes(frontmatter.layout) || frontmatter.catalogue) return false;
      if (["archivesPage", "TkArchivesPage"].includes(frontmatter.layout) || frontmatter.archivesPage) return false;
      if (frontmatter.titleTag) {
        return `${title} <TkTitleTag size="large">${frontmatter.titleTag}</TkTitleTag>`;
      }
      return selfBeforeInject?.(frontmatter, id, title);
    };
    mdH1Option.ignoreList = [...mdH1Option?.ignoreList || [], ...ignoreDir.mdH1];
    plugins.push(MdH1(mdH1Option));
  }
  if (docAnalysis) {
    docAnalysisOption.ignoreList = [...docAnalysisOption?.ignoreList || [], ...ignoreDir.docAnalysis];
    plugins.push(DocAnalysis(docAnalysisOption));
  }
  return plugins;
};
const registerTightPlugins = (vitePlugins, ignoreDir) => {
  const plugins = [];
  const { catalogueOption = {}, fileContentLoaderIgnore = [] } = vitePlugins || {};
  plugins.push(Catalogue(catalogueOption));
  const fileContentLoaderOptions = {
    pattern: ["**/*.md"],
    // 指定摘录格式
    excerpt: "<!-- more -->",
    includeSrc: true,
    transformData,
    transformRaw,
    themeConfigKey: "posts",
    globOptions: {
      ignore: [...ignoreDir.fileContentLoader, ...fileContentLoaderIgnore]
    }
  };
  plugins.push(FileContentLoader(fileContentLoaderOptions));
  return plugins;
};

export { registerPluginAndGet, registerTightPlugins };
