// vitepress-theme-teek@1.5.1/lib/components/theme/ThemeEnhance/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var reading = require('../../../../static/icons/reading.js');
require('vitepress');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useMediaQuery = require('../../../../composables/useMediaQuery.js');
require('../../../../composables/useZIndex.js');
var index = require('../../ConfigProvider/index.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../../common/Icon/src/index.vue2.js');
require('@iconify/vue');
var index_vue_vue_type_script_setup_true_name_Popover_lang = require('../../../common/Popover/src/index.vue2.js');
var namespace = require('./namespace.js');
var themeEnhance = require('./themeEnhance.js');
var LayoutSwitch_vue_vue_type_script_setup_true_name_LayoutSwitch_lang = require('./LayoutSwitch.vue2.js');
var LayoutPageWidthSlide_vue_vue_type_script_setup_true_name_LayoutPageWidthSlide_lang = require('./LayoutPageWidthSlide.vue2.js');
var LayoutDocWidthSlide_vue_vue_type_script_setup_true_name_LayoutDocWidthSlide_lang = require('./LayoutDocWidthSlide.vue2.js');
var ThemeColor_vue_vue_type_script_setup_true_name_ThemeColor_lang = require('./ThemeColor.vue2.js');
var Spotlight_vue_vue_type_script_setup_true_name_Spotlight_lang = require('./Spotlight.vue2.js');
var SpotlightStyle_vue_vue_type_script_setup_true_name_SpotlightStyle_lang = require('./SpotlightStyle.vue2.js');

var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "ThemeEnhance" },
  __name: "index",
  setup(__props) {
    const { getTeekConfigRef } = index.useTeekConfig();
    const themeEnhanceConfig = getTeekConfigRef("themeEnhance", { position: "top" });
    const isMobile = useMediaQuery.useMediaQuery(themeEnhance.mobileMaxWidthMedia);
    const disabledList = Vue.computed(() => {
      return {
        layoutSwitch: themeEnhanceConfig.value.layoutSwitch?.disabled ?? false,
        themeColor: themeEnhanceConfig.value.themeColor?.disabled ?? false,
        spotlight: themeEnhanceConfig.value.spotlight?.disabled ?? false
      };
    });
    return (_ctx, _cache) => {
      return !Vue.unref(isMobile) && Vue.unref(themeEnhanceConfig).position === "top" ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_Popover_lang.default), {
        key: 0,
        class: Vue.normalizeClass([Vue.unref(namespace.ns).b(), "flx-align-center"]),
        "popper-class": Vue.unref(namespace.ns).e("popover"),
        "y-offset": -15
      }, {
        reference: Vue.withCtx(() => [
          Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
            icon: Vue.unref(reading.default),
            size: 20
          }, null, 8, ["icon"])
        ]),
        default: Vue.withCtx(() => [
          Vue.createElementVNode(
            "div",
            {
              class: Vue.normalizeClass(Vue.unref(namespace.ns).e("content"))
            },
            [
              Vue.renderSlot(_ctx.$slots, "teek-theme-enhance-top"),
              !disabledList.value.layoutSwitch ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 0 },
                [
                  Vue.createVNode(LayoutSwitch_vue_vue_type_script_setup_true_name_LayoutSwitch_lang.default),
                  Vue.createVNode(LayoutPageWidthSlide_vue_vue_type_script_setup_true_name_LayoutPageWidthSlide_lang.default),
                  Vue.createVNode(LayoutDocWidthSlide_vue_vue_type_script_setup_true_name_LayoutDocWidthSlide_lang.default)
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true),
              !disabledList.value.themeColor ? (Vue.openBlock(), Vue.createBlock(ThemeColor_vue_vue_type_script_setup_true_name_ThemeColor_lang.default, { key: 1 })) : Vue.createCommentVNode("v-if", true),
              !disabledList.value.spotlight ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 2 },
                [
                  Vue.createVNode(Spotlight_vue_vue_type_script_setup_true_name_Spotlight_lang.default),
                  Vue.createVNode(SpotlightStyle_vue_vue_type_script_setup_true_name_SpotlightStyle_lang.default)
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true),
              Vue.renderSlot(_ctx.$slots, "teek-theme-enhance-bottom")
            ],
            2
            /* CLASS */
          )
        ]),
        _: 3
        /* FORWARDED */
      }, 8, ["class", "popper-class"])) : Vue.createCommentVNode("v-if", true);
    };
  }
});

exports.default = _sfc_main;
