// vitepress-theme-teek@1.5.1/lib/components/theme/ThemeEnhance/src/namespace.js
'use strict';

require('vue');
require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
require('../../../../composables/useZIndex.js');

const ns = useNamespace.useNamespace("theme-enhance");
const transitionName = ns.join("theme-enhance-slide");
const pageMaxWidthVar = ns.cssVarName("page-max-width");
const docMaxWidthVar = ns.cssVarName("doc-max-width");
const layoutModeStorageKey = ns.storageKey("layoutMode");
const pageMaxWidthSlideStorageKey = ns.storageKey("pageMaxWidthSlide");
const docMaxWidthSlideStorageKey = ns.storageKey("docMaxWidthSlide");
const themeColorStorageKey = ns.storageKey("themeColor");
const themeBgColorStorageKey = ns.storageKey("themeBgColor");
const spotlightStorageKey = ns.storageKey("spotlight");
const spotlightStyleStorageKey = ns.storageKey("spotlightStyle");

exports.docMaxWidthSlideStorageKey = docMaxWidthSlideStorageKey;
exports.docMaxWidthVar = docMaxWidthVar;
exports.layoutModeStorageKey = layoutModeStorageKey;
exports.ns = ns;
exports.pageMaxWidthSlideStorageKey = pageMaxWidthSlideStorageKey;
exports.pageMaxWidthVar = pageMaxWidthVar;
exports.spotlightStorageKey = spotlightStorageKey;
exports.spotlightStyleStorageKey = spotlightStyleStorageKey;
exports.themeBgColorStorageKey = themeBgColorStorageKey;
exports.themeColorStorageKey = themeColorStorageKey;
exports.transitionName = transitionName;
