// vitepress-theme-teek@1.5.1/lib/components/common/ImageViewer/src/ImageViewer.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var useEventListener = require('../../../../composables/useEventListener.js');
require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useDebounce = require('../../../../composables/useDebounce.js');
var useLocale = require('../../../../composables/useLocale.js');
var useZIndex = require('../../../../composables/useZIndex.js');
var close = require('../../../../static/icons/close.js');
var arrowLeft = require('../../../../static/icons/arrowLeft.js');
var arrowRight = require('../../../../static/icons/arrowRight.js');
var fullscreen = require('../../../../static/icons/fullscreen.js');
var refreshLeft = require('../../../../static/icons/refreshLeft.js');
var refreshRight = require('../../../../static/icons/refreshRight.js');
var scaleToOriginal = require('../../../../static/icons/scaleToOriginal.js');
var zoomIn = require('../../../../static/icons/zoomIn.js');
var zoomOut = require('../../../../static/icons/zoomOut.js');
var index = require('../../FocusTrap/src/index.vue.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../Icon/src/index.vue2.js');
require('@iconify/vue');

const _hoisted_1 = ["src", "crossorigin"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "ImageViewer" },
  __name: "ImageViewer",
  props: {
    urlList: { default: () => [] },
    zIndex: {},
    initialIndex: { default: 0 },
    infinite: { type: Boolean, default: true },
    hideOnClickModal: { type: Boolean, default: false },
    teleported: { type: Boolean },
    closeOnPressEscape: { type: Boolean, default: true },
    zoomRate: { default: 1.2 },
    minScale: { default: 0.2 },
    maxScale: { default: 7 },
    showProgress: { type: Boolean, default: false },
    crossorigin: {}
  },
  emits: ["close", "switch", "rotate"],
  setup(__props, { expose: __expose, emit: __emit }) {
    const modes = {
      CONTAIN: {
        name: "contain",
        icon: fullscreen.default
      },
      ORIGINAL: {
        name: "original",
        icon: scaleToOriginal.default
      }
    };
    const props = __props;
    const emit = __emit;
    let prevOverflow = "";
    const ns = useNamespace.useNamespace("image-viewer");
    const { t } = useLocale.useLocale();
    const { nextZIndex } = useZIndex.useZIndex();
    const wrapper = Vue.ref();
    const imgRefs = Vue.ref([]);
    const scopeEventListener = Vue.effectScope();
    const loading = Vue.ref(true);
    const activeIndex = Vue.ref(props.initialIndex);
    const mode = Vue.shallowRef(modes.CONTAIN);
    const transform = Vue.ref({
      scale: 1,
      deg: 0,
      offsetX: 0,
      offsetY: 0,
      enableTransition: false
    });
    const zIndex = Vue.ref(props.zIndex ?? nextZIndex());
    const isSingle = Vue.computed(() => {
      const { urlList } = props;
      return urlList.length <= 1;
    });
    const isFirst = Vue.computed(() => activeIndex.value === 0);
    const isLast = Vue.computed(() => activeIndex.value === props.urlList.length - 1);
    const currentImg = Vue.computed(() => props.urlList[activeIndex.value]);
    const arrowPrevKls = Vue.computed(() => [ns.e("btn"), ns.e("prev"), ns.is("disabled", !props.infinite && isFirst.value)]);
    const arrowNextKls = Vue.computed(() => [ns.e("btn"), ns.e("next"), ns.is("disabled", !props.infinite && isLast.value)]);
    const imgStyle = Vue.computed(() => {
      const { scale, deg, offsetX, offsetY, enableTransition } = transform.value;
      let translateX = offsetX / scale;
      let translateY = offsetY / scale;
      const radian = deg * Math.PI / 180;
      const cosRadian = Math.cos(radian);
      const sinRadian = Math.sin(radian);
      translateX = translateX * cosRadian + translateY * sinRadian;
      translateY = translateY * cosRadian - offsetX / scale * sinRadian;
      const style = {
        transform: `scale(${scale}) rotate(${deg}deg) translate(${translateX}px, ${translateY}px)`,
        transition: enableTransition ? "transform .3s" : ""
      };
      if (mode.value.name === modes.CONTAIN.name) {
        style.maxWidth = style.maxHeight = "100%";
      }
      return style;
    });
    const progress = Vue.computed(() => `${activeIndex.value + 1} / ${props.urlList.length}`);
    const hide = () => {
      unregisterEventListener();
      stopWheelListener?.();
      document.body.style.overflow = prevOverflow;
      emit("close");
    };
    const registerEventListener = () => {
      const keydownHandler = useDebounce.useDebounce((e) => {
        switch (e.code) {
          // ESC
          case "Escape":
            props.closeOnPressEscape && hide();
            break;
          // SPACE
          case "Space":
            toggleMode();
            break;
          // LEFT_ARROW
          case "ArrowLeft":
            prev();
            break;
          // UP_ARROW
          case "ArrowUp":
            handleActions("zoomIn");
            break;
          // RIGHT_ARROW
          case "ArrowRight":
            next();
            break;
          // DOWN_ARROW
          case "ArrowDown":
            handleActions("zoomOut");
            break;
        }
      });
      const mousewheelHandler = useDebounce.useDebounce((e) => {
        const delta = e.deltaY || e.deltaX;
        handleActions(delta < 0 ? "zoomIn" : "zoomOut", {
          zoomRate: props.zoomRate,
          enableTransition: false
        });
      });
      scopeEventListener.run(() => {
        useEventListener.useEventListener(document, "keydown", keydownHandler);
        useEventListener.useEventListener(document, "wheel", mousewheelHandler);
      });
    };
    const unregisterEventListener = () => {
      scopeEventListener.stop();
    };
    const handleImgLoad = () => {
      loading.value = false;
    };
    const handleImgError = (e) => {
      loading.value = false;
      e.target.alt = t("tk.image.error");
    };
    const handleMouseDown = (e) => {
      if (loading.value || e.button !== 0 || !wrapper.value) return;
      transform.value.enableTransition = false;
      const { offsetX, offsetY } = transform.value;
      const startX = e.pageX;
      const startY = e.pageY;
      const dragHandler = useDebounce.useDebounce((ev) => {
        transform.value = {
          ...transform.value,
          offsetX: offsetX + ev.pageX - startX,
          offsetY: offsetY + ev.pageY - startY
        };
      });
      const removeDragHandler = () => {
        document.removeEventListener("mousemove", dragHandler);
        document.removeEventListener("mouseup", removeDragHandler);
      };
      document.addEventListener("mousemove", dragHandler);
      document.addEventListener("mouseup", removeDragHandler);
      e.preventDefault();
    };
    const reset = () => {
      transform.value = {
        scale: 1,
        deg: 0,
        offsetX: 0,
        offsetY: 0,
        enableTransition: false
      };
    };
    const toggleMode = () => {
      if (loading.value) return;
      const modeNames = Object.keys(modes);
      const modeValues = Object.values(modes);
      const currentMode = mode.value.name;
      const index = modeValues.findIndex((i) => i.name === currentMode);
      const nextIndex = (index + 1) % modeNames.length;
      mode.value = modes[modeNames[nextIndex]];
      reset();
    };
    const setActiveItem = (index) => {
      const len = props.urlList.length;
      activeIndex.value = (index + len) % len;
    };
    const prev = () => {
      if (isFirst.value && !props.infinite) return;
      setActiveItem(activeIndex.value - 1);
    };
    const next = () => {
      if (isLast.value && !props.infinite) return;
      setActiveItem(activeIndex.value + 1);
    };
    const handleActions = (action, options = {}) => {
      if (loading.value) return;
      const { minScale, maxScale } = props;
      const { zoomRate, rotateDeg, enableTransition } = {
        zoomRate: props.zoomRate,
        rotateDeg: 90,
        enableTransition: true,
        ...options
      };
      switch (action) {
        case "zoomOut":
          if (transform.value.scale > minScale) {
            transform.value.scale = Number.parseFloat((transform.value.scale / zoomRate).toFixed(3));
          }
          break;
        case "zoomIn":
          if (transform.value.scale < maxScale) {
            transform.value.scale = Number.parseFloat((transform.value.scale * zoomRate).toFixed(3));
          }
          break;
        case "clockwise":
          transform.value.deg += rotateDeg;
          emit("rotate", transform.value.deg);
          break;
        case "anticlockwise":
          transform.value.deg -= rotateDeg;
          emit("rotate", transform.value.deg);
          break;
      }
      transform.value.enableTransition = enableTransition;
    };
    const onFocusoutPrevented = (event) => {
      if (event.detail?.focusReason === "pointer") {
        event.preventDefault();
      }
    };
    const onCloseRequested = () => {
      if (props.closeOnPressEscape) {
        hide();
      }
    };
    const wheelHandler = (e) => {
      if (!e.ctrlKey) return;
      if (e.deltaY < 0) {
        e.preventDefault();
        return false;
      } else if (e.deltaY > 0) {
        e.preventDefault();
        return false;
      }
    };
    Vue.watch(currentImg, () => {
      Vue.nextTick(() => {
        const $img = imgRefs.value[0];
        if (!$img?.complete) {
          loading.value = true;
        }
      });
    });
    Vue.watch(activeIndex, (val) => {
      reset();
      emit("switch", val);
    });
    registerEventListener();
    const stopWheelListener = useEventListener.useEventListener(document, "wheel", wheelHandler, { passive: false });
    Vue.onMounted(() => {
      prevOverflow = document.body.style.overflow;
      document.body.style.overflow = "hidden";
    });
    __expose({
      /**
       * 手动切换图片
       */
      setActiveItem
    });
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createBlock(Vue.Teleport, {
        disabled: !__props.teleported,
        to: "body"
      }, [
        Vue.createVNode(Vue.Transition, {
          name: "viewer-fade",
          appear: ""
        }, {
          default: Vue.withCtx(() => [
            Vue.createElementVNode(
              "div",
              {
                ref_key: "wrapper",
                ref: wrapper,
                tabindex: -1,
                class: Vue.normalizeClass(Vue.unref(ns).e("wrapper")),
                style: Vue.normalizeStyle({ zIndex: zIndex.value })
              },
              [
                Vue.createVNode(Vue.unref(index.default), {
                  loop: "",
                  trapped: "",
                  "focus-trap-el": wrapper.value,
                  "focus-start-el": "container",
                  onFocusoutPrevented,
                  onReleaseRequested: onCloseRequested
                }, {
                  default: Vue.withCtx(() => [
                    Vue.createElementVNode(
                      "div",
                      {
                        class: Vue.normalizeClass(Vue.unref(ns).e("mask")),
                        onClick: _cache[0] || (_cache[0] = Vue.withModifiers(($event) => __props.hideOnClickModal && hide(), ["self"]))
                      },
                      null,
                      2
                      /* CLASS */
                    ),
                    Vue.createCommentVNode(" CLOSE "),
                    Vue.createElementVNode(
                      "span",
                      {
                        class: Vue.normalizeClass([Vue.unref(ns).e("btn"), Vue.unref(ns).e("close")]),
                        onClick: hide
                      },
                      [
                        Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), { icon: Vue.unref(close.default) }, null, 8, ["icon"])
                      ],
                      2
                      /* CLASS */
                    ),
                    Vue.createCommentVNode(" ARROW "),
                    !isSingle.value ? (Vue.openBlock(), Vue.createElementBlock(
                      Vue.Fragment,
                      { key: 0 },
                      [
                        Vue.createElementVNode(
                          "span",
                          {
                            class: Vue.normalizeClass(arrowPrevKls.value),
                            onClick: prev
                          },
                          [
                            Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), { icon: Vue.unref(arrowLeft.default) }, null, 8, ["icon"])
                          ],
                          2
                          /* CLASS */
                        ),
                        Vue.createElementVNode(
                          "span",
                          {
                            class: Vue.normalizeClass(arrowNextKls.value),
                            onClick: next
                          },
                          [
                            Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), { icon: Vue.unref(arrowRight.default) }, null, 8, ["icon"])
                          ],
                          2
                          /* CLASS */
                        )
                      ],
                      64
                      /* STABLE_FRAGMENT */
                    )) : Vue.createCommentVNode("v-if", true),
                    _ctx.$slots.progress || __props.showProgress ? (Vue.openBlock(), Vue.createElementBlock(
                      "div",
                      {
                        key: 1,
                        class: Vue.normalizeClass([Vue.unref(ns).e("btn"), Vue.unref(ns).e("progress")])
                      },
                      [
                        Vue.renderSlot(_ctx.$slots, "progress", {
                          activeIndex: activeIndex.value,
                          total: __props.urlList.length
                        }, () => [
                          Vue.createTextVNode(
                            Vue.toDisplayString(progress.value),
                            1
                            /* TEXT */
                          )
                        ])
                      ],
                      2
                      /* CLASS */
                    )) : Vue.createCommentVNode("v-if", true),
                    Vue.createCommentVNode(" ACTIONS "),
                    Vue.createElementVNode(
                      "div",
                      {
                        class: Vue.normalizeClass([Vue.unref(ns).e("btn"), Vue.unref(ns).e("actions")])
                      },
                      [
                        Vue.createElementVNode(
                          "div",
                          {
                            class: Vue.normalizeClass(Vue.unref(ns).e("actions__inner"))
                          },
                          [
                            Vue.renderSlot(_ctx.$slots, "toolbar", {
                              actions: handleActions,
                              prev,
                              next,
                              reset: toggleMode,
                              activeIndex: activeIndex.value,
                              setActiveItem
                            }, () => [
                              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                                icon: Vue.unref(zoomOut.default),
                                onClick: _cache[1] || (_cache[1] = ($event) => handleActions("zoomOut"))
                              }, null, 8, ["icon"]),
                              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                                icon: Vue.unref(zoomIn.default),
                                onClick: _cache[2] || (_cache[2] = ($event) => handleActions("zoomIn"))
                              }, null, 8, ["icon"]),
                              Vue.createElementVNode(
                                "i",
                                {
                                  class: Vue.normalizeClass(Vue.unref(ns).e("actions__divider"))
                                },
                                null,
                                2
                                /* CLASS */
                              ),
                              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                                icon: mode.value.icon,
                                onClick: toggleMode
                              }, null, 8, ["icon"]),
                              Vue.createElementVNode(
                                "i",
                                {
                                  class: Vue.normalizeClass(Vue.unref(ns).e("actions__divider"))
                                },
                                null,
                                2
                                /* CLASS */
                              ),
                              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                                icon: Vue.unref(refreshLeft.default),
                                onClick: _cache[3] || (_cache[3] = ($event) => handleActions("anticlockwise"))
                              }, null, 8, ["icon"]),
                              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                                icon: Vue.unref(refreshRight.default),
                                onClick: _cache[4] || (_cache[4] = ($event) => handleActions("clockwise"))
                              }, null, 8, ["icon"])
                            ])
                          ],
                          2
                          /* CLASS */
                        )
                      ],
                      2
                      /* CLASS */
                    ),
                    Vue.createCommentVNode(" CANVAS "),
                    Vue.createElementVNode(
                      "div",
                      {
                        class: Vue.normalizeClass(Vue.unref(ns).e("canvas"))
                      },
                      [
                        (Vue.openBlock(true), Vue.createElementBlock(
                          Vue.Fragment,
                          null,
                          Vue.renderList(__props.urlList, (url, i) => {
                            return Vue.withDirectives((Vue.openBlock(), Vue.createElementBlock("img", {
                              ref_for: true,
                              ref: (el) => imgRefs.value[i] = el,
                              key: url,
                              src: url,
                              style: Vue.normalizeStyle(imgStyle.value),
                              class: Vue.normalizeClass([Vue.unref(ns).e("img"), "image-viewer__img"]),
                              crossorigin: __props.crossorigin,
                              onLoad: handleImgLoad,
                              onError: handleImgError,
                              onMousedown: handleMouseDown
                            }, null, 46, _hoisted_1)), [
                              [Vue.vShow, i === activeIndex.value]
                            ]);
                          }),
                          128
                          /* KEYED_FRAGMENT */
                        ))
                      ],
                      2
                      /* CLASS */
                    ),
                    Vue.renderSlot(_ctx.$slots, "default")
                  ]),
                  _: 3
                  /* FORWARDED */
                }, 8, ["focus-trap-el"])
              ],
              6
              /* CLASS, STYLE */
            )
          ]),
          _: 3
          /* FORWARDED */
        })
      ], 8, ["disabled"]);
    };
  }
});

exports.default = _sfc_main;
