// vitepress-theme-teek/es/components/theme/ConfigProvider/index.mjs
import { defineComponent, provide, h, inject, computed, unref, ref, watch, nextTick } from 'vue';
import { useData, useRoute } from 'vitepress';
import { isFunction, isObject, isClient } from '../../../helper/dist/index.mjs';
import { useAnchorScroll } from '../../../composables/useAnchorScroll.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useMediaQuery } from '../../../composables/useMediaQuery.mjs';
import { useViewTransition } from '../../../composables/useViewTransition.mjs';
import '../../../composables/useZIndex.mjs';
import { emptyPost } from '../../../config/post/helper.mjs';

const postsContext = Symbol("posts");
const teekConfigContext = Symbol("teekConfig");
const TeekConfigProvider = (layout) => {
  return defineComponent({
    name: "TeekConfigProvider",
    setup(_, { slots }) {
      provide(postsContext, useAllPosts());
      useAnchorScroll().startWatch();
      const { getTeekConfig } = useTeekConfig();
      const viewTransitionConfig = getTeekConfig("viewTransition", {
        enabled: true,
        mode: "out-in",
        easing: "ease-in"
      });
      viewTransitionConfig.enabled && useViewTransition(viewTransitionConfig);
      return () => h(layout, null, slots);
    }
  });
};
const useTeekConfig = () => {
  const { theme, frontmatter } = useData();
  const teekConfigProvide = inject(teekConfigContext, {});
  const getTeekConfig = (key, defaultValue) => {
    let dv = defaultValue;
    if (isFunction(defaultValue)) dv = defaultValue();
    if (!key) {
      return { ...dv, ...theme.value, ...frontmatter.value, ...frontmatter.value.tk, ...unref(teekConfigProvide) };
    }
    const valueFromTheme = theme.value[key];
    const valueFromFrontmatter = frontmatter.value.tk?.[key] ?? frontmatter.value[key];
    const valueFromInject = unref(teekConfigProvide)?.[key];
    if (isObject(valueFromTheme) || isObject(valueFromFrontmatter) || isObject(valueFromInject)) {
      return { ...dv, ...valueFromTheme, ...valueFromFrontmatter, ...valueFromInject };
    }
    return valueFromInject ?? valueFromFrontmatter ?? valueFromTheme ?? dv;
  };
  const getTeekConfigRef = (key, defaultValue) => {
    return computed(() => getTeekConfig(key, defaultValue));
  };
  return { getTeekConfig, getTeekConfigRef };
};
const usePageState = () => {
  const { frontmatter } = useData();
  const isHomePage = computed(
    () => !isCategoriesPage.value && !isTagsPage.value && frontmatter.value.layout === "home"
  );
  const isCategoriesPage = computed(() => !!frontmatter.value.categoriesPage);
  const isTagsPage = computed(() => !!frontmatter.value.tagsPage);
  const isArchivesPage = computed(() => !!frontmatter.value.archivesPage);
  const isCataloguePage = computed(() => !!frontmatter.value.catalogue);
  const isArticleOverviewPage = computed(() => !!frontmatter.value.articleOverviewPage);
  const isLoginUrl = computed(() => !!frontmatter.value.loginPage);
  const isRiskLinkPage = computed(() => !!frontmatter.value.riskLinkPage);
  return {
    isHomePage,
    isCategoriesPage,
    isTagsPage,
    isArchivesPage,
    isCataloguePage,
    isArticleOverviewPage,
    isLoginUrl,
    isRiskLinkPage
  };
};
const usePagePath = () => {
  const { getTeekConfigRef } = useTeekConfig();
  const { localeIndex, site } = useData();
  const posts = usePosts();
  const categoryConfig = getTeekConfigRef("category", { path: "/categories" });
  const tagConfig = getTeekConfigRef("tag", { path: "/tags" });
  const categoryPath = computed(() => {
    const localeIndexConst = localeIndex.value;
    const localeName = localeIndexConst !== "root" ? `/${localeIndexConst}` : "";
    return `${localeName}${categoryConfig.value.path}${site.value.cleanUrls ? "" : ".html"}`;
  });
  const tagPath = computed(() => {
    const localeIndexConst = localeIndex.value;
    const localeName = localeIndexConst !== "root" ? `/${localeIndexConst}` : "";
    return `${localeName}${tagConfig.value.path}${site.value.cleanUrls ? "" : ".html"}`;
  });
  const postPagePath = computed(() => {
    let archivesUrl = "";
    let articleOverviewUrl = "";
    let loginUrl = "";
    let riskLinkUrl = "";
    posts.value.allPosts.forEach((item) => {
      const {
        frontmatter: { layout, archivesPage, articleOverviewPage, loginPage, riskLinkPage },
        url
      } = item;
      const isPageLayout = layout === "page";
      if (layout === "TkCataloguePage" || isPageLayout && archivesPage === true) archivesUrl = url;
      if (layout === "TkArticleOverviewPage" || isPageLayout && articleOverviewPage === true) {
        articleOverviewUrl = url;
      }
      if (layout === false && loginPage === true) loginUrl = url;
      if (layout === false && riskLinkPage === true) riskLinkUrl = url;
    });
    return { archivesUrl, articleOverviewUrl, loginUrl, riskLinkUrl };
  });
  return {
    categoryPath,
    tagPath,
    archivesPath: computed(() => postPagePath.value.archivesUrl),
    articleOverviewPath: computed(() => postPagePath.value.articleOverviewUrl),
    loginPath: computed(() => postPagePath.value.loginUrl),
    riskLinkPath: computed(() => postPagePath.value.riskLinkUrl)
  };
};
const useAllPosts = () => {
  const { theme } = useData();
  const posts = theme.value.posts;
  return posts || emptyPost;
};
const usePosts = () => {
  const { localeIndex } = useData();
  const posts = useAllPosts();
  return computed(() => posts.locales?.[localeIndex.value] || posts);
};
const useTagColor = () => {
  const { getTeekConfigRef } = useTeekConfig();
  return getTeekConfigRef("tagColor", [
    { border: "#bfdbfe", bg: "#eff6ff", text: "#2563eb" },
    { border: "#e9d5ff", bg: "#faf5ff", text: "#9333ea" },
    { border: "#fbcfe8", bg: "#fdf2f8", text: "#db2777" },
    { border: "#a7f3d0", bg: "#ecfdf5", text: "#059669" },
    { border: "#fde68a", bg: "#fffbeb", text: "#d97706" },
    { border: "#a5f3fc", bg: "#ecfeff", text: "#0891b2" },
    { border: "#c7d2fe", bg: "#eef2ff", text: "#4f46e5" }
  ]);
};
const useWindowTransitionConfig = (condition) => {
  const { getTeekConfigRef } = useTeekConfig();
  const windowTransitionConfig = getTeekConfigRef("windowTransition", true);
  return computed(() => {
    const windowTransition = windowTransitionConfig.value;
    if (windowTransition === void 0) return true;
    return isObject(windowTransition) ? condition?.(windowTransition) ?? true : windowTransition !== false;
  });
};
const useCommon = () => {
  const isMobile = useMediaQuery("(max-width: 768px)");
  return { isMobile };
};
const useSidebar = () => {
  const hasSidebar = ref(true);
  const route = useRoute();
  watch(
    () => route.path,
    async () => {
      if (!isClient) return;
      await nextTick();
      const sidebarDom = document.querySelector(".VPSidebar");
      if (sidebarDom) hasSidebar.value = true;
      else hasSidebar.value = false;
    },
    { immediate: true, flush: "post" }
  );
  return { hasSidebar };
};

export { TeekConfigProvider, postsContext, teekConfigContext, useAllPosts, useCommon, usePagePath, usePageState, usePosts, useSidebar, useTagColor, useTeekConfig, useWindowTransitionConfig };
