// vitepress-theme-teek@1.5.1/lib/components/common/Pagination/src/components/jumper.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('vitepress');
var useNamespace = require('../../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../../composables/useLocale.js');
require('../../../../../composables/useZIndex.js');
var usePagination = require('../usePagination.js');

const _hoisted_1 = ["disabled"];
const _hoisted_2 = ["disabled", "aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "PaginationJumper" },
  __name: "jumper",
  props: {
    size: { default: "default" }
  },
  setup(__props) {
    const ns = useNamespace.useNamespace("pagination");
    const { t } = useLocale.useLocale();
    const { pageCount, disabled, currentPage, changeEvent } = usePagination.usePagination();
    const userInput = Vue.ref(currentPage?.value || 1);
    const handleChange = (event) => {
      let value = event.target?.value || userInput.value;
      if (value < 1) value = 1;
      if (pageCount?.value && value > pageCount.value) value = pageCount.value;
      const val = Math.trunc(+value);
      changeEvent?.(val);
      userInput.value = val;
    };
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("span", {
        class: Vue.normalizeClass(Vue.unref(ns).e("jump")),
        disabled: Vue.unref(disabled)
      }, [
        Vue.createElementVNode(
          "span",
          {
            class: Vue.normalizeClass([Vue.unref(ns).e("goto")])
          },
          Vue.toDisplayString(Vue.unref(t)("tk.pagination.goto")),
          3
          /* TEXT, CLASS */
        ),
        Vue.withDirectives(Vue.createElementVNode("input", {
          type: "number",
          "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => userInput.value = $event),
          disabled: Vue.unref(disabled),
          onChange: handleChange,
          "aria-label": Vue.unref(t)("tk.pagination.page"),
          class: Vue.normalizeClass([Vue.unref(ns).e("input"), Vue.unref(ns).em("input", __props.size)])
        }, null, 42, _hoisted_2), [
          [Vue.vModelText, userInput.value]
        ]),
        Vue.createElementVNode(
          "span",
          {
            class: Vue.normalizeClass([Vue.unref(ns).e("classifier")])
          },
          Vue.toDisplayString(Vue.unref(t)("tk.pagination.pageClassifier")),
          3
          /* TEXT, CLASS */
        )
      ], 10, _hoisted_1);
    };
  }
});

exports.default = _sfc_main;
