// vitepress-theme-teek@1.5.1/lib/components/theme/HomeCategoryCard/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var index$1 = require('../../../../helper/dist/index.js');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var category = require('../../../../static/icons/category.js');
var homePostList = require('../../HomePostList/src/homePostList.js');
var index = require('../../ConfigProvider/index.js');
var home = require('../../Home/src/home.js');
var index_vue_vue_type_script_setup_true_name_PageCard_lang = require('../../../common/PageCard/src/index.vue2.js');

const _hoisted_1 = ["onClick", "aria-label"];
const _hoisted_2 = { class: "sle" };
const _hoisted_3 = ["href", "aria-label"];
const _hoisted_4 = ["aria-label"];
const categoryKey = "category";
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "HomeCategoryCard" },
  __name: "index",
  props: {
    categoriesPage: { type: Boolean, default: false }
  },
  setup(__props) {
    const ns = useNamespace.useNamespace("category");
    const { t } = useLocale.useLocale();
    const { getTeekConfigRef } = index.useTeekConfig();
    const { categoryPath } = index.usePagePath();
    const categoryConfig = getTeekConfigRef("category", {
      path: "/categories",
      pageTitle: t("tk.categoryCard.pageTitle", { icon: category.default }),
      homeTitle: t("tk.categoryCard.homeTitle", { icon: category.default }),
      emptyLabel: t("tk.categoryCard.emptyLabel"),
      moreLabel: t("tk.categoryCard.moreLabel"),
      limit: 5,
      autoPage: false,
      pageSpeed: 4e3
    });
    const posts = index.usePosts();
    const pageNum = Vue.ref(1);
    const categories = Vue.computed(() => posts.value.groupCards.categories);
    const currentCategories = Vue.computed(() => {
      const { limit } = categoryConfig.value;
      const c = categories.value;
      const p = pageNum.value;
      return __props.categoriesPage ? c : c.slice((p - 1) * limit, p * limit);
    });
    const finalTitle = Vue.computed(() => {
      const { pageTitle, homeTitle } = categoryConfig.value;
      const pt = index$1.isFunction(pageTitle) ? pageTitle(category.default) : pageTitle;
      const ht = index$1.isFunction(homeTitle) ? homeTitle(category.default) : homeTitle;
      return { pt, ht };
    });
    const updatePostListData = Vue.inject(home.postDataUpdateSymbol, () => {
    });
    const router = vitepress.useRouter();
    const selectedCategory = Vue.ref("");
    const handleSwitchCategory = (category = "") => {
      const { pathname, searchParams } = new URL(window.location.href);
      const categoriesPathConst = vitepress.withBase(categoryPath.value);
      const inCategoriesPage = categoriesPathConst === pathname;
      searchParams.delete(homePostList.pageNumKey);
      searchParams.append(homePostList.pageNumKey, "1");
      searchParams.delete(categoryKey);
      if (category) searchParams.append(categoryKey, category);
      const searchParamsStr = category ? `?${searchParams.toString()}` : "";
      if (inCategoriesPage && selectedCategory.value === category) return;
      selectedCategory.value = category;
      if (!inCategoriesPage) return router.go(categoriesPathConst + searchParamsStr);
      window.history.pushState({}, "", pathname + searchParamsStr);
      updatePostListData();
    };
    Vue.onMounted(() => {
      const { searchParams } = new URL(window.location.href);
      const category = searchParams.get(categoryKey);
      if (category) selectedCategory.value = category;
    });
    Vue.watch(
      () => __props.categoriesPage,
      () => {
        if (!__props.categoriesPage) {
          selectedCategory.value = "";
          return;
        }
        const { searchParams } = new URL(window.location.href);
        const category = searchParams.get(categoryKey);
        if (category && selectedCategory.value !== category) selectedCategory.value = category;
      }
    );
    const itemRefs = Vue.ref([]);
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_PageCard_lang.default), {
        page: !__props.categoriesPage,
        modelValue: pageNum.value,
        "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => pageNum.value = $event),
        pageSize: Vue.unref(categoryConfig).limit,
        total: categories.value.length,
        title: finalTitle.value[__props.categoriesPage ? "pt" : "ht"],
        titleClick: handleSwitchCategory,
        autoPage: Vue.unref(categoryConfig).autoPage,
        pageSpeed: Vue.unref(categoryConfig).pageSpeed,
        class: Vue.normalizeClass([Vue.unref(ns).b(), Vue.unref(ns).is("page", __props.categoriesPage)]),
        "aria-label": Vue.unref(t)("tk.categoryCard.label")
      }, {
        default: Vue.withCtx(({ transitionName }) => [
          categories.value.length ? (Vue.openBlock(), Vue.createBlock(Vue.TransitionGroup, {
            key: 0,
            name: transitionName,
            tag: "div",
            mode: "out-in",
            class: Vue.normalizeClass(`${Vue.unref(ns).e("list")} flx-column`),
            "aria-label": Vue.unref(t)("tk.categoryCard.listLabel")
          }, {
            default: Vue.withCtx(() => [
              (Vue.openBlock(true), Vue.createElementBlock(
                Vue.Fragment,
                null,
                Vue.renderList(currentCategories.value, (item, index) => {
                  return Vue.openBlock(), Vue.createElementBlock("a", {
                    ref_for: true,
                    ref_key: "itemRefs",
                    ref: itemRefs,
                    key: item.name,
                    onClick: ($event) => handleSwitchCategory(item.name),
                    class: Vue.normalizeClass([{ active: item.name === selectedCategory.value }, "hover-color"]),
                    style: Vue.normalizeStyle(`top: ${index * itemRefs.value?.[index]?.getBoundingClientRect().height || 0}px`),
                    "aria-label": item.name
                  }, [
                    Vue.createElementVNode(
                      "span",
                      _hoisted_2,
                      Vue.toDisplayString(item.name),
                      1
                      /* TEXT */
                    ),
                    Vue.createElementVNode(
                      "span",
                      null,
                      Vue.toDisplayString(item.length),
                      1
                      /* TEXT */
                    )
                  ], 14, _hoisted_1);
                }),
                128
                /* KEYED_FRAGMENT */
              )),
              !__props.categoriesPage && Vue.unref(categoryConfig).limit < categories.value.length ? (Vue.openBlock(), Vue.createElementBlock("a", {
                key: 0,
                href: Vue.unref(vitepress.withBase)(Vue.unref(categoryPath)),
                "aria-label": Vue.unref(categoryConfig).moreLabel
              }, Vue.toDisplayString(Vue.unref(categoryConfig).moreLabel), 9, _hoisted_3)) : Vue.createCommentVNode("v-if", true)
            ]),
            _: 1
            /* STABLE */
          }, 8, ["name", "class", "aria-label"])) : (Vue.openBlock(), Vue.createElementBlock("div", {
            key: 1,
            class: Vue.normalizeClass(Vue.unref(ns).m("empty")),
            "aria-label": Vue.unref(categoryConfig).emptyLabel
          }, Vue.toDisplayString(Vue.unref(categoryConfig).emptyLabel), 11, _hoisted_4))
        ]),
        _: 1
        /* STABLE */
      }, 8, ["page", "modelValue", "pageSize", "total", "title", "autoPage", "pageSpeed", "class", "aria-label"]);
    };
  }
});

exports.default = _sfc_main;
