// vitepress-theme-teek@1.5.1/lib/components/theme/DemoCode/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var useClipboard = require('../../../../composables/useClipboard.js');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var playground = require('../../../../static/icons/playground.js');
var github = require('../../../../static/icons/github.js');
var copy = require('../../../../static/icons/copy.js');
var code = require('../../../../static/icons/code.js');
var caretTop = require('../../../../static/icons/caretTop.js');
var method = require('../../../common/Message/src/method.js');
var index_vue_vue_type_script_setup_true_name_TransitionCollapse_lang = require('../../../common/TransitionCollapse/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../../common/Icon/src/index.vue2.js');
require('@iconify/vue');

const _hoisted_1 = ["innerHTML"];
const _hoisted_2 = ["innerHTML"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "DemoCode" },
  __name: "index",
  props: {
    source: {},
    rawSource: {},
    path: {},
    description: {},
    options: {},
    effect: {}
  },
  setup(__props) {
    const props = __props;
    const ns = useNamespace.useNamespace("demo-code");
    const { t } = useLocale.useLocale();
    const { copy: copy$1, copied, isSupported } = useClipboard.useClipboard();
    const { frontmatter, isDark } = vitepress.useData();
    const {
      playgroundUrl = "",
      playgroundMainFileName = "App.vue",
      githubUrl = "",
      playgroundButtonTip = t("tk.demoCode.playground"),
      githubButtonTip = t("tk.demoCode.github"),
      copyButtonTip = t("tk.demoCode.copy"),
      collapseSourceButtonTip = t("tk.demoCode.collapseSource"),
      expandSourceButtonTip = t("tk.demoCode.expandSource")
    } = { ...JSON.parse(decodeURIComponent(props.options)), ...frontmatter.value.demo };
    const decodeSource = Vue.computed(() => decodeURIComponent(props.source));
    const decodeRawSource = Vue.computed(() => decodeURIComponent(props.rawSource));
    const decodedDescription = Vue.computed(() => decodeURIComponent(props.description));
    const effect = Vue.computed(() => props.effect === "true");
    const moduleFiles = undefined("/examples/**/*.vue", { eager: true });
    const DemoComponent = Vue.defineAsyncComponent(async () => {
      try {
        const key = Object.keys(moduleFiles).find((i) => i.endsWith(`/${props.path}`));
        return moduleFiles[key];
      } catch (error) {
        console.error(`[Teek Error] Failed to load component: '/${props.path}'`, error);
      }
    });
    const sourceVisible = Vue.ref(false);
    const handleToggleSourceVisible = (bol) => {
      if (bol !== void 0) sourceVisible.value = bol;
      else sourceVisible.value = !sourceVisible.value;
    };
    const handleEditPlayground = () => {
      const encoded = getPlaygroundEncoded(props.source);
      const darkParam = isDark.value ? "?theme=dark" : "";
      const link = playgroundUrl.includes("?") ? `${playgroundUrl}${darkParam.replace("?", "&")}` : `${playgroundUrl}${darkParam}`;
      const url = `${link.replace(/\/$/, "")}/#${encoded}`;
      window.open(url, "_blank");
    };
    const getPlaygroundEncoded = (source) => {
      const code = decodeURIComponent(source);
      const originCode = {
        [playgroundMainFileName]: code
      };
      const encoded = btoa(JSON.stringify(originCode));
      return encoded;
    };
    const handleEditGithub = () => {
      const url = `${githubUrl}/${props.path}`;
      window.open(url, "_blank");
    };
    const copyCode = async () => {
      if (!isSupported) console.error(t("tk.demoCode.notSupport"));
      await copy$1(decodeRawSource.value);
      copied.value ? method.default.success({
        message: t("tk.demoCode.copySuccess"),
        plain: true
      }) : method.default.error({
        message: t("tk.demoCode.copyFail"),
        plain: true
      });
    };
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock(
        Vue.Fragment,
        null,
        [
          decodedDescription.value ? (Vue.openBlock(), Vue.createElementBlock("div", {
            key: 0,
            class: Vue.normalizeClass(Vue.unref(ns).b("description")),
            innerHTML: decodedDescription.value
          }, null, 10, _hoisted_1)) : Vue.createCommentVNode("v-if", true),
          effect.value ? (Vue.openBlock(), Vue.createBlock(Vue.resolveDynamicComponent(Vue.unref(DemoComponent)), { key: 1 })) : (Vue.openBlock(), Vue.createElementBlock(
            "div",
            {
              key: 2,
              class: Vue.normalizeClass(Vue.unref(ns).b())
            },
            [
              Vue.createElementVNode(
                "div",
                {
                  class: Vue.normalizeClass(Vue.unref(ns).e("effect"))
                },
                [
                  (Vue.openBlock(), Vue.createBlock(Vue.resolveDynamicComponent(Vue.unref(DemoComponent))))
                ],
                2
                /* CLASS */
              ),
              Vue.createElementVNode(
                "div",
                {
                  class: Vue.normalizeClass(Vue.unref(ns).e("button-group"))
                },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-demo-code-button-left"),
                  Vue.unref(playgroundUrl) ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                    key: 0,
                    title: Vue.unref(playgroundButtonTip),
                    onClick: handleEditPlayground,
                    icon: Vue.unref(playground.default),
                    role: "link",
                    "aria-label": Vue.unref(playgroundButtonTip)
                  }, null, 8, ["title", "icon", "aria-label"])) : Vue.createCommentVNode("v-if", true),
                  Vue.unref(githubUrl) ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                    key: 1,
                    title: Vue.unref(githubButtonTip),
                    onClick: handleEditGithub,
                    icon: Vue.unref(github.default),
                    role: "link",
                    "aria-label": Vue.unref(githubUrl)
                  }, null, 8, ["title", "icon", "aria-label"])) : Vue.createCommentVNode("v-if", true),
                  Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                    title: Vue.unref(copyButtonTip),
                    icon: Vue.unref(copy.default),
                    onClick: copyCode,
                    role: "button",
                    "aria-label": Vue.unref(copyButtonTip)
                  }, null, 8, ["title", "icon", "aria-label"]),
                  Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                    title: sourceVisible.value ? Vue.unref(expandSourceButtonTip) : Vue.unref(collapseSourceButtonTip),
                    onClick: _cache[0] || (_cache[0] = ($event) => handleToggleSourceVisible()),
                    icon: Vue.unref(code.default),
                    role: "button",
                    "aria-label": sourceVisible.value ? Vue.unref(expandSourceButtonTip) : Vue.unref(collapseSourceButtonTip)
                  }, null, 8, ["title", "icon", "aria-label"]),
                  Vue.renderSlot(_ctx.$slots, "teek-demo-code-button-right")
                ],
                2
                /* CLASS */
              ),
              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_TransitionCollapse_lang.default), null, {
                default: Vue.withCtx(() => [
                  Vue.withDirectives(Vue.createElementVNode("div", {
                    class: Vue.normalizeClass(Vue.unref(ns).join("vp-code")),
                    innerHTML: decodeSource.value
                  }, null, 10, _hoisted_2), [
                    [Vue.vShow, sourceVisible.value]
                  ])
                ]),
                _: 1
                /* STABLE */
              }),
              Vue.createVNode(Vue.Transition, {
                name: Vue.unref(ns).join("fade-linear"),
                persisted: ""
              }, {
                default: Vue.withCtx(() => [
                  Vue.withDirectives(Vue.createElementVNode(
                    "div",
                    {
                      class: Vue.normalizeClass(Vue.unref(ns).e("float-control")),
                      onClick: _cache[1] || (_cache[1] = ($event) => handleToggleSourceVisible(false)),
                      role: "button"
                    },
                    [
                      Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), { icon: Vue.unref(caretTop.default) }, null, 8, ["icon"]),
                      Vue.createElementVNode(
                        "span",
                        null,
                        Vue.toDisplayString(Vue.unref(expandSourceButtonTip)),
                        1
                        /* TEXT */
                      )
                    ],
                    2
                    /* CLASS */
                  ), [
                    [Vue.vShow, sourceVisible.value]
                  ])
                ]),
                _: 1
                /* STABLE */
              }, 8, ["name"])
            ],
            2
            /* CLASS */
          ))
        ],
        64
        /* STABLE_FRAGMENT */
      );
    };
  }
});

exports.default = _sfc_main;
