// vitepress-theme-teek/es/index.mjs
import DefaultTheme from 'vitepress/theme';
import { isClient, umamiAnalytics, googleAnalytics, baiduAnalytics, trackPageview } from './helper/dist/index.mjs';
export { addUnit, formatDate, formatDiffDate, formatDiffDateToDay, get, getDarkColor, getLightColor, getNowDate, hexToRgb, inBrowser, is, isArray, isAsyncFunction, isBoolean, isDate, isDef, isElement, isEmpty, isExternal, isFocusable, isFunction, isImageDom, isImagePath, isIos, isNull, isNullAndUnDef, isNullOrUnDef, isNumber, isObject, isPhone, isPlainFunction, isPromise, isServer, isString, isStringNumber, isType, isUnDef, isValidURL, removeStorageItem, removeUnit, rgbToHex, upperFirst, withBase } from './helper/dist/index.mjs';
import './static/iconfont/social/iconfont.mjs';
export { default as aliPayIcon } from './static/icons/aliPay.mjs';
export { default as arrowDownIcon } from './static/icons/arrowDown.mjs';
export { default as categoryIcon } from './static/icons/category.mjs';
export { default as closeIcon } from './static/icons/close.mjs';
export { default as copyrightIcon } from './static/icons/copyright.mjs';
export { default as docAnalysisIcon } from './static/icons/docAnalysis.mjs';
export { default as emptyIcon } from './static/icons/empty.mjs';
export { default as friendLinkIcon } from './static/icons/friendLink.mjs';
export { default as icpRecordIcon } from './static/icons/icpRecord.mjs';
export { default as noticeIcon } from './static/icons/notice.mjs';
export { default as playgroundIcon } from './static/icons/playground.mjs';
export { default as rocketIcon } from './static/icons/rocket.mjs';
export { default as sizeIcon } from './static/icons/size.mjs';
export { default as tagIcon } from './static/icons/tag.mjs';
export { default as themeIcon } from './static/icons/theme.mjs';
export { default as topArticleIcon } from './static/icons/topArticle.mjs';
export { default as readingIcon } from './static/icons/reading.mjs';
export { default as clockIcon } from './static/icons/clock.mjs';
export { default as viewIcon } from './static/icons/view.mjs';
export { default as houseIcon } from './static/icons/house.mjs';
export { default as arrowLeftIcon } from './static/icons/arrowLeft.mjs';
export { default as arrowRightIcon } from './static/icons/arrowRight.mjs';
export { default as magicIcon } from './static/icons/magic.mjs';
export { default as commentIcon } from './static/icons/comment.mjs';
export { default as userIcon } from './static/icons/user.mjs';
export { default as calendarIcon } from './static/icons/calendar.mjs';
export { default as folderOpenedIcon } from './static/icons/folderOpened.mjs';
export { default as collectionTagIcon } from './static/icons/collectionTag.mjs';
export { default as editPenIcon } from './static/icons/editPen.mjs';
export { default as githubIcon } from './static/icons/github.mjs';
export { default as copyIcon } from './static/icons/copy.mjs';
export { default as codeIcon } from './static/icons/code.mjs';
export { default as caretTopIcon } from './static/icons/caretTop.mjs';
export { default as weChatPayIcon } from './static/icons/weChatPay.mjs';
export { default as shareIcon } from './static/icons/share.mjs';
export { default as thumbsIcon } from './static/icons/thumbs.mjs';
export { default as externalLinkIcon } from './static/icons/externalLink.mjs';
export { default as dArrowRightIcon } from './static/icons/dArrowRight.mjs';
export { default as dArrowLeftIcon } from './static/icons/dArrowLeft.mjs';
export { default as moreFilledIcon } from './static/icons/moreFilled.mjs';
export { default as fullscreenIcon } from './static/icons/fullscreen.mjs';
export { default as refreshLeftIcon } from './static/icons/refreshLeft.mjs';
export { default as refreshRightIcon } from './static/icons/refreshRight.mjs';
export { default as scaleToOriginalIcon } from './static/icons/scaleToOriginal.mjs';
export { default as zoomInIcon } from './static/icons/zoomIn.mjs';
export { default as zoomOutIcon } from './static/icons/zoomOut.mjs';
export { default as circleCloseFilledIcon } from './static/icons/circleCloseFilled.mjs';
export { default as infoFilledIcon } from './static/icons/infoFilled.mjs';
export { default as successFilledIcon } from './static/icons/successFilled.mjs';
export { default as warningFilledIcon } from './static/icons/warningFilled.mjs';
export { default as questionFilledIcon } from './static/icons/questionFilled.mjs';
export { default as topIcon } from './static/icons/top.mjs';
export { default as lockIcon } from './static/icons/lock.mjs';
export { default as autoWidthIcon } from './static/icons/theme-enhance/autoWidth.mjs';
export { default as clickIcon } from './static/icons/theme-enhance/click.mjs';
export { default as fullScreenOneIcon } from './static/icons/theme-enhance/fullScreenOne.mjs';
export { default as fullscreenTwoIcon } from './static/icons/theme-enhance/fullscreenTwo.mjs';
export { default as layoutIcon } from './static/icons/theme-enhance/layout.mjs';
export { default as overallReductionIcon } from './static/icons/theme-enhance/overallReduction.mjs';
export { default as alignLeftIcon } from './static/icons/theme-enhance/alignLeft.mjs';
export { default as alignTextLeftIcon } from './static/icons/theme-enhance/alignTextLeft.mjs';
export { default as scaleIcon } from './static/icons/theme-enhance/scale.mjs';
export { default as waterIcon } from './static/icons/theme-enhance/water.mjs';
export { onClickOutside } from './composables/onClickOutside.mjs';
export { useAnchorScroll } from './composables/useAnchorScroll.mjs';
export { callBusuanzi, callVercount, createScript, useUvPv } from './composables/useUvPv.mjs';
export { useClipboard } from './composables/useClipboard.mjs';
export { useCopyBanner } from './composables/useCopyBanner.mjs';
export { useDebounce } from './composables/useDebounce.mjs';
export { useElementHover } from './composables/useElementHover.mjs';
export { useEventListener } from './composables/useEventListener.mjs';
export { localeContextKey, useLocale } from './composables/useLocale.mjs';
export { useMediaQuery } from './composables/useMediaQuery.mjs';
export { useMounted } from './composables/useMounted.mjs';
export { useNamespace } from './composables/useNamespace.mjs';
export { usePopoverSize } from './composables/usePopoverSize.mjs';
export { useScopeDispose } from './composables/useScopeDispose.mjs';
export { useScrollData } from './composables/useScrollData.mjs';
export { StorageSerializers, guessSerializerType, useStorage } from './composables/useStorage.mjs';
export { useSwitchData } from './composables/useSwitchData.mjs';
export { useTextTypes } from './composables/useTextTypes.mjs';
export { useThemeColor, varNameList } from './composables/useThemeColor.mjs';
export { useViewTransition } from './composables/useViewTransition.mjs';
export { useVpRouter } from './composables/useVpRouter.mjs';
export { useWindowSize } from './composables/useWindowSize.mjs';
export { useIntersectionObserver, useWindowTransition } from './composables/useWindowTransition.mjs';
export { Z_INDEX_INJECTION_KEY, defaultInitialZIndex, useZIndex, zIndexContextKey } from './composables/useZIndex.mjs';
export { default as zhCn } from './locale/lang/zh-cn.mjs';
export { default as en } from './locale/lang/en.mjs';
export { createContainerThenGet, createContainerThenUse, createContainersThenGet, createContainersThenUse } from './markdown/helper/simpleContainer.mjs';
export { createCardContainer, createCardContainers } from './markdown/helper/cardContainer.mjs';
export { version } from './version.mjs';
import { TeekConfigProvider } from './components/theme/ConfigProvider/index.mjs';
export { postsContext, teekConfigContext, useAllPosts, useCommon, usePagePath, usePageState, usePosts, useSidebar, useTagColor, useTeekConfig, useWindowTransitionConfig } from './components/theme/ConfigProvider/index.mjs';
import _sfc_main from './components/theme/CataloguePage/src/index.vue2.mjs';
export { default as TkCatalogueItem } from './components/theme/CataloguePage/src/CatalogueItem.vue2.mjs';
import _sfc_main$1 from './components/theme/ArchivesPage/src/index.vue2.mjs';
import _sfc_main$2 from './components/theme/ArticleOverviewPage/src/index.vue2.mjs';
import _sfc_main$3 from './components/theme/LoginPage/src/index.vue2.mjs';
export { defaultPrivateConfig, getLoginStorageKey, loginUrlKeyMap, verifyModeMap } from './components/theme/LoginPage/src/login.mjs';
export { useWatchLogin } from './components/theme/LoginPage/src/useWatchLogin.mjs';
import _sfc_main$4 from './components/theme/RiskLinkPage/src/index.vue2.mjs';
export { useRiskLink } from './components/theme/RiskLinkPage/src/useRiskLink.mjs';
import _sfc_main$5 from './components/theme/DemoCode/src/index.vue2.mjs';
import _sfc_main$6 from './components/common/TitleTag/src/index.vue2.mjs';
import _sfc_main$7 from './components/common/Icon/src/index.vue2.mjs';
export { WhatsApp, bilibili, email, gitee, github, moblieQQ, music, telegram } from './components/common/Icon/src/SocialIcons.mjs';
export { addCollection as addIcons } from '@iconify/vue';
import _sfc_main$8 from './components/theme/Layout/src/index.vue2.mjs';
export { default as TkArticlePage } from './components/common/ArticlePage/src/index.vue2.mjs';
export { default as TkDocAsideOutline } from './components/common/ArticlePage/src/components/DocAsideOutline.vue2.mjs';
export { default as TkAvatar } from './components/common/Avatar/src/index.vue2.mjs';
export { default as TkBreadcrumb } from './components/common/Breadcrumb/src/Breadcrumb.vue2.mjs';
export { default as TkBreadcrumbItem } from './components/common/Breadcrumb/src/BreadcrumbItem.vue2.mjs';
export { default as TkImageViewer } from './components/common/ImageViewer/src/index.vue2.mjs';
export { default as TkInputSlide } from './components/common/InputSlide/src/index.vue2.mjs';
export { default as TkMessage } from './components/common/Message/src/method.mjs';
export { default as TkPagination } from './components/common/Pagination/src/index.vue2.mjs';
export { default as TkPopover } from './components/common/Popover/src/index.vue2.mjs';
export { default as TkSegmented } from './components/common/Segmented/src/Segmented.vue2.mjs';
export { default as TkSegmentedItem } from './components/common/Segmented/src/SegmentedItem.vue2.mjs';
export { default as TkTransitionCollapse } from './components/common/TransitionCollapse/src/index.vue2.mjs';
export { default as TkVerifyCode } from './components/common/VerifyCode/src/index.vue2.mjs';
export { default as TkVpContainer } from './components/common/VpContainer/src/index.vue2.mjs';
export { default as TkArticleAnalyze } from './components/theme/ArticleAnalyze/src/index.vue2.mjs';
export { default as TkAsideBottomAppreciation } from './components/theme/ArticleAppreciation/src/AsideBottomAppreciation.vue2.mjs';
export { default as TkDocAfterAppreciation } from './components/theme/ArticleAppreciation/src/DocAfterAppreciation.vue2.mjs';
export { default as TkDocAfterAppreciationPopper } from './components/theme/ArticleAppreciation/src/DocAfterAppreciationPopper.vue2.mjs';
export { default as TkArticleBreadcrumb } from './components/theme/ArticleBreadcrumb/src/index.vue2.mjs';
export { default as TkArticleHeadingHighlight } from './components/theme/ArticleHeadingHighlight/src/index.vue2.mjs';
export { default as TkArticleImagePreview } from './components/theme/ArticleImagePreview/src/index.vue2.mjs';
export { default as TkArticleInfo } from './components/theme/ArticleInfo/src/index.vue2.mjs';
export { default as TkArticlePageStyle } from './components/theme/ArticlePageStyle/src/index.vue2.mjs';
export { default as TkArticleShare } from './components/theme/ArticleShare/src/index.vue2.mjs';
export { default as TkArticleTitle } from './components/theme/ArticleTitle/src/index.vue2.mjs';
export { default as TkArticleUpdate } from './components/theme/ArticleUpdate/src/index.vue2.mjs';
export { default as TkBodyBgImage } from './components/theme/BodyBgImage/src/index.vue2.mjs';
export { default as TkCodeBlockToggle } from './components/theme/CodeBlockToggle/src/index.vue2.mjs';
export { default as TkCommentArtalk } from './components/theme/CommentArtalk/src/index.vue2.mjs';
export { default as TkCommentGiscus } from './components/theme/CommentGiscus/src/index.vue2.mjs';
export { default as TkCommentTwikoo } from './components/theme/CommentTwikoo/src/index.vue2.mjs';
export { default as TkCommentWaline } from './components/theme/CommentWaline/src/index.vue2.mjs';
export { default as TkFooterGroup } from './components/theme/FooterGroup/src/index.vue2.mjs';
export { default as TkFooterInfo } from './components/theme/FooterInfo/src/index.vue2.mjs';
export { default as TkHome } from './components/theme/Home/src/index.vue2.mjs';
export { default as TkHomeBanner } from './components/theme/HomeBanner/src/index.vue2.mjs';
export { default as TkHomeBannerBgImage } from './components/theme/HomeBanner/src/HomeBannerBgImage.vue2.mjs';
export { default as TkHomeBannerBgPure } from './components/theme/HomeBanner/src/HomeBannerBgPure.vue2.mjs';
export { default as TkHomeBannerContent } from './components/theme/HomeBanner/src/HomeBannerContent.vue2.mjs';
export { default as TkHomeBannerFeature } from './components/theme/HomeBanner/src/HomeBannerFeature.vue2.mjs';
export { default as TkHomeBannerWaves } from './components/theme/HomeBanner/src/HomeBannerWaves.vue2.mjs';
export { default as TkPageCard } from './components/common/PageCard/src/index.vue2.mjs';
export { default as TkHomeCategoryCard } from './components/theme/HomeCategoryCard/src/index.vue2.mjs';
export { default as TkHomeDocAnalysisCard } from './components/theme/HomeDocAnalysisCard/src/index.vue2.mjs';
export { default as TkHomeFeature } from './components/theme/HomeFeature/src/index.vue2.mjs';
export { default as TkHomeFriendLinkCard } from './components/theme/HomeFriendLinkCard/src/index.vue2.mjs';
export { default as TkHomeFullscreenWallpaper } from './components/theme/HomeFullscreenWallpaper/src/index.vue2.mjs';
export { default as TkHomeCardList } from './components/theme/HomeCardList/src/index.vue2.mjs';
export { default as TkHomeMyCard } from './components/theme/HomeMyCard/src/index.vue2.mjs';
export { default as TkHomeMyCardScreen } from './components/theme/HomeMyCard/src/HomeMyCardScreen.vue2.mjs';
export { default as TkHomePostList } from './components/theme/HomePostList/src/index.vue2.mjs';
export { default as TkHomePostItem } from './components/theme/HomePostList/src/HomePostItem.vue2.mjs';
export { default as TkHomeTagCard } from './components/theme/HomeTagCard/src/index.vue2.mjs';
export { default as TkHomeTopArticleCard } from './components/theme/HomeTopArticleCard/src/index.vue2.mjs';
export { default as TkNotice } from './components/theme/Notice/src/index.vue2.mjs';
export { default as TkRightBottomButton } from './components/theme/RightBottomButton/src/index.vue2.mjs';
export { default as TkRouteLoading } from './components/theme/RouteLoading/src/index.vue.mjs';
export { default as TkSidebarTrigger } from './components/theme/SidebarTrigger/src/index.vue2.mjs';
export { default as TkThemeEnhance } from './components/theme/ThemeEnhance/src/index.vue2.mjs';
export { default as TkSwitch } from './components/theme/ThemeEnhance/src/components/Switch.vue2.mjs';
export { default as TkThemeEnhanceBaseTemplate } from './components/theme/ThemeEnhance/src/components/BaseTemplate.vue2.mjs';
export { default as TkFocusTrap } from './components/common/FocusTrap/src/index.vue.mjs';
export { createImageViewer } from './components/common/ImageViewer/index.mjs';
export { getInstance, getLastOffset, getOffsetOrSpace, instances } from './components/common/Message/src/instance.mjs';
export { createDynamicComponent } from './components/theme/ArticleTitle/src/compile.mjs';
export { artalkContext } from './components/theme/CommentArtalk/src/artalk.mjs';
export { giscusContext } from './components/theme/CommentGiscus/src/giscus.mjs';
export { twikooContext } from './components/theme/CommentTwikoo/src/twikoo.mjs';
export { walineContext } from './components/theme/CommentWaline/src/waline.mjs';
export { postDataUpdateSymbol } from './components/theme/Home/src/home.mjs';
export { pageNumKey } from './components/theme/HomePostList/src/homePostList.mjs';
export { LayoutMode, SpotlightStyle, ThemeColorName, activateMaxWidthSlideMedia, layoutModeAttribute, mobileMaxWidthMedia, themeColorAttribute, themeColorList, touchMedia } from './components/theme/ThemeEnhance/src/themeEnhance.mjs';
export { docMaxWidthSlideStorageKey, docMaxWidthVar, layoutModeStorageKey, ns, pageMaxWidthSlideStorageKey, pageMaxWidthVar, spotlightStorageKey, spotlightStyleStorageKey, themeBgColorStorageKey, themeColorStorageKey, transitionName } from './components/theme/ThemeEnhance/src/namespace.mjs';
export { useThemeColorList } from './components/theme/ThemeEnhance/src/useThemeColorList.mjs';

var index = {
  extends: DefaultTheme,
  Layout: TeekConfigProvider(_sfc_main$8),
  async enhanceApp({ app, siteData, router }) {
    app.component("TkCataloguePage", _sfc_main);
    app.component("TkArchivesPage", _sfc_main$1);
    app.component("TkArticleOverviewPage", _sfc_main$2);
    app.component("TkLoginPage", _sfc_main$3);
    app.component("TkRiskLinkPage", _sfc_main$4);
    app.component("TkDemoCode", _sfc_main$5);
    app.component("TkTitleTag", _sfc_main$6);
    app.component("TkIcon", _sfc_main$7);
    if (!isClient) return;
    const { themeConfig } = siteData.value;
    processSiteAnalytics(themeConfig);
    if (themeConfig.permalinks) await processPermalinkNotFoundWhenFirstLoaded({ siteData, router });
  }
};
const processSiteAnalytics = (themeConfig) => {
  const siteAnalytics = themeConfig.siteAnalytics || [];
  const siteAnalysis = {
    baidu: (options) => {
      baiduAnalytics(options);
      if (isClient) trackPageview(options, window.location.href);
    },
    google: (options) => googleAnalytics(options),
    umami: (options) => umamiAnalytics(options)
  };
  siteAnalytics.forEach((item) => {
    siteAnalysis[item.provider]?.(item.options);
  });
};
const processPermalinkNotFoundWhenFirstLoaded = async ({ siteData, router }) => {
  const { base, cleanUrls, themeConfig } = siteData.value;
  if (router.route.path === base && router.route.data.isNotFound) {
    const { pathname, search, hash } = new URL(location.href);
    const decodePath = "/" + decodeURIComponent(pathname.slice(base.length)).replace(/\/$/, "").replace(/\.html/, "");
    const link = cleanUrls ? decodePath : decodePath + ".html";
    const filePath = themeConfig.permalinks.inv[link];
    if (filePath) {
      const targetUrl = base + filePath + search + hash;
      await router.go(targetUrl);
    }
  }
};

export { TeekConfigProvider, _sfc_main$1 as TkArchivesPage, _sfc_main$2 as TkArticleOverviewPage, _sfc_main as TkCataloguePage, _sfc_main$5 as TkDemoCode, _sfc_main$7 as TkIcon, _sfc_main$8 as TkLayout, _sfc_main$3 as TkLoginPage, _sfc_main$4 as TkRiskLinkPage, _sfc_main$6 as TkTitleTag, baiduAnalytics, index as default, googleAnalytics, isClient, trackPageview, umamiAnalytics };
