// vitepress-theme-teek@1.5.1/es/components/theme/HomeBanner/src/HomeBannerFeature.vue2.mjs
import { defineComponent, ref, computed, onBeforeUnmount, createBlock, createCommentVNode, openBlock, TransitionGroup, unref, normalizeClass, withCtx, createElementBlock, Fragment, renderList, withDirectives, resolveDynamicComponent, createElementVNode, toDisplayString, vShow } from 'vue';
import { useData, withBase } from 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../composables/useLocale.mjs';
import { useWindowSize } from '../../../../composables/useWindowSize.mjs';
import '../../../../composables/useZIndex.mjs';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';

const _hoisted_1 = ["aria-labelledby"];
const _hoisted_2 = ["src", "alt", "aria-label"];
const _hoisted_3 = ["id"];
const _hoisted_4 = { class: "feature-details" };
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "HomeBannerFeature" },
  __name: "HomeBannerFeature",
  setup(__props) {
    const ns = useNamespace("banner-feature");
    const { t } = useLocale();
    const { getTeekConfigRef } = useTeekConfig();
    const { frontmatter } = useData();
    const bannerConfig = getTeekConfigRef("banner", {
      features: frontmatter.value.tk?.features || [],
      featureCarousel: 4e3
    });
    const featuresConfig = getTeekConfigRef("features", []);
    const active = ref(0);
    const isMobile = ref(false);
    let timer;
    const features = computed(() => bannerConfig.value.features || featuresConfig.value || []);
    const clearTimer = () => {
      if (timer) {
        clearTimeout(timer);
        timer = null;
      }
    };
    useWindowSize((width) => {
      const { featureCarousel } = bannerConfig.value;
      if (width <= 719) {
        isMobile.value = true;
        clearTimer();
        timer = setInterval(() => {
          active.value = (active.value + 1) % features.value.length;
        }, featureCarousel);
      } else {
        isMobile.value = false;
        clearTimer();
      }
    });
    onBeforeUnmount(() => {
      clearTimer();
    });
    return (_ctx, _cache) => {
      return features.value.length ? (openBlock(), createBlock(TransitionGroup, {
        key: 0,
        name: unref(ns).join("slide-next"),
        tag: "div",
        class: normalizeClass([unref(ns).b(), unref(ns).join("wallpaper-outside"), "flx-wrap-between"]),
        "aria-label": unref(t)("tk.homeBanner.featureLabel")
      }, {
        default: withCtx(() => [
          (openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(unref(bannerConfig).features, (feature, index) => {
              return withDirectives((openBlock(), createElementBlock("div", {
                class: normalizeClass(unref(ns).e("feature__item")),
                key: index,
                role: "group",
                "aria-labelledby": `feature-title-${index}`
              }, [
                (openBlock(), createBlock(resolveDynamicComponent(feature.link ? "a" : "div"), {
                  href: unref(withBase)(feature.link ?? ""),
                  class: "flx-column-center hover-color",
                  "aria-label": feature.title
                }, {
                  default: withCtx(() => [
                    feature.image ? (openBlock(), createElementBlock("img", {
                      key: 0,
                      class: "feature-img",
                      src: unref(withBase)(feature.image),
                      alt: feature.title,
                      "aria-label": feature.title
                    }, null, 8, _hoisted_2)) : createCommentVNode("v-if", true),
                    createElementVNode("p", {
                      id: `feature-title-${index}`,
                      class: "feature-title"
                    }, toDisplayString(feature.title), 9, _hoisted_3),
                    createElementVNode(
                      "p",
                      _hoisted_4,
                      toDisplayString(feature.details),
                      1
                      /* TEXT */
                    )
                  ]),
                  _: 2
                  /* DYNAMIC */
                }, 1032, ["href", "aria-label"]))
              ], 10, _hoisted_1)), [
                [vShow, !isMobile.value || active.value === index]
              ]);
            }),
            128
            /* KEYED_FRAGMENT */
          ))
        ]),
        _: 1
        /* STABLE */
      }, 8, ["name", "class", "aria-label"])) : createCommentVNode("v-if", true);
    };
  }
});

export { _sfc_main as default };
