// vitepress-theme-teek@1.5.1/es/components/theme/HomeBanner/src/HomeBannerWaves.vue2.mjs
import { defineComponent, createElementBlock, openBlock, normalizeClass, unref, createCommentVNode, createElementVNode, createStaticVNode } from 'vue';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import '../../../../composables/useZIndex.mjs';

var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "HomeBannerWaves" },
  __name: "HomeBannerWaves",
  setup(__props) {
    const ns = useNamespace("banner-waves");
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        "svg",
        {
          class: normalizeClass(unref(ns).b()),
          xmlns: "http://www.w3.org/2000/svg",
          "xmlns:xlink": "http://www.w3.org/1999/xlink",
          viewBox: "0 24 150 28",
          preserveAspectRatio: "none",
          "shape-rendering": "auto"
        },
        [
          createCommentVNode("\u5F62\u72B6\u5BB9\u5668"),
          _cache[0] || (_cache[0] = createElementVNode(
            "defs",
            null,
            [
              createElementVNode("path", {
                id: "gentle-wave",
                d: "M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"
              })
            ],
            -1
            /* CACHED */
          )),
          createCommentVNode("\u7EC4\u5408\u6CE2\u6D6A"),
          _cache[1] || (_cache[1] = createStaticVNode('<g class="parallax"><use class="use" xlink:href="#gentle-wave" x="48" y="0"></use><use class="use dark:fill-black" xlink:href="#gentle-wave" x="48" y="3"></use><use class="use dark:fill-black" xlink:href="#gentle-wave" x="48" y="5"></use><use class="use dark:fill-black" xlink:href="#gentle-wave" x="48" y="7"></use></g>', 1))
        ],
        2
        /* CLASS */
      );
    };
  }
});

export { _sfc_main as default };
