// vitepress-theme-teek/lib/components/theme/LoginPage/src/login.js
'use strict';

require('vue');
var index = require('../../../../helper/dist/index.js');
require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
require('../../../../composables/useZIndex.js');

const getLoginStorageKey = () => {
  if (!index.isClient) return { siteLoginKey: "", pagesLoginKey: "", pageLoginKey: "", realmLoginKey: "" };
  const ns = useNamespace.useNamespace();
  const siteLoginKey = ns.storageKey("private", "site", window.location.hostname);
  const pagesLoginKey = ns.storageKey("private", "pages", window.location.hostname);
  const pageLoginKey = ns.storageKey("private", "page", window.location.hostname);
  const realmLoginKey = ns.storageKey("private", "realm", window.location.hostname);
  return { siteLoginKey, pagesLoginKey, pageLoginKey, realmLoginKey };
};
const loginUrlKeyMap = {
  realm: "realm",
  toPath: "toPath",
  verifyMode: "verifyMode"
};
const verifyModeMap = {
  site: "site",
  pages: "pages",
  page: "page",
  realm: "realm"
};
const defaultPrivateConfig = {
  enabled: false,
  expire: "1d",
  session: false,
  siteLogin: false,
  site: [],
  pages: [],
  realm: {}
};

exports.defaultPrivateConfig = defaultPrivateConfig;
exports.getLoginStorageKey = getLoginStorageKey;
exports.loginUrlKeyMap = loginUrlKeyMap;
exports.verifyModeMap = verifyModeMap;
