// vitepress-theme-teek@1.5.1/es/components/theme/ThemeEnhance/src/components/BaseTemplate.vue2.mjs
import { defineComponent, ref, createElementBlock, openBlock, createElementVNode, createBlock, createVNode, createCommentVNode, withCtx, renderSlot, normalizeClass, unref, createTextVNode, toDisplayString, Fragment, renderList } from 'vue';
import { ns } from '../namespace.mjs';
import _sfc_main$3 from '../../../../common/Icon/src/index.vue2.mjs';
import '@iconify/vue';
import _sfc_main$1 from './Title.vue2.mjs';
import _sfc_main$2 from './Helper.vue2.mjs';
import _sfc_main$4 from './BorderHighlight.vue2.mjs';

const _hoisted_1 = {
  class: "flx-align-center",
  style: { "margin-bottom": "6px" }
};
const _hoisted_2 = {
  key: 1,
  style: { "font-weight": "600" }
};
const _hoisted_3 = { key: 0 };
const _hoisted_4 = {
  key: 1,
  style: { "margin-top": "8px" }
};
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "BaseTemplate" },
  __name: "BaseTemplate",
  props: {
    title: {},
    helperDesc: {},
    icon: {},
    tips: {},
    disabled: { type: Boolean },
    helper: { type: Boolean },
    borderHighlight: { type: Boolean, default: true }
  },
  setup(__props) {
    const helperVisible = ref(false);
    const titleElementRef = ref();
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", null, [
        createElementVNode(
          "div",
          {
            ref_key: "titleElementRef",
            ref: titleElementRef,
            class: "flx-align-center"
          },
          [
            createVNode(_sfc_main$1, {
              title: __props.title,
              icon: __props.icon,
              disabled: __props.disabled,
              "aria-label": __props.title
            }, {
              default: withCtx(() => [
                renderSlot(_ctx.$slots, "title")
              ]),
              _: 3
              /* FORWARDED */
            }, 8, ["title", "icon", "disabled", "aria-label"]),
            __props.helper ? (openBlock(), createBlock(_sfc_main$2, {
              key: 0,
              modelValue: helperVisible.value,
              "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => helperVisible.value = $event),
              "trigger-el": titleElementRef.value
            }, {
              default: withCtx(() => [
                createElementVNode(
                  "div",
                  {
                    class: normalizeClass(unref(ns).e("helper__body"))
                  },
                  [
                    createElementVNode(
                      "h4",
                      {
                        class: normalizeClass(unref(ns).em("helper", "title"))
                      },
                      [
                        renderSlot(_ctx.$slots, "helper-title", {}, () => [
                          createTextVNode(
                            toDisplayString(__props.title),
                            1
                            /* TEXT */
                          )
                        ])
                      ],
                      2
                      /* CLASS */
                    ),
                    createElementVNode(
                      "p",
                      {
                        class: normalizeClass(unref(ns).em("helper", "desc"))
                      },
                      [
                        renderSlot(_ctx.$slots, "helper-desc", {}, () => [
                          createTextVNode(
                            toDisplayString(__props.helperDesc),
                            1
                            /* TEXT */
                          )
                        ])
                      ],
                      2
                      /* CLASS */
                    ),
                    (openBlock(true), createElementBlock(
                      Fragment,
                      null,
                      renderList(__props.tips, (tip, index) => {
                        return openBlock(), createElementBlock(
                          "div",
                          {
                            key: index,
                            class: normalizeClass(unref(ns).e("helper__body__tip"))
                          },
                          [
                            createElementVNode("div", _hoisted_1, [
                              tip.icon ? (openBlock(), createBlock(unref(_sfc_main$3), {
                                key: 0,
                                icon: tip.icon,
                                size: 16,
                                style: { "margin-right": "4px" }
                              }, null, 8, ["icon"])) : createCommentVNode("v-if", true),
                              tip.title ? (openBlock(), createElementBlock(
                                "span",
                                _hoisted_2,
                                toDisplayString(tip.title),
                                1
                                /* TEXT */
                              )) : createCommentVNode("v-if", true)
                            ]),
                            tip.content ? (openBlock(), createElementBlock(
                              "span",
                              _hoisted_3,
                              toDisplayString(tip.content),
                              1
                              /* TEXT */
                            )) : createCommentVNode("v-if", true)
                          ],
                          2
                          /* CLASS */
                        );
                      }),
                      128
                      /* KEYED_FRAGMENT */
                    ))
                  ],
                  2
                  /* CLASS */
                )
              ]),
              _: 3
              /* FORWARDED */
            }, 8, ["modelValue", "trigger-el"])) : createCommentVNode("v-if", true)
          ],
          512
          /* NEED_PATCH */
        ),
        __props.borderHighlight ? (openBlock(), createBlock(_sfc_main$4, {
          key: 0,
          active: helperVisible.value,
          style: { "margin-top": "8px" }
        }, {
          default: withCtx(() => [
            renderSlot(_ctx.$slots, "default")
          ]),
          _: 3
          /* FORWARDED */
        }, 8, ["active"])) : (openBlock(), createElementBlock("div", _hoisted_4, [
          renderSlot(_ctx.$slots, "default")
        ]))
      ]);
    };
  }
});

export { _sfc_main as default };
