// vitepress-theme-teek@1.5.1/lib/components/theme/SidebarTrigger/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
var useMediaQuery = require('../../../../composables/useMediaQuery.js');
require('../../../../composables/useZIndex.js');
var autoWidth = require('../../../../static/icons/theme-enhance/autoWidth.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../../common/Icon/src/index.vue2.js');
require('@iconify/vue');

const _hoisted_1 = ["title", "aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  __name: "index",
  setup(__props) {
    const ns = useNamespace.useNamespace("sidebar-trigger");
    const { t } = useLocale.useLocale();
    const active = Vue.ref(false);
    const isLt960 = useMediaQuery.useMediaQuery("(max-width: 960px)");
    const toggleSideBar = () => {
      const layoutDom = document.querySelector(`.${ns.join("layout")}`);
      layoutDom && layoutDom.classList.toggle(ns.is("sidebar-collapse"));
      if (!active.value) {
        active.value = true;
        setTimeout(() => {
          active.value = false;
        }, 300);
      }
    };
    Vue.onMounted(async () => {
      await Vue.nextTick();
      const docDom = document.querySelector(".VPContent.has-sidebar");
      docDom?.classList.add("sidebar-trigger-transition");
    });
    return (_ctx, _cache) => {
      return !Vue.unref(isLt960) ? Vue.renderSlot(_ctx.$slots, "default", {
        key: 0,
        active: active.value,
        icon: Vue.unref(autoWidth.default),
        toggleSideBar
      }, () => [
        Vue.createElementVNode("div", {
          class: Vue.normalizeClass([Vue.unref(ns).b(), Vue.unref(ns).is("active", active.value)]),
          onClick: toggleSideBar,
          title: Vue.unref(t)("tk.sidebarTrigger.label"),
          role: "button",
          "aria-label": Vue.unref(t)("tk.sidebarTrigger.label")
        }, [
          Vue.createElementVNode(
            "div",
            {
              class: Vue.normalizeClass([Vue.unref(ns).join("right-bottom-button__button")])
            },
            [
              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                icon: Vue.unref(autoWidth.default),
                "aria-hidden": "true"
              }, null, 8, ["icon"])
            ],
            2
            /* CLASS */
          )
        ], 10, _hoisted_1)
      ]) : Vue.createCommentVNode("v-if", true);
    };
  }
});

exports.default = _sfc_main;
