// vitepress-theme-teek@1.5.1/lib/components/theme/ArticleTitle/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var index_vue_vue_type_script_setup_true_name_TitleTag_lang = require('../../../common/TitleTag/src/index.vue2.js');
var compile = require('./compile.js');

const _hoisted_1 = ["aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "ArticleTitle" },
  __name: "index",
  props: {
    post: {},
    titleTagProps: {}
  },
  setup(__props) {
    const ns = useNamespace.useNamespace("article-title");
    const { t } = useLocale.useLocale();
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("span", {
        class: Vue.normalizeClass(Vue.unref(ns).b()),
        "aria-label": Vue.unref(t)("tk.articleTitle.label")
      }, [
        __props.post.frontmatter.titleTag && __props.titleTagProps?.position === "left" ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_TitleTag_lang.default), Vue.mergeProps({
          key: 0,
          text: __props.post.frontmatter.titleTag
        }, __props.titleTagProps, {
          "aria-label": __props.post.frontmatter.titleTag
        }), null, 16, ["text", "aria-label"])) : Vue.createCommentVNode("v-if", true),
        Vue.renderSlot(_ctx.$slots, "default", {}, () => [
          __props.post.title ? (Vue.openBlock(), Vue.createBlock(Vue.resolveDynamicComponent(Vue.unref(compile.createDynamicComponent)(__props.post.title)), { key: 0 })) : Vue.createCommentVNode("v-if", true)
        ]),
        __props.post.frontmatter.titleTag && __props.titleTagProps?.position === "right" ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_TitleTag_lang.default), Vue.mergeProps({
          key: 1,
          text: __props.post.frontmatter.titleTag
        }, __props.titleTagProps, {
          "aria-label": __props.post.frontmatter.titleTag
        }), null, 16, ["text", "aria-label"])) : Vue.createCommentVNode("v-if", true)
      ], 10, _hoisted_1);
    };
  }
});

exports.default = _sfc_main;
