// vitepress-theme-teek/lib/composables/useScrollbarSize.js
'use strict';

var Vue = require('vue');
var index = require('../helper/dist/index.js');
var useEventListener = require('./useEventListener.js');
var useMounted = require('./useMounted.js');
var useScopeDispose = require('./useScopeDispose.js');

const useScrollbarSize = () => {
  const width = Vue.ref(0);
  const height = Vue.ref(0);
  let measureElement = null;
  const createMeasureElement = () => {
    if (measureElement) return measureElement;
    document.querySelector("#measure-element")?.remove();
    measureElement = document.createElement("div");
    measureElement.id = "measure-element";
    measureElement.style.cssText = `
        position: fixed;
        top: -9999px;
        left: 0;
        width: 100px;
        height: 100px;
        overflow: scroll;
        visibility: hidden;
      `;
    document.body.appendChild(measureElement);
    return measureElement;
  };
  const clearMeasureElement = () => {
    if (!measureElement) return;
    if (document.body.contains(measureElement)) {
      document.body.removeChild(measureElement);
      measureElement = null;
    }
  };
  const calculate = () => {
    if (!index.isClient) return;
    const docElem = document.documentElement;
    const isQuirksMode = document.compatMode === "BackCompat";
    const hasVertical = isQuirksMode ? document.body.scrollHeight > document.body.clientHeight : docElem.scrollHeight > docElem.clientHeight;
    const hasHorizontal = isQuirksMode ? document.body.scrollWidth > document.body.clientWidth : docElem.scrollWidth > docElem.clientWidth;
    if (!hasVertical && !hasHorizontal) {
      width.value = 0;
      height.value = 0;
      return;
    }
    const measure = createMeasureElement();
    width.value = hasVertical ? measure.offsetWidth - measure.clientWidth : 0;
    height.value = hasHorizontal ? measure.offsetHeight - measure.clientHeight : 0;
  };
  const update = () => {
    calculate();
  };
  useEventListener.useEventListener(() => window, "resize", update);
  useMounted.useMounted(() => {
    createMeasureElement();
    calculate();
  });
  calculate();
  useScopeDispose.useScopeDispose(clearMeasureElement);
  return { width, height, update };
};

exports.useScrollbarSize = useScrollbarSize;
