// vitepress-theme-teek@1.5.1/es/components/theme/CodeBlockToggle/src/index.vue2.mjs
import { defineComponent, watch, nextTick } from 'vue';
import { isClient, addUnit, isBoolean } from '../../../../helper/dist/index.mjs';
import { useEventListener } from '../../../../composables/useEventListener.mjs';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import '../../../../composables/useZIndex.mjs';
import arrowDownIcon from '../../../../static/icons/arrowDown.mjs';
import message from '../../../common/Message/src/method.mjs';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';
import { createOverlay } from './createOverlay.mjs';

const documentAttribute = "code-block";
const foldClass = "fold";
const arrowClass = "code-arrow";
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "CodeBlockToggle" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace();
    const { getTeekConfigRef } = useTeekConfig();
    const codeBlockConfig = getTeekConfigRef("codeBlock", {
      enabled: true,
      collapseHeight: 700,
      copiedDone: void 0,
      overlay: false,
      overlayHeight: 400,
      langTextTransform: ""
    });
    watch(
      codeBlockConfig,
      (newVal) => {
        if (!isClient) return;
        const { enabled = true, langTextTransform } = newVal || {};
        if (!enabled) return document.documentElement.removeAttribute(documentAttribute);
        document.documentElement.setAttribute(documentAttribute, ns.namespace);
        if (langTextTransform) {
          document.documentElement.style.setProperty(ns.cssVarName("code-block-lang-transform"), langTextTransform);
        }
        nextTick(() => initCodeBlock());
      },
      { immediate: true }
    );
    const initCodeBlock = () => {
      const modes = document.querySelectorAll(".vp-doc div[class*='language-']");
      Array.from(modes).forEach((item) => {
        const copyDom = item.querySelector(`.copy`);
        copyDom?.addEventListener("click", () => {
          codeBlockConfig.value.copiedDone?.(message);
        });
        const className = item.parentElement?.className;
        if (className?.includes("details") || className?.includes(ns.join("vp-code"))) return;
        const arrowElement = item.querySelector(`.${arrowClass}`);
        if (arrowElement) return;
        const newArrowElement = document.createElement("div");
        newArrowElement.setAttribute("aria-hidden", "true");
        newArrowElement.classList.add(arrowClass);
        newArrowElement.innerHTML = arrowDownIcon;
        if (codeBlockConfig.value.overlay) {
          const overlay = createOverlay(() => newArrowElement.click());
          item.appendChild(overlay);
        }
        addClickEvent(newArrowElement, item);
        item.append(newArrowElement);
      });
    };
    const addClickEvent = (arrowDom, codeDom) => {
      const modeHeight = getElementHeight(codeDom);
      codeDom.style.height = addUnit(modeHeight);
      const preDom = codeDom.querySelector("pre");
      const lineNumbersWrapperDom = codeDom.querySelector(".line-numbers-wrapper");
      const codeBlockOverlayDom = codeDom.querySelector(".code-block-overlay");
      const codeBlockState = {
        expand: { height: addUnit(modeHeight), display: "block", overlayDisplay: "none", speed: 80 },
        fold: {
          height: codeBlockConfig.value.overlay ? addUnit(codeBlockConfig.value.overlayHeight) ?? "400px" : ns.cssVar("code-block-fold-height"),
          display: codeBlockConfig.value.overlay ? "block" : "none",
          overlayDisplay: "flex",
          speed: 400
        }
      };
      let timer;
      const clearTimer = () => {
        if (timer) {
          clearTimeout(timer);
          timer = null;
        }
      };
      const toggle = () => {
        const isFold = arrowDom.classList.contains(foldClass);
        const state = codeBlockState[isFold ? "expand" : "fold"];
        codeDom.style.height = state.height;
        clearTimer();
        if (preDom || lineNumbersWrapperDom) {
          timer = setTimeout(() => {
            if (preDom) preDom.style.display = state.display;
            if (lineNumbersWrapperDom) lineNumbersWrapperDom.style.display = state.display;
            if (codeBlockOverlayDom) codeBlockOverlayDom.style.display = state.overlayDisplay;
            clearTimer();
          }, state.speed);
        }
        arrowDom.classList.toggle(foldClass);
      };
      useEventListener(arrowDom, "click", toggle);
      const collapseHeight = codeBlockConfig.value.collapseHeight;
      if (isBoolean(collapseHeight)) {
        if (collapseHeight) toggle();
        else if (codeBlockOverlayDom) codeBlockOverlayDom.style.display = "none";
      } else if (collapseHeight && modeHeight > collapseHeight) toggle();
      else if (codeBlockOverlayDom) codeBlockOverlayDom.style.display = "none";
    };
    const getElementHeight = (item) => {
      const parentElementClass = item.parentElement?.className || "";
      if (!parentElementClass.includes("blocks")) return item.offsetHeight;
      if (parentElementClass.includes("blocks") && item.className.includes("active")) return item.offsetHeight;
      item.style.display = "block";
      const height = item.offsetHeight;
      item.style.display = "";
      return height;
    };
    return (_ctx, _cache) => {
      return null;
    };
  }
});

export { _sfc_main as default };
