// vitepress-theme-teek@1.5.1/lib/composables/usePopoverSize.js
'use strict';

var Vue = require('vue');
var useWindowSize = require('./useWindowSize.js');
var useScrollbarSize = require('./useScrollbarSize.js');
var index = require('../helper/dist/index.js');
var useEventListener = require('./useEventListener.js');

const AUTO = "auto";
const defaultSpace = 10;
const usePopoverSize = (trigger, popover, options = {}) => {
  const { placement = "bottom", offset = 0, xOffset = 0, yOffset = 0 } = options;
  const top = Vue.ref(AUTO);
  const right = Vue.ref(AUTO);
  const bottom = Vue.ref(AUTO);
  const left = Vue.ref(AUTO);
  const { width: windowWidth, height: windowHeight } = useWindowSize.useWindowSize();
  const { width: scrollbarWidth, height: scrollbarHeight } = useScrollbarSize.useScrollbarSize();
  const triggerEl = Vue.computed(() => {
    const plain = Vue.toValue(trigger);
    return plain?.$el ?? plain;
  });
  const popoverEl = Vue.computed(() => {
    const plain = Vue.toValue(popover);
    return plain?.$el ?? plain;
  });
  const calculatePosition = async () => {
    if (!index.isClient || !triggerEl.value || !popoverEl.value) return;
    await Vue.nextTick();
    const {
      top: triggerTop,
      right: triggerLeftWidth,
      // 等于 left + width
      bottom: triggerTopHeigh,
      // 等于 top + height
      left: triggerLeft,
      width: triggerWidth,
      height: triggerHeight
    } = triggerEl.value.getBoundingClientRect();
    const triggerRight = windowWidth.value - triggerLeftWidth;
    const triggerBottom = windowHeight.value - triggerTopHeigh;
    const popoverWidth = popoverEl.value.offsetWidth;
    const popoverHeight = popoverEl.value.offsetHeight;
    const x = window.scrollX + (offset || xOffset);
    const y = window.scrollY + (offset || yOffset);
    let popoverTop = AUTO;
    let popoverRight = AUTO;
    let popoverBottom = AUTO;
    let popoverLeft = AUTO;
    const placementIsY = ["top", "bottom"].some((item) => placement.startsWith(item));
    const placementIsX = ["left", "right"].some((item) => placement.startsWith(item));
    const expectTop = () => triggerTop + triggerHeight + y;
    const expectRight = () => triggerRight + triggerWidth - x - scrollbarWidth.value;
    const expectBottom = () => triggerBottom + triggerHeight - y - scrollbarHeight.value;
    const expectLeft = () => triggerLeft + triggerWidth + x;
    if (placementIsY) {
      if (placement.endsWith("start")) popoverLeft = expectLeft() - triggerWidth;
      else if (placement.endsWith("end")) popoverRight = expectRight() - triggerWidth;
      else popoverLeft = triggerLeft + triggerWidth / 2 - popoverWidth / 2 + x;
    } else if (placementIsX) {
      if (placement.endsWith("start")) popoverTop = expectTop() - triggerHeight;
      else if (placement.endsWith("end")) popoverBottom = expectBottom() - triggerHeight;
      else popoverTop = triggerTop + triggerHeight / 2 - popoverHeight / 2 + y;
    }
    if (placement.startsWith("top")) popoverBottom = expectBottom() + defaultSpace;
    else if (placement.startsWith("right")) popoverLeft = expectLeft() + defaultSpace;
    else if (placement.startsWith("bottom")) popoverTop = expectTop() + defaultSpace;
    else if (placement.startsWith("left")) popoverRight = expectRight() + defaultSpace;
    const isOverTop = () => !index.isString(popoverBottom) && popoverBottom + popoverHeight > windowHeight.value - y;
    const isOverRight = () => !index.isString(popoverLeft) && popoverLeft + popoverWidth > windowWidth.value + x;
    const isOverBottom = () => !index.isString(popoverTop) && popoverTop + popoverHeight > windowHeight.value + y;
    const isOverLeft = () => !index.isString(popoverRight) && popoverRight + popoverWidth > windowWidth.value - x;
    if (isOverTop()) {
      popoverTop = (placementIsX ? expectTop() - triggerHeight : expectTop()) + defaultSpace;
      if (isOverBottom()) popoverTop = AUTO;
      else popoverBottom = AUTO;
    }
    if (isOverBottom()) {
      popoverBottom = (placementIsX ? expectBottom() - triggerHeight : expectBottom()) + defaultSpace;
      if (isOverTop()) popoverBottom = AUTO;
      else popoverTop = AUTO;
    }
    if (isOverRight()) {
      popoverRight = (placementIsY ? expectRight() - triggerWidth : expectRight()) + defaultSpace;
      if (isOverLeft()) popoverRight = AUTO;
      else popoverLeft = AUTO;
    }
    if (isOverLeft()) {
      popoverLeft = (placementIsY ? expectLeft() - triggerWidth : expectLeft()) + defaultSpace;
      if (isOverRight()) popoverLeft = AUTO;
      else popoverRight = AUTO;
    }
    top.value = popoverTop;
    right.value = popoverRight;
    bottom.value = popoverBottom;
    left.value = popoverLeft;
  };
  calculatePosition();
  const update = () => {
    calculatePosition();
  };
  useEventListener.useEventListener(() => window, "scroll", update);
  useEventListener.useEventListener(() => window, "resize", update);
  return { top, right, bottom, left, update };
};

exports.usePopoverSize = usePopoverSize;
