// vitepress-theme-teek@1.5.1/es/components/theme/LoginPage/src/login.mjs
import 'vue';
import { isClient } from '../../../../helper/dist/index.mjs';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import '../../../../composables/useZIndex.mjs';

const getLoginStorageKey = () => {
  if (!isClient) return { siteLoginKey: "", pagesLoginKey: "", pageLoginKey: "", realmLoginKey: "" };
  const ns = useNamespace();
  const siteLoginKey = ns.storageKey("private", "site", window.location.hostname);
  const pagesLoginKey = ns.storageKey("private", "pages", window.location.hostname);
  const pageLoginKey = ns.storageKey("private", "page", window.location.hostname);
  const realmLoginKey = ns.storageKey("private", "realm", window.location.hostname);
  return { siteLoginKey, pagesLoginKey, pageLoginKey, realmLoginKey };
};
const loginUrlKeyMap = {
  realm: "realm",
  toPath: "toPath",
  verifyMode: "verifyMode"
};
const verifyModeMap = {
  site: "site",
  pages: "pages",
  page: "page",
  realm: "realm"
};
const defaultPrivateConfig = {
  enabled: false,
  expire: "1d",
  session: false,
  siteLogin: false,
  site: [],
  pages: [],
  realm: {}
};

export { defaultPrivateConfig, getLoginStorageKey, loginUrlKeyMap, verifyModeMap };
