// vitepress-theme-teek@1.5.1/lib/components/theme/CommentArtalk/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var index$1 = require('../../../../helper/dist/index.js');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useVpRouter = require('../../../../composables/useVpRouter.js');
require('../../../../composables/useZIndex.js');
var index = require('../../ConfigProvider/index.js');
var artalk = require('./artalk.js');

const _hoisted_1 = ["href"];
const artalkId = "artalk";
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "CommentArtalk" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace.useNamespace();
    const vpRouter = useVpRouter.useVpRouter();
    const { getTeekConfig } = index.useTeekConfig();
    const { isDark, page } = vitepress.useData();
    const artalkOptions = getTeekConfig("comment", {}).options;
    const { server, site, ...options } = artalkOptions;
    const artalkRef = Vue.ref(null);
    const artalkJs = Vue.ref(null);
    const artalk$1 = Vue.ref();
    const initArtalkByInject = () => {
      const getArtalkInstance = Vue.inject(artalk.artalkContext, () => null);
      const el = artalkRef.value || `#${artalkId}`;
      const artalkInstance = getArtalkInstance?.(el, artalkOptions);
      if (!artalkInstance) return false;
      artalk$1.value = artalkInstance;
      switchDark();
      return true;
    };
    const initArtalkByJs = () => {
      if (!index$1.isClient) return console.error("[Teek Error] Not in a client");
      const Artalk = window.Artalk;
      const el = artalkRef.value || `#${artalkId}`;
      if (!Artalk || !artalkRef.value) {
        return console.error("[Teek Error] Artalk initialization failed. Unable to load online js file from " + server);
      }
      artalk$1.value = Artalk.init({
        darkMode: isDark.value,
        ...options,
        el,
        pageKey: vpRouter.route.path,
        pageTitle: page.value.title,
        server,
        site
      });
      switchDark();
    };
    const initJs = () => {
      const t = artalkJs.value;
      if (t) t.onload = initArtalkByJs;
    };
    const reloadArtalk = () => {
      const a = artalk$1.value;
      a?.update({
        pageKey: vpRouter.route.path,
        pageTitle: page.value.title
      });
      a?.reload();
    };
    Vue.onMounted(() => {
      if (!initArtalkByInject() && server) {
        initJs();
        return artalk$1.value && vpRouter.bindAfterRouteChange(ns.join("artalk"), () => reloadArtalk());
      }
      console.error(
        "[Teek Error] Artalk initialization failed. Please configure the 'server' and 'site' or provide the artalk instance"
      );
    });
    Vue.onUnmounted(() => {
      const a = artalk$1.value;
      if (a) a.destroy();
    });
    const switchDark = () => {
      setTimeout(() => {
        const a = artalk$1.value;
        if (a) a.setDarkMode(isDark.value);
      }, 100);
    };
    Vue.watch(isDark, () => switchDark());
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock(
        "div",
        {
          class: Vue.normalizeClass(Vue.unref(ns).b("artalk"))
        },
        [
          Vue.unref(server) ? (Vue.openBlock(), Vue.createElementBlock("link", {
            key: 0,
            rel: "stylesheet",
            href: `${Vue.unref(server)}/dist/Artalk.css`,
            crossorigin: "anonymous"
          }, null, 8, _hoisted_1)) : Vue.createCommentVNode("v-if", true),
          Vue.createElementVNode(
            "div",
            {
              id: artalkId,
              ref_key: "artalkRef",
              ref: artalkRef
            },
            null,
            512
            /* NEED_PATCH */
          ),
          Vue.unref(server) ? (Vue.openBlock(), Vue.createBlock(Vue.resolveDynamicComponent("script"), {
            key: 1,
            src: `${Vue.unref(server)}/dist/Artalk.js`,
            crossorigin: "anonymous",
            ref_key: "artalkJs",
            ref: artalkJs
          }, null, 8, ["src"])) : Vue.createCommentVNode("v-if", true)
        ],
        2
        /* CLASS */
      );
    };
  }
});

exports.default = _sfc_main;
