// vitepress-theme-teek/lib/components/common/Breadcrumb/src/Breadcrumb.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('vitepress');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var breadcrumb = require('./breadcrumb.js');
var namespace = require('./namespace.js');

const _hoisted_1 = ["aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "Breadcrumb" },
  __name: "Breadcrumb",
  props: {
    separator: { default: "/" }
  },
  setup(__props) {
    const { t } = useLocale.useLocale();
    const breadcrumb$1 = Vue.ref();
    Vue.provide(breadcrumb.breadcrumbKey, { separator: __props.separator });
    Vue.onMounted(() => {
      const items = breadcrumb$1.value?.querySelectorAll(`.${namespace.ns.e("item")}`);
      if (items?.length) {
        items[items.length - 1].setAttribute("aria-current", "page");
      }
    });
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("div", {
        ref_key: "breadcrumb",
        ref: breadcrumb$1,
        class: Vue.normalizeClass(Vue.unref(namespace.ns).b()),
        role: "navigation",
        "aria-label": Vue.unref(t)("tk.breadcrumb.label")
      }, [
        Vue.renderSlot(_ctx.$slots, "default")
      ], 10, _hoisted_1);
    };
  }
});

exports.default = _sfc_main;
