// vitepress-theme-teek/es/components/common/VpContainer/src/index.vue2.mjs
import { defineComponent, useSlots, computed, createElementBlock, createCommentVNode, openBlock, normalizeClass, unref, createElementVNode, renderSlot } from 'vue';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import '../../../../composables/useZIndex.mjs';

const _hoisted_1 = {
  key: 0,
  class: "custom-block-title"
};
const _hoisted_2 = ["innerHTML"];
const _hoisted_3 = { key: 1 };
const _hoisted_4 = ["innerHTML"];
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "VpContainer" },
  __name: "index",
  props: {
    type: { default: "tip" },
    title: {},
    text: { default: "" }
  },
  setup(__props) {
    const ns = useNamespace("vp-container");
    const slots = useSlots();
    const hasTitle = computed(() => __props.title || slots.title);
    const hasText = computed(() => __props.text || slots.default);
    return (_ctx, _cache) => {
      return hasTitle.value || hasText.value ? (openBlock(), createElementBlock(
        "div",
        {
          key: 0,
          class: normalizeClass([unref(ns).b(), "vp-doc"])
        },
        [
          createElementVNode(
            "div",
            {
              class: normalizeClass([__props.type, "custom-block", { "no-title": !__props.title }])
            },
            [
              hasTitle.value ? (openBlock(), createElementBlock("div", _hoisted_1, [
                renderSlot(_ctx.$slots, "title", {}, () => [
                  createElementVNode("span", { innerHTML: __props.title }, null, 8, _hoisted_2)
                ])
              ])) : createCommentVNode("v-if", true),
              hasText.value ? (openBlock(), createElementBlock("p", _hoisted_3, [
                renderSlot(_ctx.$slots, "default", {}, () => [
                  createElementVNode("span", { innerHTML: __props.text }, null, 8, _hoisted_4)
                ])
              ])) : createCommentVNode("v-if", true)
            ],
            2
            /* CLASS */
          )
        ],
        2
        /* CLASS */
      )) : createCommentVNode("v-if", true);
    };
  }
});

export { _sfc_main as default };
