// vitepress-theme-teek/es/components/common/TitleTag/src/index.vue2.mjs
import { defineComponent, createElementBlock, createCommentVNode, openBlock, normalizeClass, unref, renderSlot, createTextVNode, toDisplayString } from 'vue';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import '../../../../composables/useZIndex.mjs';

var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "TitleTag" },
  __name: "index",
  props: {
    text: {},
    type: {},
    position: {},
    size: {}
  },
  setup(__props) {
    const ns = useNamespace("title-tag");
    return (_ctx, _cache) => {
      return __props.text || _ctx.$slots.default ? (openBlock(), createElementBlock(
        "span",
        {
          key: 0,
          class: normalizeClass([unref(ns).b(), __props.type && unref(ns).m(__props.type), __props.position && unref(ns).m(__props.position), __props.size && unref(ns).m(__props.size)])
        },
        [
          renderSlot(_ctx.$slots, "default", {}, () => [
            createTextVNode(
              toDisplayString(__props.text),
              1
              /* TEXT */
            )
          ])
        ],
        2
        /* CLASS */
      )) : createCommentVNode("v-if", true);
    };
  }
});

export { _sfc_main as default };
