// vitepress-theme-teek@1.5.1/lib/components/common/PageCard/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var index = require('../../../../helper/dist/index.js');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var arrowLeft = require('../../../../static/icons/arrowLeft.js');
var arrowRight = require('../../../../static/icons/arrowRight.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../Icon/src/index.vue2.js');
require('@iconify/vue');

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["href", "target", "aria-label"];
const _hoisted_3 = ["innerHTML"];
const _hoisted_4 = ["aria-label"];
const _hoisted_5 = ["innerHTML"];
const _hoisted_6 = ["innerHTML"];
const _hoisted_7 = { key: 0 };
const _hoisted_8 = ["aria-label", "aria-disabled"];
const _hoisted_9 = ["aria-label", "aria-disabled"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "PageCard" },
  __name: "index",
  props: /* @__PURE__ */ Vue.mergeModels({
    title: { default: "" },
    titleLink: {},
    titleClick: { type: Function },
    page: { type: Boolean, default: false },
    pageSize: { default: 4 },
    total: { default: 0 },
    autoPage: { type: Boolean, default: false },
    pageSpeed: { default: 4e3 }
  }, {
    "modelValue": { default: 1 },
    "modelModifiers": {}
  }),
  emits: /* @__PURE__ */ Vue.mergeModels(["pagination"], ["update:modelValue"]),
  setup(__props, { emit: __emit }) {
    const ns = useNamespace.useNamespace("page-card");
    const pointClass = ns.join("pointer");
    const { t } = useLocale.useLocale();
    const emit = __emit;
    const pageNum = Vue.useModel(__props, "modelValue");
    const pageTotalNum = Math.ceil(__props.total / __props.pageSize);
    const hasNextData = __props.total !== 0 && pageTotalNum !== 1;
    const transitionName = Vue.ref(ns.join("scroll"));
    const pagination = (to, type) => {
      emit("pagination", to, type);
      transitionName.value = ns.join(`slide-${type}`);
      if (__props.page && __props.autoPage) startAutoPage();
      const index = pageNum.value % pageTotalNum;
      const res = (index + to) % pageTotalNum;
      if (res < 1) pageNum.value = pageTotalNum + res;
      else if (res > pageTotalNum) pageNum.value = 1;
      else pageNum.value = res;
    };
    let timer;
    const startAutoPage = () => {
      closeAutoPage();
      if (__props.pageSpeed > 0) {
        timer = setTimeout(() => {
          pagination(1, "next");
        }, __props.pageSpeed);
      }
    };
    const closeAutoPage = () => {
      if (timer) {
        clearTimeout(timer);
        timer = null;
      }
    };
    Vue.onMounted(() => {
      if (__props.page && __props.autoPage) startAutoPage();
    });
    Vue.onUnmounted(() => {
      if (__props.page && __props.autoPage) closeAutoPage();
    });
    const removeHtmlTag = (str) => {
      return str.replace(/<[^>]*>/g, "").trim();
    };
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("div", {
        class: Vue.normalizeClass(`${Vue.unref(ns).b()} card`),
        "aria-label": Vue.unref(t)("tk.pageCard.label")
      }, [
        __props.title ? (Vue.openBlock(), Vue.createElementBlock(
          "div",
          {
            key: 0,
            class: Vue.normalizeClass(`${Vue.unref(ns).e("header")} flx-justify-between`)
          },
          [
            Vue.renderSlot(_ctx.$slots, "title", {}, () => [
              __props.titleLink ? (Vue.openBlock(), Vue.createElementBlock("a", {
                key: 0,
                href: Vue.unref(vitepress.withBase)(__props.titleLink),
                target: Vue.unref(index.isValidURL)(__props.titleLink) ? "_blank" : "_self",
                "aria-label": removeHtmlTag(__props.title)
              }, [
                Vue.createElementVNode("span", {
                  class: "title flx-align-center",
                  innerHTML: __props.title
                }, null, 8, _hoisted_3)
              ], 8, _hoisted_2)) : __props.titleClick ? (Vue.openBlock(), Vue.createElementBlock("a", {
                key: 1,
                onClick: _cache[0] || (_cache[0] = () => __props.titleClick()),
                class: Vue.normalizeClass(Vue.unref(pointClass)),
                "aria-label": removeHtmlTag(__props.title)
              }, [
                Vue.createElementVNode("span", {
                  class: "title flx-align-center",
                  innerHTML: __props.title
                }, null, 8, _hoisted_5)
              ], 10, _hoisted_4)) : (Vue.openBlock(), Vue.createElementBlock("span", {
                key: 2,
                class: "title flx-align-center",
                innerHTML: __props.title
              }, null, 8, _hoisted_6))
            ]),
            Vue.renderSlot(_ctx.$slots, "page", Vue.normalizeProps(Vue.guardReactiveProps({ pagination })), () => [
              __props.page ? (Vue.openBlock(), Vue.createElementBlock("div", _hoisted_7, [
                Vue.renderSlot(_ctx.$slots, "page-left", Vue.normalizeProps(Vue.guardReactiveProps({ pagination })), () => [
                  Vue.createElementVNode("span", {
                    class: Vue.normalizeClass(["page-button", Vue.unref(hasNextData) ? Vue.unref(pointClass) : "disabled"]),
                    onClick: _cache[1] || (_cache[1] = ($event) => pagination(-1, "prev")),
                    role: "button",
                    "aria-label": Vue.unref(t)("tk.pageCard.prev"),
                    "aria-disabled": !Vue.unref(hasNextData)
                  }, [
                    Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                      icon: Vue.unref(arrowLeft.default),
                      size: 14,
                      "aria-hidden": "true"
                    }, null, 8, ["icon"])
                  ], 10, _hoisted_8)
                ]),
                Vue.renderSlot(_ctx.$slots, "page-right", Vue.normalizeProps(Vue.guardReactiveProps({ pagination })), () => [
                  Vue.createElementVNode("span", {
                    class: Vue.normalizeClass(["page-button", Vue.unref(hasNextData) ? Vue.unref(pointClass) : "disabled"]),
                    onClick: _cache[2] || (_cache[2] = ($event) => pagination(1, "next")),
                    role: "button",
                    "aria-label": Vue.unref(t)("tk.pageCard.next"),
                    "aria-disabled": !Vue.unref(hasNextData)
                  }, [
                    Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                      icon: Vue.unref(arrowRight.default),
                      size: 14,
                      "aria-hidden": "true"
                    }, null, 8, ["icon"])
                  ], 10, _hoisted_9)
                ])
              ])) : Vue.createCommentVNode("v-if", true)
            ])
          ],
          2
          /* CLASS */
        )) : Vue.createCommentVNode("v-if", true),
        Vue.renderSlot(_ctx.$slots, "default", Vue.normalizeProps(Vue.guardReactiveProps({ transitionName: transitionName.value, startAutoPage, closeAutoPage })))
      ], 10, _hoisted_1);
    };
  }
});

exports.default = _sfc_main;
