// vitepress-theme-teek@1.5.1/lib/components/common/Message/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var DefaultTheme = require('vitepress/theme');
var useEventListener = require('../../../../composables/useEventListener.js');
require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useZIndex = require('../../../../composables/useZIndex.js');
var close = require('../../../../static/icons/close.js');
var circleCloseFilled = require('../../../../static/icons/circleCloseFilled.js');
var infoFilled = require('../../../../static/icons/infoFilled.js');
var successFilled = require('../../../../static/icons/successFilled.js');
var warningFilled = require('../../../../static/icons/warningFilled.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../Icon/src/index.vue2.js');
require('@iconify/vue');
var message = require('./message.js');
var instance = require('./instance.js');

const _hoisted_1 = ["id"];
const _hoisted_2 = ["innerHTML"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "Message" },
  __name: "index",
  props: /* @__PURE__ */ Vue.mergeDefaults({
    customClass: {},
    center: { type: Boolean },
    dangerouslyUseHTMLString: { type: Boolean },
    duration: {},
    icon: {},
    id: {},
    message: { type: [String, Object, Function] },
    onClose: { type: Function },
    showClose: { type: Boolean },
    type: {},
    plain: { type: Boolean },
    offset: {},
    zIndex: {},
    grouping: { type: Boolean },
    repeatNum: {}
  }, message.messagePropsDefaults),
  emits: ["destroy"],
  setup(__props, { expose: __expose, emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const iconsMap = {
      success: successFilled.default,
      warning: warningFilled.default,
      error: circleCloseFilled.default,
      info: infoFilled.default
    };
    const isStartTransition = Vue.ref(false);
    const ns = useNamespace.useNamespace("message");
    const { currentZIndex, nextZIndex } = useZIndex.useZIndex();
    const messageRef = Vue.ref();
    const visible = Vue.ref(false);
    const height = Vue.ref(0);
    const badeTypeMap = {
      info: "info",
      primary: "primary",
      success: "success",
      warning: "warning",
      error: "danger"
    };
    const badgeType = Vue.computed(() => props.type ? badeTypeMap[props.type] : "info");
    const typeClass = Vue.computed(() => {
      const type = props.type;
      return { [ns.bm("icon", type)]: type && iconsMap[type] };
    });
    const iconComponent = Vue.computed(() => props.icon || iconsMap[props.type] || "");
    const lastOffset = Vue.computed(() => instance.getLastOffset(props.id));
    const offset = Vue.computed(() => instance.getOffsetOrSpace(props.id, props.offset) + lastOffset.value);
    const bottom = Vue.computed(() => height.value + offset.value);
    const customStyle = Vue.computed(() => ({
      top: `${offset.value}px`,
      zIndex: currentZIndex.value
    }));
    let timer;
    const clearTimer = () => {
      if (timer) {
        clearTimeout(timer);
        timer = null;
      }
    };
    const startTimer = () => {
      if (props.duration === 0) return;
      clearTimer();
      timer = setTimeout(() => {
        close$1();
      }, props.duration);
    };
    const close$1 = () => {
      visible.value = false;
      Vue.nextTick(() => {
        if (!isStartTransition.value) {
          props.onClose?.();
          emit("destroy");
        }
      });
    };
    const keydown = ({ code }) => {
      if (code === "Space") {
        close$1();
      }
    };
    let resizeObserver;
    Vue.onMounted(() => {
      startTimer();
      nextZIndex();
      visible.value = true;
      resizeObserver = new ResizeObserver(() => {
        if (messageRef.value) height.value = messageRef.value.getBoundingClientRect().height;
      });
      if (messageRef.value) resizeObserver.observe(messageRef.value);
    });
    Vue.onUnmounted(() => {
      if (resizeObserver && messageRef.value) resizeObserver.unobserve(messageRef.value);
    });
    Vue.watch(
      () => props.repeatNum,
      () => {
        clearTimer();
        startTimer();
      }
    );
    useEventListener.useEventListener(document, "keydown", keydown);
    __expose({
      visible,
      bottom,
      close: close$1
    });
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createBlock(Vue.Transition, {
        name: Vue.unref(ns).b("fade"),
        onBeforeEnter: _cache[0] || (_cache[0] = ($event) => isStartTransition.value = true),
        onBeforeLeave: __props.onClose,
        onAfterLeave: _cache[1] || (_cache[1] = ($event) => _ctx.$emit("destroy")),
        persisted: ""
      }, {
        default: Vue.withCtx(() => [
          Vue.withDirectives(Vue.createElementVNode("div", {
            id: __props.id,
            ref_key: "messageRef",
            ref: messageRef,
            class: Vue.normalizeClass([
              Vue.unref(ns).b(),
              { [Vue.unref(ns).m(__props.type)]: __props.type },
              Vue.unref(ns).is("center", __props.center),
              Vue.unref(ns).is("closable", __props.showClose),
              Vue.unref(ns).is("plain", __props.plain),
              __props.customClass
            ]),
            style: Vue.normalizeStyle(customStyle.value),
            role: "alert",
            onMouseenter: clearTimer,
            onMouseleave: startTimer
          }, [
            __props.repeatNum > 1 ? (Vue.openBlock(), Vue.createBlock(Vue.unref(DefaultTheme.VPBadge), {
              key: 0,
              text: __props.repeatNum,
              type: badgeType.value,
              class: Vue.normalizeClass(Vue.unref(ns).e("badge"))
            }, null, 8, ["text", "type", "class"])) : Vue.createCommentVNode("v-if", true),
            iconComponent.value ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
              key: 1,
              icon: iconComponent.value,
              class: Vue.normalizeClass([Vue.unref(ns).e("icon"), typeClass.value])
            }, null, 8, ["icon", "class"])) : Vue.createCommentVNode("v-if", true),
            Vue.renderSlot(_ctx.$slots, "default", {}, () => [
              !__props.dangerouslyUseHTMLString ? (Vue.openBlock(), Vue.createElementBlock(
                "p",
                {
                  key: 0,
                  class: Vue.normalizeClass(Vue.unref(ns).e("content"))
                },
                Vue.toDisplayString(__props.message),
                3
                /* TEXT, CLASS */
              )) : (Vue.openBlock(), Vue.createElementBlock("p", {
                key: 1,
                class: Vue.normalizeClass(Vue.unref(ns).e("content")),
                innerHTML: __props.message
              }, null, 10, _hoisted_2))
            ]),
            __props.showClose ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
              key: 2,
              icon: Vue.unref(close.default),
              class: Vue.normalizeClass(Vue.unref(ns).e("closeBtn")),
              onClick: Vue.withModifiers(close$1, ["stop"])
            }, null, 8, ["icon", "class"])) : Vue.createCommentVNode("v-if", true)
          ], 46, _hoisted_1), [
            [Vue.vShow, visible.value]
          ])
        ]),
        _: 3
        /* FORWARDED */
      }, 8, ["name", "onBeforeLeave"]);
    };
  }
});

exports.default = _sfc_main;
