// vitepress-theme-teek@1.5.1/es/components/common/ArticlePage/src/components/DocAsideOutlineItem.vue2.mjs
import { defineComponent, resolveComponent, createElementBlock, openBlock, normalizeClass, unref, Fragment, renderList, createElementVNode, createBlock, createCommentVNode, toDisplayString } from 'vue';
import 'vitepress';
import { useNamespace } from '../../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import '../../../../../composables/useZIndex.mjs';

const _hoisted_1 = ["href", "title"];
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "DocAsideOutlineItem" },
  __name: "DocAsideOutlineItem",
  props: {
    headers: {},
    root: { type: Boolean }
  },
  setup(__props) {
    const ns = useNamespace("aside-outline-item");
    function onClick({ target: el }) {
      const id = el.href.split("#")[1];
      const heading = document.getElementById(decodeURIComponent(id));
      heading?.focus({ preventScroll: true });
    }
    return (_ctx, _cache) => {
      const _component_DocAsideOutlineItem = resolveComponent("DocAsideOutlineItem", true);
      return openBlock(), createElementBlock(
        "ul",
        {
          class: normalizeClass([unref(ns).b(), __props.root ? unref(ns).is("root") : unref(ns).is("nested")])
        },
        [
          (openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(__props.headers, ({ children, link, title }) => {
              return openBlock(), createElementBlock("li", { key: link }, [
                createElementVNode("a", {
                  class: normalizeClass(unref(ns).m("link")),
                  href: link,
                  onClick,
                  title
                }, toDisplayString(title), 11, _hoisted_1),
                children?.length ? (openBlock(), createBlock(_component_DocAsideOutlineItem, {
                  key: 0,
                  headers: children
                }, null, 8, ["headers"])) : createCommentVNode("v-if", true)
              ]);
            }),
            128
            /* KEYED_FRAGMENT */
          ))
        ],
        2
        /* CLASS */
      );
    };
  }
});

export { _sfc_main as default };
