// vitepress-theme-teek@1.5.1/lib/components/common/VerifyCode/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var useImgCode = require('./useImgCode.js');

var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "ImageVerifyCode" },
  __name: "index",
  props: {
    "modelValue": { required: true },
    "modelModifiers": {}
  },
  emits: ["update:modelValue"],
  setup(__props, { expose: __expose }) {
    const { domRef, imgCode, setImgCode, getImgCode } = useImgCode.useImageVerify();
    const code = Vue.useModel(__props, "modelValue");
    Vue.watch(code, (newValue) => {
      setImgCode(newValue);
    });
    Vue.watch(imgCode, (newValue) => {
      code.value = newValue;
    });
    __expose({ getImgCode });
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock(
        "canvas",
        {
          ref_key: "domRef",
          ref: domRef,
          width: "120",
          height: "40",
          class: "cursor-pointer",
          onClick: _cache[0] || (_cache[0] = //@ts-ignore
          (...args) => Vue.unref(getImgCode) && Vue.unref(getImgCode)(...args))
        },
        null,
        512
        /* NEED_PATCH */
      );
    };
  }
});

exports.default = _sfc_main;
