// vitepress-theme-teek/lib/components/theme/Layout/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var DefaultTheme = require('vitepress/theme');
var vitepress = require('vitepress');
var index$1 = require('../../../../helper/dist/index.js');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var index = require('../../ConfigProvider/index.js');
var index_vue_vue_type_script_setup_true_name_Home_lang = require('../../Home/src/index.vue2.js');
var HomeMyCardScreen_vue_vue_type_script_setup_true_name_HomeMyCardScreen_lang = require('../../HomeMyCard/src/HomeMyCardScreen.vue2.js');
var index_vue_vue_type_script_setup_true_name_BodyBgImage_lang = require('../../BodyBgImage/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_FooterGroup_lang = require('../../FooterGroup/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_FooterInfo_lang = require('../../FooterInfo/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleImagePreview_lang = require('../../ArticleImagePreview/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleAnalyze_lang = require('../../ArticleAnalyze/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleShare_lang = require('../../ArticleShare/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleUpdate_lang = require('../../ArticleUpdate/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleHeadingHighlight_lang = require('../../ArticleHeadingHighlight/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticlePageStyle_lang = require('../../ArticlePageStyle/src/index.vue2.js');
var AsideBottomAppreciation_vue_vue_type_script_setup_true_name_AsideBottomAppreciation_lang = require('../../ArticleAppreciation/src/AsideBottomAppreciation.vue2.js');
var DocAfterAppreciation_vue_vue_type_script_setup_true_name_DocAfterAppreciation_lang = require('../../ArticleAppreciation/src/DocAfterAppreciation.vue2.js');
var DocAfterAppreciationPopper_vue_vue_type_script_setup_true_name_DocAfterAppreciationPopper_lang = require('../../ArticleAppreciation/src/DocAfterAppreciationPopper.vue2.js');
var index_vue_vue_type_script_setup_true_name_CommentTwikoo_lang = require('../../CommentTwikoo/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_CommentArtalk_lang = require('../../CommentArtalk/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_CommentGiscus_lang = require('../../CommentGiscus/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_CommentWaline_lang = require('../../CommentWaline/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_CodeBlockToggle_lang = require('../../CodeBlockToggle/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_RightBottomButton_lang = require('../../RightBottomButton/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_Notice_lang = require('../../Notice/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ThemeEnhance_lang = require('../../ThemeEnhance/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_VpContainer_lang = require('../../../common/VpContainer/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArchivesPage_lang = require('../../ArchivesPage/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_CataloguePage_lang = require('../../CataloguePage/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleOverviewPage_lang = require('../../ArticleOverviewPage/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_LoginPage_lang = require('../../LoginPage/src/index.vue2.js');
var useWatchLogin = require('../../LoginPage/src/useWatchLogin.js');
var index_vue_vue_type_script_setup_true_name_RiskLinkPage_lang = require('../../RiskLinkPage/src/index.vue2.js');
var useRiskLink = require('../../RiskLinkPage/src/useRiskLink.js');
var index_vue_vue_type_script_setup_true_name_SidebarTrigger_lang = require('../../SidebarTrigger/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_lang = require('../../HomeFeature/src/index.vue2.js');
var index$2 = require('../../RouteLoading/src/index.vue.js');
var index_vue_vue_type_script_setup_true_name_ArticleBanner_lang = require('../../ArticleBanner/src/index.vue2.js');

var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "TeekLayout" },
  __name: "index",
  props: {
    locale: {}
  },
  setup(__props) {
    const props = __props;
    Vue.provide(
      useLocale.localeContextKey,
      Vue.computed(() => props.locale)
    );
    const { Layout } = DefaultTheme;
    const ns = useNamespace.useNamespace("layout");
    const { getTeekConfigRef } = index.useTeekConfig();
    const { isHomePage, isArchivesPage, isCataloguePage, isArticleOverviewPage } = index.usePageState();
    const { frontmatter, localeIndex, page } = vitepress.useData();
    const { hasSidebar } = index.useSidebar();
    const teekConfig = getTeekConfigRef(null, {
      teekTheme: true,
      teekHome: true,
      vpHome: true,
      sidebarTrigger: false,
      loading: false,
      codeBlock: { enabled: true },
      themeSize: "",
      bodyBgImg: {},
      notice: {},
      comment: { provider: "" },
      articleUpdate: { enabled: true },
      articleTopTip: void 0,
      articleBottomTip: void 0,
      articleShare: {},
      appreciation: {},
      riskLink: { enabled: false },
      themeEnhance: { enabled: true },
      articleBanner: { enabled: false },
      pageStyle: "default"
    });
    const loading = Vue.ref(teekConfig.value.loading);
    const commentConfig = Vue.computed(() => {
      const comment = frontmatter.value.comment ?? teekConfig.value.comment;
      if (index$1.isBoolean(comment)) return { enabled: comment };
      return {
        enabled: true,
        components: {
          twikoo: index_vue_vue_type_script_setup_true_name_CommentTwikoo_lang.default,
          waline: index_vue_vue_type_script_setup_true_name_CommentWaline_lang.default,
          giscus: index_vue_vue_type_script_setup_true_name_CommentGiscus_lang.default,
          artalk: index_vue_vue_type_script_setup_true_name_CommentArtalk_lang.default
        },
        provider: comment.provider,
        options: comment.options
      };
    });
    const topTipConfig = Vue.computed(() => {
      if (index$1.isBoolean(teekConfig.value.articleTopTip)) return teekConfig.value.articleTopTip;
      return teekConfig.value.articleTopTip?.(frontmatter.value, localeIndex.value, page.value);
    });
    const bottomTipConfig = Vue.computed(() => {
      if (index$1.isBoolean(teekConfig.value.articleBottomTip)) return teekConfig.value.articleBottomTip;
      return teekConfig.value.articleBottomTip?.(frontmatter.value, localeIndex.value, page.value);
    });
    const showArticleBanner = Vue.computed(
      () => frontmatter.value.articleBanner !== false && teekConfig.value.articleBanner.enabled && !hasSidebar.value && frontmatter.value.article !== false && (!frontmatter.value.layout || frontmatter.value.layout === "doc") && teekConfig.value.pageStyle === "default"
    );
    const themeSizeAttribute = ns.join("theme-size");
    Vue.watch(
      () => teekConfig.value.themeSize,
      (newValue) => {
        if (!index$1.isClient) return;
        if (newValue) document.documentElement.setAttribute(themeSizeAttribute, newValue);
        else document.documentElement.removeAttribute(themeSizeAttribute);
      },
      { immediate: true, flush: "post" }
    );
    const { watchSite, watchPages } = useWatchLogin.useWatchLogin();
    const { restart } = useRiskLink.useRiskLink({
      whitelist: teekConfig.value.riskLink.whitelist,
      blacklist: teekConfig.value.riskLink.blacklist
    });
    watchSite();
    watchPages();
    vitepress.onContentUpdated(() => {
      if (teekConfig.value.riskLink.enabled) restart();
    });
    const usedSlots = [
      "home-hero-before",
      "home-features-after",
      "nav-bar-content-after",
      "layout-top",
      "layout-bottom",
      "doc-footer-before",
      "doc-before",
      "doc-after",
      "aside-bottom",
      "page-top",
      "aside-outline-before",
      "sidebar-nav-before"
    ];
    return (_ctx, _cache) => {
      return Vue.unref(teekConfig).teekTheme ? (Vue.openBlock(), Vue.createElementBlock(
        Vue.Fragment,
        { key: 0 },
        [
          Vue.unref(teekConfig).loading ?? false ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index$2.default), {
            key: 0,
            modelValue: loading.value,
            "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => loading.value = $event)
          }, {
            default: Vue.withCtx((scope) => [
              Vue.renderSlot(_ctx.$slots, "teek-loading", Vue.normalizeProps(Vue.guardReactiveProps(scope)))
            ]),
            _: 3
            /* FORWARDED */
          }, 8, ["modelValue"])) : Vue.createCommentVNode("v-if", true),
          Vue.unref(frontmatter).loginPage === true ? Vue.renderSlot(_ctx.$slots, "teek-login-page", { key: 1 }, () => [
            Vue.withDirectives(Vue.createVNode(
              Vue.unref(index_vue_vue_type_script_setup_true_name_LoginPage_lang.default),
              null,
              null,
              512
              /* NEED_PATCH */
            ), [
              [Vue.vShow, !loading.value]
            ])
          ]) : Vue.createCommentVNode("v-if", true),
          Vue.unref(frontmatter).riskLinkPage === true ? Vue.renderSlot(_ctx.$slots, "teek-risk-link-page", { key: 2 }, () => [
            Vue.withDirectives(Vue.createVNode(
              Vue.unref(index_vue_vue_type_script_setup_true_name_RiskLinkPage_lang.default),
              null,
              null,
              512
              /* NEED_PATCH */
            ), [
              [Vue.vShow, !loading.value]
            ])
          ]) : Vue.createCommentVNode("v-if", true),
          Vue.unref(frontmatter).layout !== false ? (Vue.openBlock(), Vue.createElementBlock(
            Vue.Fragment,
            { key: 3 },
            [
              Vue.unref(teekConfig).bodyBgImg?.imgSrc ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_BodyBgImage_lang.default), { key: 0 })) : Vue.createCommentVNode("v-if", true),
              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleHeadingHighlight_lang.default)),
              Vue.unref(teekConfig).notice?.enabled ? (Vue.openBlock(), Vue.createBlock(
                Vue.unref(index_vue_vue_type_script_setup_true_name_Notice_lang.default),
                { key: 1 },
                Vue.createSlots({
                  _: 2
                  /* DYNAMIC */
                }, [
                  Vue.renderList(_ctx.$slots, (_, name) => {
                    return {
                      name,
                      fn: Vue.withCtx((scope) => [
                        Vue.renderSlot(_ctx.$slots, name, Vue.normalizeProps(Vue.guardReactiveProps(scope)))
                      ])
                    };
                  })
                ]),
                1024
                /* DYNAMIC_SLOTS */
              )) : Vue.createCommentVNode("v-if", true),
              Vue.createVNode(
                Vue.unref(index_vue_vue_type_script_setup_true_name_RightBottomButton_lang.default),
                null,
                Vue.createSlots({
                  _: 2
                  /* DYNAMIC */
                }, [
                  Vue.renderList(_ctx.$slots, (_, name) => {
                    return {
                      name,
                      fn: Vue.withCtx((scope) => [
                        Vue.renderSlot(_ctx.$slots, name, Vue.normalizeProps(Vue.guardReactiveProps(scope)))
                      ])
                    };
                  })
                ]),
                1024
                /* DYNAMIC_SLOTS */
              )
            ],
            64
            /* STABLE_FRAGMENT */
          )) : Vue.createCommentVNode("v-if", true),
          Vue.withDirectives(Vue.createVNode(Vue.unref(Layout), {
            class: Vue.normalizeClass([
              Vue.unref(ns).b(),
              { [Vue.unref(ns).m("hide-vp-home")]: !Vue.unref(teekConfig).vpHome },
              Vue.unref(ns).has("sidebar-trigger", Vue.unref(teekConfig).sidebarTrigger)
            ])
          }, Vue.createSlots({
            "home-hero-before": Vue.withCtx(() => [
              Vue.renderSlot(_ctx.$slots, "home-hero-before"),
              Vue.renderSlot(_ctx.$slots, "teek-home-before"),
              Vue.createCommentVNode(" \u81EA\u5B9A\u4E49\u9996\u9875 "),
              Vue.unref(teekConfig).teekHome ? (Vue.openBlock(), Vue.createBlock(
                Vue.unref(index_vue_vue_type_script_setup_true_name_Home_lang.default),
                { key: 0 },
                Vue.createSlots({
                  _: 2
                  /* DYNAMIC */
                }, [
                  Vue.renderList(_ctx.$slots, (_, name) => {
                    return {
                      name,
                      fn: Vue.withCtx((scope) => [
                        Vue.renderSlot(_ctx.$slots, name, Vue.normalizeProps(Vue.guardReactiveProps(scope)))
                      ])
                    };
                  })
                ]),
                1024
                /* DYNAMIC_SLOTS */
              )) : Vue.createCommentVNode("v-if", true),
              Vue.renderSlot(_ctx.$slots, "teek-home-after")
            ]),
            "home-features-after": Vue.withCtx(() => [
              !Vue.unref(teekConfig).teekHome ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 0 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-home-features-before"),
                  Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_lang.default)),
                  Vue.renderSlot(_ctx.$slots, "teek-home-features-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true),
              Vue.renderSlot(_ctx.$slots, "home-features-after")
            ]),
            "nav-bar-content-after": Vue.withCtx(() => [
              Vue.renderSlot(_ctx.$slots, "nav-bar-content-after"),
              Vue.unref(teekConfig).themeEnhance.enabled ?? true ? (Vue.openBlock(), Vue.createBlock(
                Vue.unref(index_vue_vue_type_script_setup_true_name_ThemeEnhance_lang.default),
                { key: 0 },
                Vue.createSlots({
                  _: 2
                  /* DYNAMIC */
                }, [
                  Vue.renderList(_ctx.$slots, (_, name) => {
                    return {
                      name,
                      fn: Vue.withCtx((scope) => [
                        Vue.renderSlot(_ctx.$slots, name, Vue.normalizeProps(Vue.guardReactiveProps(scope)))
                      ])
                    };
                  })
                ]),
                1024
                /* DYNAMIC_SLOTS */
              )) : Vue.createCommentVNode("v-if", true)
            ]),
            "layout-top": Vue.withCtx(() => [
              Vue.renderSlot(_ctx.$slots, "layout-top"),
              showArticleBanner.value ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 0 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-article-banner-before"),
                  Vue.createVNode(
                    Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleBanner_lang.default),
                    null,
                    Vue.createSlots({
                      _: 2
                      /* DYNAMIC */
                    }, [
                      Vue.renderList(_ctx.$slots, (_, name) => {
                        return {
                          name,
                          fn: Vue.withCtx((scope) => [
                            Vue.renderSlot(_ctx.$slots, name, Vue.normalizeProps(Vue.guardReactiveProps(scope)))
                          ])
                        };
                      })
                    ]),
                    1024
                    /* DYNAMIC_SLOTS */
                  ),
                  Vue.renderSlot(_ctx.$slots, "teek-article-banner-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true)
            ]),
            "layout-bottom": Vue.withCtx(() => [
              Vue.unref(isHomePage) ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 0 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-footer-group-before"),
                  Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_FooterGroup_lang.default)),
                  Vue.renderSlot(_ctx.$slots, "teek-footer-group-after"),
                  Vue.renderSlot(_ctx.$slots, "teek-footer-info-before"),
                  Vue.renderSlot(_ctx.$slots, "teek-footer-info", {}, () => [
                    Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_FooterInfo_lang.default))
                  ]),
                  Vue.renderSlot(_ctx.$slots, "teek-footer-info-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true),
              Vue.renderSlot(_ctx.$slots, "layout-bottom")
            ]),
            "sidebar-nav-before": Vue.withCtx(() => [
              Vue.createVNode(Vue.unref(HomeMyCardScreen_vue_vue_type_script_setup_true_name_HomeMyCardScreen_lang.default))
            ]),
            "doc-before": Vue.withCtx(() => [
              Vue.renderSlot(_ctx.$slots, "doc-before"),
              Vue.unref(frontmatter).article !== false && !showArticleBanner.value ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 0 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-article-analyze-before"),
                  Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleAnalyze_lang.default)),
                  Vue.renderSlot(_ctx.$slots, "teek-article-analyze-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true),
              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleImagePreview_lang.default)),
              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticlePageStyle_lang.default)),
              Vue.unref(teekConfig).codeBlock.enabled ?? true ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_CodeBlockToggle_lang.default), { key: 1 })) : Vue.createCommentVNode("v-if", true),
              topTipConfig.value ? (Vue.openBlock(), Vue.createBlock(
                Vue.unref(index_vue_vue_type_script_setup_true_name_VpContainer_lang.default),
                Vue.normalizeProps(Vue.mergeProps({ key: 2 }, Vue.unref(index$1.isBoolean)(topTipConfig.value) ? {} : topTipConfig.value)),
                null,
                16
                /* FULL_PROPS */
              )) : Vue.createCommentVNode("v-if", true),
              Vue.unref(teekConfig).sidebarTrigger ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_SidebarTrigger_lang.default), { key: 3 }, {
                default: Vue.withCtx((scope) => [
                  Vue.renderSlot(_ctx.$slots, "teek-sidebar-trigger", Vue.normalizeProps(Vue.guardReactiveProps(scope)))
                ]),
                _: 3
                /* FORWARDED */
              })) : Vue.createCommentVNode("v-if", true)
            ]),
            "doc-footer-before": Vue.withCtx(() => [
              Vue.renderSlot(_ctx.$slots, "doc-footer-before"),
              bottomTipConfig.value ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 0 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-article-bottom-tip-before"),
                  Vue.createVNode(
                    Vue.unref(index_vue_vue_type_script_setup_true_name_VpContainer_lang.default),
                    Vue.normalizeProps(Vue.guardReactiveProps(Vue.unref(index$1.isBoolean)(bottomTipConfig.value) ? {} : bottomTipConfig.value)),
                    null,
                    16
                    /* FULL_PROPS */
                  ),
                  Vue.renderSlot(_ctx.$slots, "teek-article-bottom-tip-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true)
            ]),
            "doc-after": Vue.withCtx(() => [
              Vue.renderSlot(_ctx.$slots, "doc-after"),
              (Vue.unref(teekConfig).articleUpdate.enabled ?? true) && Vue.unref(frontmatter).articleUpdate !== false ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 0 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-doc-update-before"),
                  Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleUpdate_lang.default)),
                  Vue.renderSlot(_ctx.$slots, "teek-doc-update-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true),
              ["doc-after", "doc-after-popper"].includes(Vue.unref(teekConfig).appreciation.position) ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 1 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-doc-after-appreciation-before"),
                  Vue.unref(teekConfig).appreciation.position === "doc-after" ? (Vue.openBlock(), Vue.createBlock(Vue.unref(DocAfterAppreciation_vue_vue_type_script_setup_true_name_DocAfterAppreciation_lang.default), { key: 0 })) : Vue.unref(teekConfig).appreciation.position === "doc-after-popper" ? (Vue.openBlock(), Vue.createBlock(Vue.unref(DocAfterAppreciationPopper_vue_vue_type_script_setup_true_name_DocAfterAppreciationPopper_lang.default), { key: 1 })) : Vue.createCommentVNode("v-if", true),
                  Vue.renderSlot(_ctx.$slots, "teek-doc-after-appreciation-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true),
              Vue.createCommentVNode(" \u8BC4\u8BBA\u533A "),
              commentConfig.value.enabled && commentConfig.value.provider ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 2 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-comment-before"),
                  commentConfig.value.provider === "render" ? Vue.renderSlot(_ctx.$slots, "teek-comment", { key: 0 }) : (Vue.openBlock(), Vue.createBlock(Vue.resolveDynamicComponent(commentConfig.value.components?.[commentConfig.value.provider]), {
                    key: 1,
                    id: `${Vue.unref(ns).namespace}-comment`,
                    class: Vue.normalizeClass(Vue.unref(ns).e("comment"))
                  }, null, 8, ["id", "class"])),
                  Vue.renderSlot(_ctx.$slots, "teek-comment-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true)
            ]),
            "aside-bottom": Vue.withCtx(() => [
              Vue.renderSlot(_ctx.$slots, "aside-bottom"),
              Vue.unref(teekConfig).appreciation.position === "aside-bottom" ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 0 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-aside-bottom-appreciation-before"),
                  Vue.createVNode(Vue.unref(AsideBottomAppreciation_vue_vue_type_script_setup_true_name_AsideBottomAppreciation_lang.default)),
                  Vue.renderSlot(_ctx.$slots, "teek-aside-bottom-appreciation-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true)
            ]),
            "page-top": Vue.withCtx(() => [
              Vue.renderSlot(_ctx.$slots, "page-top"),
              Vue.unref(isArchivesPage) || Vue.unref(isCataloguePage) || Vue.unref(isArticleOverviewPage) ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 0 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-page-top-before"),
                  Vue.unref(isArchivesPage) ? (Vue.openBlock(), Vue.createBlock(
                    Vue.unref(index_vue_vue_type_script_setup_true_name_ArchivesPage_lang.default),
                    { key: 0 },
                    Vue.createSlots({
                      _: 2
                      /* DYNAMIC */
                    }, [
                      Vue.renderList(_ctx.$slots, (_, name) => {
                        return {
                          name,
                          fn: Vue.withCtx((scope) => [
                            Vue.renderSlot(_ctx.$slots, name, Vue.normalizeProps(Vue.guardReactiveProps(scope)))
                          ])
                        };
                      })
                    ]),
                    1024
                    /* DYNAMIC_SLOTS */
                  )) : Vue.unref(isCataloguePage) ? (Vue.openBlock(), Vue.createBlock(
                    Vue.unref(index_vue_vue_type_script_setup_true_name_CataloguePage_lang.default),
                    { key: 1 },
                    Vue.createSlots({
                      _: 2
                      /* DYNAMIC */
                    }, [
                      Vue.renderList(_ctx.$slots, (_, name) => {
                        return {
                          name,
                          fn: Vue.withCtx((scope) => [
                            Vue.renderSlot(_ctx.$slots, name, Vue.normalizeProps(Vue.guardReactiveProps(scope)))
                          ])
                        };
                      })
                    ]),
                    1024
                    /* DYNAMIC_SLOTS */
                  )) : Vue.unref(isArticleOverviewPage) ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleOverviewPage_lang.default), { key: 2 })) : Vue.createCommentVNode("v-if", true),
                  Vue.renderSlot(_ctx.$slots, "teek-page-top-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true)
            ]),
            "aside-outline-before": Vue.withCtx(() => [
              Vue.unref(teekConfig).articleShare.enabled ? (Vue.openBlock(), Vue.createElementBlock(
                Vue.Fragment,
                { key: 0 },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-article-share-before"),
                  Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleShare_lang.default)),
                  Vue.renderSlot(_ctx.$slots, "teek-article-share-after")
                ],
                64
                /* STABLE_FRAGMENT */
              )) : Vue.createCommentVNode("v-if", true),
              Vue.renderSlot(_ctx.$slots, "aside-outline-before")
            ]),
            _: 2
            /* DYNAMIC */
          }, [
            Vue.renderList(Object.keys(_ctx.$slots).filter((name) => !usedSlots.includes(name)), (name) => {
              return {
                name,
                fn: Vue.withCtx((slotData) => [
                  Vue.renderSlot(_ctx.$slots, name, Vue.normalizeProps(Vue.guardReactiveProps(slotData)))
                ])
              };
            })
          ]), 1032, ["class"]), [
            [Vue.vShow, !loading.value]
          ])
        ],
        64
        /* STABLE_FRAGMENT */
      )) : (Vue.openBlock(), Vue.createBlock(
        Vue.unref(Layout),
        { key: 1 },
        Vue.createSlots({
          _: 2
          /* DYNAMIC */
        }, [
          Vue.renderList(_ctx.$slots, (_, name) => {
            return {
              name,
              fn: Vue.withCtx((slotData) => [
                Vue.renderSlot(_ctx.$slots, name, Vue.normalizeProps(Vue.guardReactiveProps(slotData)))
              ])
            };
          })
        ]),
        1024
        /* DYNAMIC_SLOTS */
      ));
    };
  }
});

exports.default = _sfc_main;
