// vitepress-theme-teek@1.5.1/es/config/index.mjs
import todoPlugin from '../markdown/plugins/todo.mjs';
import containerPlugin from '../markdown/plugins/container.mjs';
import shareCardPlugin from '../markdown/plugins/shareCard.mjs';
import imgCardPlugin from '../markdown/plugins/imgCard.mjs';
import navCardPlugin from '../markdown/plugins/navCard.mjs';
import demoPlugin from '../markdown/plugins/demo.mjs';
import videoPlugin from '../markdown/plugins/video.mjs';
import { createContainersThenUse } from '../markdown/helper/simpleContainer.mjs';
import 'markdown-it-container';
import 'js-yaml';
import { registerPluginAndGet } from './vitePlugins.mjs';
export { createRewrites } from 'vitepress-plugin-permalink';
export { LayoutMode, SpotlightStyle, ThemeColorName } from '../components/theme/ThemeEnhance/src/themeEnhance.mjs';

const defineTeekConfig = (config = {}) => {
  const { vitePlugins, markdown = {}, ...teekConfig } = config;
  const plugins = registerPluginAndGet(vitePlugins, teekConfig.teekTheme);
  const head = [];
  if (teekConfig.docAnalysis?.statistics?.provider === "busuanzi") {
    head.push(["meta", { name: "referrer", content: "no-referrer-when-downgrade" }]);
  }
  return {
    // 使用永久链接插件需要忽略死链提醒
    ignoreDeadLinks: true,
    metaChunk: true,
    head,
    vite: {
      plugins,
      ssr: { noExternal: ["vitepress-theme-teek"] },
      // 解决项目启动后终端打印 Scss 的废弃警告：The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.
      css: { preprocessorOptions: { scss: { api: "modern" } } }
    },
    markdown: {
      config: (md) => {
        [todoPlugin, shareCardPlugin, imgCardPlugin, navCardPlugin, videoPlugin].forEach((plugin) => md.use(plugin));
        const { container = {}, demo, config: config2 } = markdown;
        if (!demo?.disabled) md.use(demoPlugin, demo).use(containerPlugin, container.label);
        createContainersThenUse(md, container.config?.() || []);
        config2?.(md);
      }
    },
    themeConfig: teekConfig
  };
};

export { defineTeekConfig };
