// vitepress-theme-teek@1.5.1/lib/components/theme/ArticleUpdate/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var editPen = require('../../../../static/icons/editPen.js');
var index = require('../../ConfigProvider/index.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../../common/Icon/src/index.vue2.js');
require('@iconify/vue');
var index_vue_vue_type_script_setup_true_name_ArticleTitle_lang = require('../../ArticleTitle/src/index.vue2.js');

const _hoisted_1 = ["href", "aria-label"];
const _hoisted_2 = { key: 1 };
const _hoisted_3 = ["href", "aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "ArticleUpdate" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace.useNamespace("article-update");
    const { t } = useLocale.useLocale();
    const posts = index.usePosts();
    const route = vitepress.useRoute();
    const { frontmatter } = vitepress.useData();
    const { getTeekConfigRef } = index.useTeekConfig();
    const { archivesPath } = index.usePagePath();
    const articleConfig = getTeekConfigRef("articleUpdate", {
      limit: 3
    });
    const updatePosts = Vue.computed(() => {
      const path = "/" + route.data.filePath.replace(".md", "");
      return [
        ...posts.value.sortPostsByDate.filter((item) => ![route.path, path, `${path}.html`].includes(item.url)).slice(0, articleConfig.value.limit),
        { title: "\u66F4\u591A\u6587\u7AE0 >", url: archivesPath.value, frontmatter: {}, date: "" }
      ];
    });
    return (_ctx, _cache) => {
      return Vue.withDirectives((Vue.openBlock(), Vue.createElementBlock(
        "div",
        {
          class: Vue.normalizeClass(Vue.unref(ns).b())
        },
        [
          Vue.createElementVNode(
            "div",
            {
              class: Vue.normalizeClass([Vue.unref(ns).e("title"), "flx-align-center"])
            },
            [
              Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                icon: Vue.unref(editPen.default),
                class: "edit-icon",
                "aria-hidden": "true"
              }, null, 8, ["icon"]),
              Vue.unref(archivesPath) ? (Vue.openBlock(), Vue.createElementBlock("a", {
                key: 0,
                href: Vue.unref(vitepress.withBase)(Vue.unref(archivesPath)),
                class: "hover-color",
                "aria-label": Vue.unref(t)("tk.articleUpdate.label")
              }, Vue.toDisplayString(Vue.unref(t)("tk.articleUpdate.label")), 9, _hoisted_1)) : (Vue.openBlock(), Vue.createElementBlock(
                "span",
                _hoisted_2,
                Vue.toDisplayString(Vue.unref(t)("tk.articleUpdate.label")),
                1
                /* TEXT */
              ))
            ],
            2
            /* CLASS */
          ),
          Vue.createElementVNode("ul", null, [
            (Vue.openBlock(true), Vue.createElementBlock(
              Vue.Fragment,
              null,
              Vue.renderList(updatePosts.value, (item, index) => {
                return Vue.openBlock(), Vue.createElementBlock("li", {
                  key: item.url,
                  class: "flx-center"
                }, [
                  Vue.createElementVNode(
                    "span",
                    {
                      class: Vue.normalizeClass(Vue.unref(ns).m("num")),
                      "aria-hidden": "true"
                    },
                    Vue.toDisplayString(index !== updatePosts.value.length - 1 ? (index + 1).toString().padStart(2, "0") : ""),
                    3
                    /* TEXT, CLASS */
                  ),
                  Vue.createElementVNode(
                    "div",
                    {
                      class: Vue.normalizeClass(Vue.unref(ns).e("content"))
                    },
                    [
                      item.url ? (Vue.openBlock(), Vue.createElementBlock("a", {
                        key: 0,
                        href: Vue.unref(vitepress.withBase)(item.url),
                        class: "flx-1 hover-color sle",
                        "aria-label": item.title
                      }, [
                        Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleTitle_lang.default), {
                          post: item,
                          "title-tag-props": { position: "right", size: "small" }
                        }, null, 8, ["post"])
                      ], 8, _hoisted_3)) : Vue.createCommentVNode("v-if", true),
                      item.date ? (Vue.openBlock(), Vue.createElementBlock(
                        "span",
                        {
                          key: 1,
                          class: Vue.normalizeClass(Vue.unref(ns).em("content", "date"))
                        },
                        Vue.toDisplayString(item.date),
                        3
                        /* TEXT, CLASS */
                      )) : Vue.createCommentVNode("v-if", true)
                    ],
                    2
                    /* CLASS */
                  )
                ]);
              }),
              128
              /* KEYED_FRAGMENT */
            ))
          ])
        ],
        2
        /* CLASS */
      )), [
        [Vue.vShow, Vue.unref(frontmatter).article !== false]
      ]);
    };
  }
});

exports.default = _sfc_main;
