// vitepress-theme-teek@1.5.1/es/components/theme/ArticleAppreciation/src/DocAfterAppreciation.vue2.mjs
import { defineComponent, computed, ref, watch, createElementBlock, openBlock, unref, normalizeClass, createCommentVNode, createVNode, createBlock, Transition, withCtx, createElementVNode } from 'vue';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../composables/useLocale.mjs';
import '../../../../composables/useZIndex.mjs';
import aliPayIcon from '../../../../static/icons/aliPay.mjs';
import weChatPayIcon from '../../../../static/icons/weChatPay.mjs';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';
import _sfc_main$1 from '../../../common/Icon/src/index.vue2.mjs';
import '@iconify/vue';

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["innerHTML", "aria-expanded", "aria-controls"];
const _hoisted_3 = ["aria-expanded", "aria-controls"];
const _hoisted_4 = ["innerHTML"];
const _hoisted_5 = ["innerHTML"];
const _hoisted_6 = ["aria-label"];
const _hoisted_7 = ["innerHTML"];
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "DocAfterAppreciation" },
  __name: "DocAfterAppreciation",
  setup(__props) {
    const ns = useNamespace("article-appreciation");
    const { t } = useLocale();
    const { getTeekConfigRef } = useTeekConfig();
    const appreciateConfig = getTeekConfigRef("appreciation", { position: "" });
    const docAfterOptions = computed(() => appreciateConfig.value.options || { expand: false });
    const showContent = ref(docAfterOptions.value.expand);
    const icon = computed(() => {
      const { icon: icon2 } = docAfterOptions.value;
      if (icon2 === "aliPay") return aliPayIcon;
      if (icon2 === "weChatPay") return weChatPayIcon;
      return icon2;
    });
    const toggleShowContent = () => {
      showContent.value = !showContent.value;
    };
    watch(
      () => docAfterOptions.value.expand,
      (newValue) => {
        showContent.value = newValue;
      }
    );
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass([unref(ns).b(), unref(ns).m("doc-after")]),
        "aria-label": unref(t)("tk.articleAppreciation.label")
      }, [
        docAfterOptions.value.buttonHtml ? (openBlock(), createElementBlock("div", {
          key: 0,
          innerHTML: docAfterOptions.value.buttonHtml,
          role: "button",
          "aria-expanded": showContent.value,
          "aria-controls": `${unref(ns).e("content")}`
        }, null, 8, _hoisted_2)) : docAfterOptions.value.expandTitle || docAfterOptions.value.collapseTitle ? (openBlock(), createElementBlock("button", {
          key: 1,
          class: normalizeClass(unref(ns).e("button")),
          onClick: toggleShowContent,
          "aria-expanded": showContent.value,
          "aria-controls": `${unref(ns).e("content")}`,
          "aria-live": "polite"
        }, [
          icon.value ? (openBlock(), createBlock(unref(_sfc_main$1), {
            key: 0,
            class: normalizeClass(unref(ns).e("button__icon")),
            icon: icon.value,
            size: 16,
            "aria-hidden": "true"
          }, null, 8, ["class", "icon"])) : createCommentVNode("v-if", true),
          showContent.value ? (openBlock(), createElementBlock("span", {
            key: 1,
            innerHTML: docAfterOptions.value.collapseTitle
          }, null, 8, _hoisted_4)) : (openBlock(), createElementBlock("span", {
            key: 2,
            innerHTML: docAfterOptions.value.expandTitle
          }, null, 8, _hoisted_5))
        ], 10, _hoisted_3)) : createCommentVNode("v-if", true),
        createVNode(Transition, {
          name: unref(ns).join("fade")
        }, {
          default: withCtx(() => [
            showContent.value ? (openBlock(), createElementBlock("div", {
              key: 0,
              class: normalizeClass(unref(ns).e("content")),
              "aria-label": unref(t)("tk.articleAppreciation.contentLabel")
            }, [
              createElementVNode("div", {
                class: normalizeClass(unref(ns).e("content")),
                innerHTML: docAfterOptions.value.content
              }, null, 10, _hoisted_7)
            ], 10, _hoisted_6)) : createCommentVNode("v-if", true)
          ]),
          _: 1
          /* STABLE */
        }, 8, ["name"])
      ], 10, _hoisted_1);
    };
  }
});

export { _sfc_main as default };
