// vitepress-theme-teek@1.5.1/lib/components/theme/ThemeEnhance/src/components/SpotlightHover.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var index = require('../../../../../helper/dist/index.js');
var useEventListener = require('../../../../../composables/useEventListener.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useStorage = require('../../../../../composables/useStorage.js');
require('../../../../../composables/useZIndex.js');
var themeEnhance = require('../themeEnhance.js');
var namespace = require('../namespace.js');

var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "SpotlightHover" },
  __name: "SpotlightHover",
  props: {
    enabled: { type: Boolean }
  },
  setup(__props) {
    const props = __props;
    const shouldRecalculate = Vue.ref(false);
    const boxStyles = Vue.ref({ display: "none" });
    const vpDocElement = Vue.ref();
    const highlightedElement = Vue.ref();
    const spotlightStyle = useStorage.useStorage(namespace.spotlightStyleStorageKey, themeEnhance.SpotlightStyle.Aside);
    const mousePosition = Vue.ref({ x: 0, y: 0 });
    const computeBoxStyles = (bounding) => {
      return {
        display: "block",
        width: `${bounding.width + 8}px`,
        height: `${bounding.height + 8}px`,
        left: `${bounding.left - 4}px`,
        top: `${bounding.top - 4}px`,
        transition: "all 0.2s ease",
        borderRadius: "8px"
      };
    };
    const findChildElementUnderVPDocElement = (element) => {
      if (element === null) return null;
      if (element.parentElement === document.querySelector(".VPDoc main .vp-doc > div")) return element;
      else return findChildElementUnderVPDocElement(element.parentElement);
    };
    const watchHandler = () => {
      if (!index.isClient) return;
      const element = document.elementFromPoint(mousePosition.value.x, mousePosition.value.y);
      if (!(element && vpDocElement.value?.contains(element))) return;
      const el = findChildElementUnderVPDocElement(element);
      highlightedElement.value = el || void 0;
      if (highlightedElement.value && highlightedElement.value.tagName === "P") {
        const val = highlightedElement.value;
        const style = window.getComputedStyle(val);
        const lineHeight = Number.parseFloat(style.lineHeight);
        const lines = Math.floor(val.offsetHeight / lineHeight);
        const rect = val.getBoundingClientRect();
        const relativeY = mousePosition.value.y - rect.top;
        for (let i = 0; i < lines; i++) {
          const top = i * lineHeight;
          const height = lineHeight;
          const left = val.offsetLeft;
          const width = val.offsetWidth;
          if (relativeY >= top && relativeY < top + height) {
            boxStyles.value = computeBoxStyles({
              top: top + rect.top,
              left: left + rect.left,
              width,
              height
            });
            break;
          }
        }
      } else {
        if (highlightedElement.value) {
          const rect = highlightedElement.value.getBoundingClientRect();
          boxStyles.value = computeBoxStyles({
            top: rect.top,
            left: rect.left,
            width: rect.width,
            height: rect.height
          });
        }
      }
    };
    useEventListener.useEventListener(
      () => document,
      "mousemove",
      (event) => {
        mousePosition.value = { x: event.clientX, y: event.clientY };
      }
    );
    useEventListener.useEventListener(() => document, "scroll", watchHandler, true);
    Vue.onMounted(() => {
      vpDocElement.value = document.querySelector(".VPDoc main .vp-doc");
    });
    const route = vitepress.useRoute();
    Vue.watch(
      route,
      () => {
        vpDocElement.value = document.querySelector(".VPDoc main .vp-doc");
        shouldRecalculate.value = true;
        boxStyles.value = { display: "none" };
        watchHandler();
        shouldRecalculate.value = false;
      },
      { flush: "post" }
    );
    Vue.watch([() => mousePosition.value.x, () => mousePosition.value.y], () => {
      if (props.enabled) watchHandler();
    });
    Vue.watch(
      () => props.enabled,
      (val) => {
        if (!val) boxStyles.value = { display: "none" };
      }
    );
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createBlock(Vue.Teleport, { to: "body" }, [
        props.enabled && !shouldRecalculate.value ? (Vue.openBlock(), Vue.createElementBlock(
          "div",
          {
            key: 0,
            style: Vue.normalizeStyle(boxStyles.value),
            class: Vue.normalizeClass([
              Vue.unref(namespace.ns).join("spotlight-hover"),
              Vue.unref(spotlightStyle) === Vue.unref(themeEnhance.SpotlightStyle).Under ? Vue.unref(namespace.ns).join("spotlight-hover__under") : "",
              Vue.unref(spotlightStyle) === Vue.unref(themeEnhance.SpotlightStyle).Aside ? Vue.unref(namespace.ns).join("spotlight-hover__aside") : ""
            ]),
            "aria-hidden": "true",
            focusable: "false"
          },
          null,
          6
          /* CLASS, STYLE */
        )) : Vue.createCommentVNode("v-if", true)
      ]);
    };
  }
});

exports.default = _sfc_main;
