// vitepress-theme-teek/lib/components/theme/ArticleBanner/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
require('../../../../composables/useZIndex.js');
var index = require('../../ConfigProvider/index.js');
var index_vue_vue_type_script_setup_true_name_ArticleBreadcrumb_lang = require('../../ArticleBreadcrumb/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleAnalyze_lang = require('../../ArticleAnalyze/src/index.vue2.js');
var HomeBannerWaves_vue_vue_type_script_setup_true_name_HomeBannerWaves_lang = require('../../HomeBanner/src/HomeBannerWaves.vue2.js');

const _hoisted_1 = ["src"];
const _hoisted_2 = {
  key: 0,
  class: "categories flx-center"
};
const _hoisted_3 = ["href", "title"];
const _hoisted_4 = {
  key: 1,
  class: "tags flx-center"
};
const _hoisted_5 = ["href", "title"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  __name: "index",
  setup(__props) {
    const loaded = Vue.ref(false);
    const ns = useNamespace.useNamespace("article-banner");
    const { frontmatter } = vitepress.useData();
    const route = vitepress.useRoute();
    const { hasSidebar } = index.useSidebar();
    const { isMobile } = index.useCommon();
    const { getTeekConfigRef } = index.useTeekConfig();
    const articleBannerConfig = getTeekConfigRef("articleBanner", {
      enabled: false,
      showCategory: true,
      showTag: true,
      defaultCoverImg: "",
      defaultCoverBgColor: ""
    });
    const postConfig = getTeekConfigRef("post", {
      defaultCoverImg: []
    });
    const imgSrc = Vue.computed(() => {
      const imgSrcList = [
        frontmatter.value.coverImg || articleBannerConfig.value.defaultCoverImg || postConfig.value.defaultCoverImg || []
      ].flat();
      if (!imgSrcList.length) return "";
      return imgSrcList[Math.floor(Math.random() * imgSrcList.length)];
    });
    const style = Vue.computed(() => {
      return {
        [ns.cssVarName("article-banner-bg-color")]: frontmatter.value.coverBgColor || articleBannerConfig.value.defaultCoverBgColor
      };
    });
    Vue.watch(
      () => route.path,
      async () => {
        await Vue.nextTick();
        loaded.value = true;
        setTimeout(() => loaded.value = false, 1);
      }
    );
    return (_ctx, _cache) => {
      return !Vue.unref(hasSidebar) && Vue.unref(articleBannerConfig).enabled ? Vue.withDirectives((Vue.openBlock(), Vue.createElementBlock(
        "div",
        {
          key: 0,
          class: Vue.normalizeClass(Vue.unref(ns).b()),
          style: Vue.normalizeStyle(style.value)
        },
        [
          Vue.createElementVNode(
            "div",
            {
              class: Vue.normalizeClass([Vue.unref(ns).e("wrapper"), "flx-justify-center"])
            },
            [
              imgSrc.value ? (Vue.openBlock(), Vue.createElementBlock(
                "div",
                {
                  key: 0,
                  class: Vue.normalizeClass(Vue.unref(ns).e("cover"))
                },
                [
                  Vue.createElementVNode("img", {
                    src: imgSrc.value,
                    class: "no-preview",
                    alt: "cover"
                  }, null, 8, _hoisted_1)
                ],
                2
                /* CLASS */
              )) : Vue.createCommentVNode("v-if", true),
              Vue.createElementVNode(
                "div",
                {
                  class: Vue.normalizeClass(Vue.unref(ns).e("info"))
                },
                [
                  Vue.renderSlot(_ctx.$slots, "teek-article-banner-info-top"),
                  !Vue.unref(isMobile) ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleBreadcrumb_lang.default), { key: 0 })) : Vue.createCommentVNode("v-if", true),
                  Vue.createElementVNode(
                    "div",
                    {
                      class: Vue.normalizeClass([Vue.unref(ns).e("meta"), "flx-center flx-wrap"])
                    },
                    [
                      Vue.unref(frontmatter).categories && Vue.unref(articleBannerConfig).showCategory ? (Vue.openBlock(), Vue.createElementBlock("div", _hoisted_2, [
                        (Vue.openBlock(true), Vue.createElementBlock(
                          Vue.Fragment,
                          null,
                          Vue.renderList(Vue.unref(frontmatter).categories, (category) => {
                            return Vue.openBlock(), Vue.createElementBlock("a", {
                              key: category,
                              href: Vue.unref(vitepress.withBase)(`/categories?category=${category}`),
                              class: "meta-info category",
                              title: category
                            }, Vue.toDisplayString(category), 9, _hoisted_3);
                          }),
                          128
                          /* KEYED_FRAGMENT */
                        ))
                      ])) : Vue.createCommentVNode("v-if", true),
                      Vue.unref(frontmatter).tags && Vue.unref(articleBannerConfig).showTag ? (Vue.openBlock(), Vue.createElementBlock("div", _hoisted_4, [
                        (Vue.openBlock(true), Vue.createElementBlock(
                          Vue.Fragment,
                          null,
                          Vue.renderList(Vue.unref(frontmatter).tags, (tag) => {
                            return Vue.openBlock(), Vue.createElementBlock("a", {
                              key: tag,
                              href: Vue.unref(vitepress.withBase)(`/tags?tag=${tag}`),
                              class: "meta-info tag",
                              title: tag
                            }, [
                              _cache[0] || (_cache[0] = Vue.createElementVNode(
                                "span",
                                null,
                                "#",
                                -1
                                /* CACHED */
                              )),
                              Vue.createElementVNode(
                                "span",
                                null,
                                Vue.toDisplayString(tag),
                                1
                                /* TEXT */
                              )
                            ], 8, _hoisted_5);
                          }),
                          128
                          /* KEYED_FRAGMENT */
                        ))
                      ])) : Vue.createCommentVNode("v-if", true)
                    ],
                    2
                    /* CLASS */
                  ),
                  Vue.createElementVNode(
                    "h1",
                    null,
                    Vue.toDisplayString(Vue.unref(frontmatter).title),
                    1
                    /* TEXT */
                  ),
                  Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleAnalyze_lang.default), {
                    breadcrumb: false,
                    scope: "article-banner"
                  }),
                  Vue.renderSlot(_ctx.$slots, "teek-article-banner-info-bottom")
                ],
                2
                /* CLASS */
              )
            ],
            2
            /* CLASS */
          ),
          Vue.createVNode(Vue.unref(HomeBannerWaves_vue_vue_type_script_setup_true_name_HomeBannerWaves_lang.default))
        ],
        6
        /* CLASS, STYLE */
      )), [
        [Vue.vShow, !loaded.value]
      ]) : Vue.createCommentVNode("v-if", true);
    };
  }
});

exports.default = _sfc_main;
