// vitepress-theme-teek@1.5.1/lib/components/common/ArticlePage/src/components/DocAsideOutline.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var useNamespace = require('../../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
require('../../../../../composables/useZIndex.js');
var outline = require('./outline.js');
var DocAsideOutlineItem_vue_vue_type_script_setup_true_name_DocAsideOutlineItem_lang = require('./DocAsideOutlineItem.vue2.js');

var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "DocAsideOutline" },
  __name: "DocAsideOutline",
  setup(__props) {
    const ns = useNamespace.useNamespace("aside-outline");
    const { frontmatter, theme } = vitepress.useData();
    const headers = Vue.shallowRef([]);
    vitepress.onContentUpdated(() => {
      headers.value = outline.getHeaders(frontmatter.value.outline ?? theme.value.outline);
    });
    Vue.onMounted(() => {
      if (!headers.value?.length) headers.value = outline.getHeaders(frontmatter.value.outline ?? theme.value.outline);
    });
    const container = Vue.ref();
    const marker = Vue.ref();
    outline.useActiveAnchor(container, marker);
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock(
        "nav",
        {
          "aria-labelledby": "doc-outline-aria-label",
          class: Vue.normalizeClass([Vue.unref(ns).b(), { "has-outline": headers.value.length > 0 }]),
          ref_key: "container",
          ref: container
        },
        [
          Vue.createElementVNode(
            "div",
            {
              class: Vue.normalizeClass(Vue.unref(ns).e("content"))
            },
            [
              Vue.createElementVNode(
                "div",
                {
                  class: Vue.normalizeClass(Vue.unref(ns).m("marker")),
                  ref_key: "marker",
                  ref: marker
                },
                null,
                2
                /* CLASS */
              ),
              Vue.createElementVNode(
                "div",
                {
                  id: "doc-outline-aria-label",
                  "aria-level": "2",
                  class: Vue.normalizeClass(Vue.unref(ns).m("title")),
                  role: "heading"
                },
                Vue.toDisplayString(Vue.unref(outline.resolveTitle)(Vue.unref(theme))),
                3
                /* TEXT, CLASS */
              ),
              Vue.createVNode(DocAsideOutlineItem_vue_vue_type_script_setup_true_name_DocAsideOutlineItem_lang.default, {
                headers: headers.value,
                root: true
              }, null, 8, ["headers"])
            ],
            2
            /* CLASS */
          )
        ],
        2
        /* CLASS */
      );
    };
  }
});

exports.default = _sfc_main;
