// vitepress-theme-teek@1.5.1/lib/components/theme/HomePostList/src/HomePostItem.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var index = require('../../ConfigProvider/index.js');
var index_vue_vue_type_script_setup_true_name_ArticleInfo_lang = require('../../ArticleInfo/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleTitle_lang = require('../../ArticleTitle/src/index.vue2.js');

const _hoisted_1 = ["title", "aria-label"];
const _hoisted_2 = { class: "list flx" };
const _hoisted_3 = ["href", "aria-label"];
const _hoisted_4 = ["aria-label"];
const _hoisted_5 = ["innerHTML"];
const _hoisted_6 = ["href", "aria-label"];
const _hoisted_7 = ["aria-label"];
const _hoisted_8 = ["aria-label"];
const _hoisted_9 = ["innerHTML"];
const _hoisted_10 = ["href", "aria-label"];
const _hoisted_11 = ["href", "aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "HomePostItem" },
  __name: "HomePostItem",
  props: {
    post: {},
    coverImgMode: {}
  },
  setup(__props) {
    const ns = useNamespace.useNamespace("post-item");
    const { t } = useLocale.useLocale();
    const { getTeekConfigRef } = index.useTeekConfig();
    const postConfig = getTeekConfigRef("post", {
      excerptPosition: "bottom",
      showMore: true,
      moreLabel: t("tk.homePost.moreLabel"),
      showCapture: false,
      splitSeparator: false,
      listStyleTitleTagPosition: "right",
      defaultCoverImg: []
    });
    const articleConfig = getTeekConfigRef("articleAnalyze", {
      showInfo: true
    });
    const postUrl = __props.post.url && vitepress.withBase(__props.post.url);
    const excerpt = Vue.computed(
      () => __props.post.frontmatter.description || __props.post.excerpt || postConfig.value.showCapture && __props.post.capture
    );
    const imgSrcList = Vue.computed(() => [__props.post.frontmatter.coverImg || postConfig.value.defaultCoverImg || []].flat());
    const coverImgMap = Vue.computed(() => {
      const imgSrc = imgSrcList.value[Math.floor(Math.random() * imgSrcList.value.length)];
      return {
        default: {
          is: "div",
          props: {
            style: `background-image: url(${vitepress.withBase(imgSrc)});`
          }
        },
        full: {
          is: "img",
          props: {
            src: vitepress.withBase(imgSrc)
          }
        }
      };
    });
    const isShowInfo = Vue.computed(() => {
      const arr = [articleConfig.value.showInfo].flat();
      return arr.includes(true) || arr.includes("post");
    });
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock(
        "div",
        {
          class: Vue.normalizeClass(Vue.unref(ns).b())
        },
        [
          !!__props.post.frontmatter.sticky ? (Vue.openBlock(), Vue.createElementBlock("i", {
            key: 0,
            class: "pin",
            title: Vue.unref(t)("tk.homePost.pin", { sticky: __props.post.frontmatter.sticky }),
            "aria-label": Vue.unref(t)("tk.homePost.pinLabel")
          }, null, 8, _hoisted_1)) : Vue.createCommentVNode("v-if", true),
          Vue.createElementVNode("div", _hoisted_2, [
            Vue.createElementVNode(
              "div",
              {
                class: Vue.normalizeClass(Vue.unref(ns).e("left"))
              },
              [
                Vue.createCommentVNode(" \u6807\u9898 "),
                Vue.createElementVNode("a", {
                  class: Vue.normalizeClass([Vue.unref(ns).e("left__title"), "hover-color", "sle"]),
                  href: Vue.unref(postUrl),
                  "aria-label": __props.post.title
                }, [
                  Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleTitle_lang.default), {
                    post: __props.post,
                    "title-tag-props": { position: Vue.unref(postConfig).listStyleTitleTagPosition }
                  }, null, 8, ["post", "title-tag-props"])
                ], 10, _hoisted_3),
                Vue.createCommentVNode(" \u6458\u8981 top "),
                excerpt.value && Vue.unref(postConfig).excerptPosition === "top" ? (Vue.openBlock(), Vue.createElementBlock("div", {
                  key: 0,
                  class: Vue.normalizeClass(`${Vue.unref(ns).e("left__excerpt")} top`),
                  "aria-label": Vue.unref(t)("tk.homePost.excerptLabel")
                }, [
                  Vue.createElementVNode("div", {
                    class: "excerpt",
                    innerHTML: excerpt.value
                  }, null, 8, _hoisted_5),
                  Vue.unref(postConfig).showMore ? (Vue.openBlock(), Vue.createElementBlock("a", {
                    key: 0,
                    class: "more",
                    href: Vue.unref(postUrl),
                    "aria-label": Vue.unref(postConfig).moreLabel
                  }, Vue.toDisplayString(Vue.unref(postConfig).moreLabel), 9, _hoisted_6)) : Vue.createCommentVNode("v-if", true)
                ], 10, _hoisted_4)) : Vue.createCommentVNode("v-if", true),
                Vue.createCommentVNode(" \u6587\u7AE0\u4FE1\u606F "),
                Vue.createElementVNode("div", {
                  class: Vue.normalizeClass(Vue.unref(ns).e("left__footer")),
                  "aria-label": Vue.unref(t)("tk.homePost.infoLabel")
                }, [
                  isShowInfo.value ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_ArticleInfo_lang.default), {
                    key: 0,
                    post: __props.post,
                    scope: "post",
                    split: Vue.unref(postConfig).splitSeparator
                  }, null, 8, ["post", "split"])) : Vue.createCommentVNode("v-if", true)
                ], 10, _hoisted_7),
                Vue.createCommentVNode(" \u6458\u8981 bottom "),
                excerpt.value && Vue.unref(postConfig).excerptPosition === "bottom" ? (Vue.openBlock(), Vue.createElementBlock("div", {
                  key: 1,
                  class: Vue.normalizeClass(`${Vue.unref(ns).e("left__excerpt")} bottom`),
                  "aria-label": Vue.unref(t)("tk.homePost.excerptLabel")
                }, [
                  Vue.createElementVNode("div", {
                    class: "excerpt",
                    innerHTML: excerpt.value
                  }, null, 8, _hoisted_9),
                  Vue.unref(postConfig).showMore ? (Vue.openBlock(), Vue.createElementBlock("a", {
                    key: 0,
                    class: "more",
                    href: Vue.unref(postUrl),
                    "aria-label": Vue.unref(postConfig).moreLabel
                  }, Vue.toDisplayString(Vue.unref(postConfig).moreLabel), 9, _hoisted_10)) : Vue.createCommentVNode("v-if", true)
                ], 10, _hoisted_8)) : Vue.createCommentVNode("v-if", true)
              ],
              2
              /* CLASS */
            ),
            Vue.createCommentVNode(" \u53F3\u4FA7\u5C01\u9762\u56FE "),
            Vue.createElementVNode(
              "div",
              {
                class: Vue.normalizeClass(`${Vue.unref(ns).e("right")} flx-align-center`)
              },
              [
                imgSrcList.value.length ? (Vue.openBlock(), Vue.createElementBlock("a", {
                  key: 0,
                  href: Vue.unref(postUrl),
                  class: Vue.normalizeClass(`${__props.coverImgMode} cover`),
                  "aria-label": Vue.unref(t)("tk.homePost.coverImgLabel")
                }, [
                  (Vue.openBlock(), Vue.createBlock(
                    Vue.resolveDynamicComponent(coverImgMap.value[__props.coverImgMode].is),
                    Vue.mergeProps(coverImgMap.value[__props.coverImgMode].props, {
                      "aria-hidden": "true",
                      class: "cover-img"
                    }),
                    null,
                    16
                    /* FULL_PROPS */
                  ))
                ], 10, _hoisted_11)) : Vue.createCommentVNode("v-if", true)
              ],
              2
              /* CLASS */
            )
          ])
        ],
        2
        /* CLASS */
      );
    };
  }
});

exports.default = _sfc_main;
