// vitepress-theme-teek@1.5.1/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var DefaultTheme = require('vitepress/theme');
var index$5 = require('./helper/dist/index.js');
require('./static/iconfont/social/iconfont.js');
var aliPay = require('./static/icons/aliPay.js');
var arrowDown = require('./static/icons/arrowDown.js');
var category = require('./static/icons/category.js');
var close = require('./static/icons/close.js');
var copyright = require('./static/icons/copyright.js');
var docAnalysis = require('./static/icons/docAnalysis.js');
var empty = require('./static/icons/empty.js');
var friendLink = require('./static/icons/friendLink.js');
var icpRecord = require('./static/icons/icpRecord.js');
var notice = require('./static/icons/notice.js');
var playground = require('./static/icons/playground.js');
var rocket = require('./static/icons/rocket.js');
var size = require('./static/icons/size.js');
var tag = require('./static/icons/tag.js');
var theme = require('./static/icons/theme.js');
var topArticle = require('./static/icons/topArticle.js');
var reading = require('./static/icons/reading.js');
var clock = require('./static/icons/clock.js');
var view = require('./static/icons/view.js');
var house = require('./static/icons/house.js');
var arrowLeft = require('./static/icons/arrowLeft.js');
var arrowRight = require('./static/icons/arrowRight.js');
var magic = require('./static/icons/magic.js');
var comment = require('./static/icons/comment.js');
var user = require('./static/icons/user.js');
var calendar = require('./static/icons/calendar.js');
var folderOpened = require('./static/icons/folderOpened.js');
var collectionTag = require('./static/icons/collectionTag.js');
var editPen = require('./static/icons/editPen.js');
var github = require('./static/icons/github.js');
var copy = require('./static/icons/copy.js');
var code = require('./static/icons/code.js');
var caretTop = require('./static/icons/caretTop.js');
var weChatPay = require('./static/icons/weChatPay.js');
var share = require('./static/icons/share.js');
var thumbs = require('./static/icons/thumbs.js');
var externalLink = require('./static/icons/externalLink.js');
var dArrowRight = require('./static/icons/dArrowRight.js');
var dArrowLeft = require('./static/icons/dArrowLeft.js');
var moreFilled = require('./static/icons/moreFilled.js');
var fullscreen = require('./static/icons/fullscreen.js');
var refreshLeft = require('./static/icons/refreshLeft.js');
var refreshRight = require('./static/icons/refreshRight.js');
var scaleToOriginal = require('./static/icons/scaleToOriginal.js');
var zoomIn = require('./static/icons/zoomIn.js');
var zoomOut = require('./static/icons/zoomOut.js');
var circleCloseFilled = require('./static/icons/circleCloseFilled.js');
var infoFilled = require('./static/icons/infoFilled.js');
var successFilled = require('./static/icons/successFilled.js');
var warningFilled = require('./static/icons/warningFilled.js');
var questionFilled = require('./static/icons/questionFilled.js');
var top = require('./static/icons/top.js');
var lock = require('./static/icons/lock.js');
var autoWidth = require('./static/icons/theme-enhance/autoWidth.js');
var click = require('./static/icons/theme-enhance/click.js');
var fullScreenOne = require('./static/icons/theme-enhance/fullScreenOne.js');
var fullscreenTwo = require('./static/icons/theme-enhance/fullscreenTwo.js');
var layout = require('./static/icons/theme-enhance/layout.js');
var overallReduction = require('./static/icons/theme-enhance/overallReduction.js');
var alignLeft = require('./static/icons/theme-enhance/alignLeft.js');
var alignTextLeft = require('./static/icons/theme-enhance/alignTextLeft.js');
var scale = require('./static/icons/theme-enhance/scale.js');
var water = require('./static/icons/theme-enhance/water.js');
var onClickOutside = require('./composables/onClickOutside.js');
var useAnchorScroll = require('./composables/useAnchorScroll.js');
var useUvPv = require('./composables/useUvPv.js');
var useClipboard = require('./composables/useClipboard.js');
var useCopyBanner = require('./composables/useCopyBanner.js');
var useDebounce = require('./composables/useDebounce.js');
var useElementHover = require('./composables/useElementHover.js');
var useEventListener = require('./composables/useEventListener.js');
var useLocale = require('./composables/useLocale.js');
var useMediaQuery = require('./composables/useMediaQuery.js');
var useMounted = require('./composables/useMounted.js');
var useNamespace = require('./composables/useNamespace.js');
var usePopoverSize = require('./composables/usePopoverSize.js');
var useScopeDispose = require('./composables/useScopeDispose.js');
var useScrollData = require('./composables/useScrollData.js');
var useStorage = require('./composables/useStorage.js');
var useSwitchData = require('./composables/useSwitchData.js');
var useTextTypes = require('./composables/useTextTypes.js');
var useThemeColor = require('./composables/useThemeColor.js');
var useViewTransition = require('./composables/useViewTransition.js');
var useVpRouter = require('./composables/useVpRouter.js');
var useWindowSize = require('./composables/useWindowSize.js');
var useWindowTransition = require('./composables/useWindowTransition.js');
var useZIndex = require('./composables/useZIndex.js');
var zhCn = require('./locale/lang/zh-cn.js');
var en = require('./locale/lang/en.js');
var simpleContainer = require('./markdown/helper/simpleContainer.js');
var cardContainer = require('./markdown/helper/cardContainer.js');
var version = require('./version.js');
var index$3 = require('./components/theme/ConfigProvider/index.js');
var index_vue_vue_type_script_setup_true_name_CataloguePage_lang = require('./components/theme/CataloguePage/src/index.vue2.js');
var CatalogueItem_vue_vue_type_script_setup_true_name_CatalogueItem_lang = require('./components/theme/CataloguePage/src/CatalogueItem.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArchivesPage_lang = require('./components/theme/ArchivesPage/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleOverviewPage_lang = require('./components/theme/ArticleOverviewPage/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_LoginPage_lang = require('./components/theme/LoginPage/src/index.vue2.js');
var login = require('./components/theme/LoginPage/src/login.js');
var useWatchLogin = require('./components/theme/LoginPage/src/useWatchLogin.js');
var index_vue_vue_type_script_setup_true_name_RiskLinkPage_lang = require('./components/theme/RiskLinkPage/src/index.vue2.js');
var useRiskLink = require('./components/theme/RiskLinkPage/src/useRiskLink.js');
var index_vue_vue_type_script_setup_true_name_DemoCode_lang = require('./components/theme/DemoCode/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_TitleTag_lang = require('./components/common/TitleTag/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('./components/common/Icon/src/index.vue2.js');
var SocialIcons = require('./components/common/Icon/src/SocialIcons.js');
var vue = require('@iconify/vue');
var index_vue_vue_type_script_setup_true_name_TeekLayout_lang = require('./components/theme/Layout/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticlePage_lang = require('./components/common/ArticlePage/src/index.vue2.js');
var DocAsideOutline_vue_vue_type_script_setup_true_name_DocAsideOutline_lang = require('./components/common/ArticlePage/src/components/DocAsideOutline.vue2.js');
var index_vue_vue_type_script_setup_true_name_Avatar_lang = require('./components/common/Avatar/src/index.vue2.js');
var Breadcrumb_vue_vue_type_script_setup_true_name_Breadcrumb_lang = require('./components/common/Breadcrumb/src/Breadcrumb.vue2.js');
var BreadcrumbItem_vue_vue_type_script_setup_true_name_BreadcrumbItem_lang = require('./components/common/Breadcrumb/src/BreadcrumbItem.vue2.js');
var index_vue_vue_type_script_setup_true_name_ImageViewer_lang = require('./components/common/ImageViewer/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_InputSlide_lang = require('./components/common/InputSlide/src/index.vue2.js');
var method = require('./components/common/Message/src/method.js');
var index_vue_vue_type_script_setup_true_name_Pagination_lang = require('./components/common/Pagination/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_Popover_lang = require('./components/common/Popover/src/index.vue2.js');
var Segmented_vue_vue_type_script_setup_true_name_Segmented_lang = require('./components/common/Segmented/src/Segmented.vue2.js');
var SegmentedItem_vue_vue_type_script_setup_true_name_SegmentedItem_lang = require('./components/common/Segmented/src/SegmentedItem.vue2.js');
var index_vue_vue_type_script_setup_true_name_TransitionCollapse_lang = require('./components/common/TransitionCollapse/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_lang = require('./components/common/VerifyCode/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_VpContainer_lang = require('./components/common/VpContainer/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleAnalyze_lang = require('./components/theme/ArticleAnalyze/src/index.vue2.js');
var AsideBottomAppreciation_vue_vue_type_script_setup_true_name_AsideBottomAppreciation_lang = require('./components/theme/ArticleAppreciation/src/AsideBottomAppreciation.vue2.js');
var DocAfterAppreciation_vue_vue_type_script_setup_true_name_DocAfterAppreciation_lang = require('./components/theme/ArticleAppreciation/src/DocAfterAppreciation.vue2.js');
var DocAfterAppreciationPopper_vue_vue_type_script_setup_true_name_DocAfterAppreciationPopper_lang = require('./components/theme/ArticleAppreciation/src/DocAfterAppreciationPopper.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleBreadcrumb_lang = require('./components/theme/ArticleBreadcrumb/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleHeadingHighlight_lang = require('./components/theme/ArticleHeadingHighlight/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleImagePreview_lang = require('./components/theme/ArticleImagePreview/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleInfo_lang = require('./components/theme/ArticleInfo/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticlePageStyle_lang = require('./components/theme/ArticlePageStyle/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleShare_lang = require('./components/theme/ArticleShare/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleTitle_lang = require('./components/theme/ArticleTitle/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ArticleUpdate_lang = require('./components/theme/ArticleUpdate/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_BodyBgImage_lang = require('./components/theme/BodyBgImage/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_CodeBlockToggle_lang = require('./components/theme/CodeBlockToggle/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_CommentArtalk_lang = require('./components/theme/CommentArtalk/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_CommentGiscus_lang = require('./components/theme/CommentGiscus/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_CommentTwikoo_lang = require('./components/theme/CommentTwikoo/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_CommentWaline_lang = require('./components/theme/CommentWaline/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_FooterGroup_lang = require('./components/theme/FooterGroup/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_FooterInfo_lang = require('./components/theme/FooterInfo/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_Home_lang = require('./components/theme/Home/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_HomeBanner_lang = require('./components/theme/HomeBanner/src/index.vue2.js');
var HomeBannerBgImage_vue_vue_type_script_setup_true_name_HomeBannerBgImage_lang = require('./components/theme/HomeBanner/src/HomeBannerBgImage.vue2.js');
var HomeBannerBgPure_vue_vue_type_script_setup_true_name_HomeBannerBgPure_lang = require('./components/theme/HomeBanner/src/HomeBannerBgPure.vue2.js');
var HomeBannerContent_vue_vue_type_script_setup_true_name_HomeBannerContent_lang = require('./components/theme/HomeBanner/src/HomeBannerContent.vue2.js');
var HomeBannerFeature_vue_vue_type_script_setup_true_name_HomeBannerFeature_lang = require('./components/theme/HomeBanner/src/HomeBannerFeature.vue2.js');
var HomeBannerWaves_vue_vue_type_script_setup_true_name_HomeBannerWaves_lang = require('./components/theme/HomeBanner/src/HomeBannerWaves.vue2.js');
var index_vue_vue_type_script_setup_true_name_PageCard_lang = require('./components/common/PageCard/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_HomeCategoryCard_lang = require('./components/theme/HomeCategoryCard/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_HomeDocAnalysisCard_lang = require('./components/theme/HomeDocAnalysisCard/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_lang$1 = require('./components/theme/HomeFeature/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_HomeFriendLinkCard_lang = require('./components/theme/HomeFriendLinkCard/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_HomeFullscreenWallpaper_lang = require('./components/theme/HomeFullscreenWallpaper/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_HomeCardList_lang = require('./components/theme/HomeCardList/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_HomeMyCard_lang = require('./components/theme/HomeMyCard/src/index.vue2.js');
var HomeMyCardScreen_vue_vue_type_script_setup_true_name_HomeMyCardScreen_lang = require('./components/theme/HomeMyCard/src/HomeMyCardScreen.vue2.js');
var index_vue_vue_type_script_setup_true_name_HomePostList_lang = require('./components/theme/HomePostList/src/index.vue2.js');
var HomePostItem_vue_vue_type_script_setup_true_name_HomePostItem_lang = require('./components/theme/HomePostList/src/HomePostItem.vue2.js');
var index_vue_vue_type_script_setup_true_name_HomeTagCard_lang = require('./components/theme/HomeTagCard/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_HomeTopArticleCard_lang = require('./components/theme/HomeTopArticleCard/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_Notice_lang = require('./components/theme/Notice/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_RightBottomButton_lang = require('./components/theme/RightBottomButton/src/index.vue2.js');
var index$4 = require('./components/theme/RouteLoading/src/index.vue.js');
var index_vue_vue_type_script_setup_true_name_SidebarTrigger_lang = require('./components/theme/SidebarTrigger/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_ThemeEnhance_lang = require('./components/theme/ThemeEnhance/src/index.vue2.js');
var Switch_vue_vue_type_script_setup_true_name_Switch_lang = require('./components/theme/ThemeEnhance/src/components/Switch.vue2.js');
var BaseTemplate_vue_vue_type_script_setup_true_name_BaseTemplate_lang = require('./components/theme/ThemeEnhance/src/components/BaseTemplate.vue2.js');
var index$1 = require('./components/common/FocusTrap/src/index.vue.js');
var index$2 = require('./components/common/ImageViewer/index.js');
var instance = require('./components/common/Message/src/instance.js');
var compile = require('./components/theme/ArticleTitle/src/compile.js');
var artalk = require('./components/theme/CommentArtalk/src/artalk.js');
var giscus = require('./components/theme/CommentGiscus/src/giscus.js');
var twikoo = require('./components/theme/CommentTwikoo/src/twikoo.js');
var waline = require('./components/theme/CommentWaline/src/waline.js');
var home = require('./components/theme/Home/src/home.js');
var homePostList = require('./components/theme/HomePostList/src/homePostList.js');
var themeEnhance = require('./components/theme/ThemeEnhance/src/themeEnhance.js');
var namespace = require('./components/theme/ThemeEnhance/src/namespace.js');
var useThemeColorList = require('./components/theme/ThemeEnhance/src/useThemeColorList.js');

var index = {
  extends: DefaultTheme,
  Layout: index$3.TeekConfigProvider(index_vue_vue_type_script_setup_true_name_TeekLayout_lang.default),
  async enhanceApp({ app, siteData, router }) {
    app.component("TkCataloguePage", index_vue_vue_type_script_setup_true_name_CataloguePage_lang.default);
    app.component("TkArchivesPage", index_vue_vue_type_script_setup_true_name_ArchivesPage_lang.default);
    app.component("TkArticleOverviewPage", index_vue_vue_type_script_setup_true_name_ArticleOverviewPage_lang.default);
    app.component("TkLoginPage", index_vue_vue_type_script_setup_true_name_LoginPage_lang.default);
    app.component("TkRiskLinkPage", index_vue_vue_type_script_setup_true_name_RiskLinkPage_lang.default);
    app.component("TkDemoCode", index_vue_vue_type_script_setup_true_name_DemoCode_lang.default);
    app.component("TkTitleTag", index_vue_vue_type_script_setup_true_name_TitleTag_lang.default);
    app.component("TkIcon", index_vue_vue_type_script_setup_true_name_Icon_lang.default);
    if (!index$5.isClient) return;
    const { themeConfig } = siteData.value;
    processSiteAnalytics(themeConfig);
    if (themeConfig.permalinks) await processPermalinkNotFoundWhenFirstLoaded({ siteData, router });
  }
};
const processSiteAnalytics = (themeConfig) => {
  const siteAnalytics = themeConfig.siteAnalytics || [];
  const siteAnalysis = {
    baidu: (options) => {
      index$5.baiduAnalytics(options);
      if (index$5.isClient) index$5.trackPageview(options, window.location.href);
    },
    google: (options) => index$5.googleAnalytics(options),
    umami: (options) => index$5.umamiAnalytics(options)
  };
  siteAnalytics.forEach((item) => {
    siteAnalysis[item.provider]?.(item.options);
  });
};
const processPermalinkNotFoundWhenFirstLoaded = async ({ siteData, router }) => {
  const { base, cleanUrls, themeConfig } = siteData.value;
  if (router.route.path === base && router.route.data.isNotFound) {
    const { pathname, search, hash } = new URL(location.href);
    const decodePath = "/" + decodeURIComponent(pathname.slice(base.length)).replace(/\/$/, "").replace(/\.html/, "");
    const link = cleanUrls ? decodePath : decodePath + ".html";
    const filePath = themeConfig.permalinks.inv[link];
    if (filePath) {
      const targetUrl = base + filePath + search + hash;
      await router.go(targetUrl);
    }
  }
};

exports.addUnit = index$5.addUnit;
exports.baiduAnalytics = index$5.baiduAnalytics;
exports.formatDate = index$5.formatDate;
exports.formatDiffDate = index$5.formatDiffDate;
exports.formatDiffDateToDay = index$5.formatDiffDateToDay;
exports.get = index$5.get;
exports.getDarkColor = index$5.getDarkColor;
exports.getLightColor = index$5.getLightColor;
exports.getNowDate = index$5.getNowDate;
exports.googleAnalytics = index$5.googleAnalytics;
exports.hexToRgb = index$5.hexToRgb;
exports.inBrowser = index$5.inBrowser;
exports.is = index$5.is;
exports.isArray = index$5.isArray;
exports.isAsyncFunction = index$5.isAsyncFunction;
exports.isBoolean = index$5.isBoolean;
exports.isClient = index$5.isClient;
exports.isDate = index$5.isDate;
exports.isDef = index$5.isDef;
exports.isElement = index$5.isElement;
exports.isEmpty = index$5.isEmpty;
exports.isExternal = index$5.isExternal;
exports.isFocusable = index$5.isFocusable;
exports.isFunction = index$5.isFunction;
exports.isImageDom = index$5.isImageDom;
exports.isImagePath = index$5.isImagePath;
exports.isIos = index$5.isIos;
exports.isNull = index$5.isNull;
exports.isNullAndUnDef = index$5.isNullAndUnDef;
exports.isNullOrUnDef = index$5.isNullOrUnDef;
exports.isNumber = index$5.isNumber;
exports.isObject = index$5.isObject;
exports.isPhone = index$5.isPhone;
exports.isPlainFunction = index$5.isPlainFunction;
exports.isPromise = index$5.isPromise;
exports.isServer = index$5.isServer;
exports.isString = index$5.isString;
exports.isStringNumber = index$5.isStringNumber;
exports.isType = index$5.isType;
exports.isUnDef = index$5.isUnDef;
exports.isValidURL = index$5.isValidURL;
exports.removeStorageItem = index$5.removeStorageItem;
exports.removeUnit = index$5.removeUnit;
exports.rgbToHex = index$5.rgbToHex;
exports.trackPageview = index$5.trackPageview;
exports.umamiAnalytics = index$5.umamiAnalytics;
exports.upperFirst = index$5.upperFirst;
exports.withBase = index$5.withBase;
exports.aliPayIcon = aliPay.default;
exports.arrowDownIcon = arrowDown.default;
exports.categoryIcon = category.default;
exports.closeIcon = close.default;
exports.copyrightIcon = copyright.default;
exports.docAnalysisIcon = docAnalysis.default;
exports.emptyIcon = empty.default;
exports.friendLinkIcon = friendLink.default;
exports.icpRecordIcon = icpRecord.default;
exports.noticeIcon = notice.default;
exports.playgroundIcon = playground.default;
exports.rocketIcon = rocket.default;
exports.sizeIcon = size.default;
exports.tagIcon = tag.default;
exports.themeIcon = theme.default;
exports.topArticleIcon = topArticle.default;
exports.readingIcon = reading.default;
exports.clockIcon = clock.default;
exports.viewIcon = view.default;
exports.houseIcon = house.default;
exports.arrowLeftIcon = arrowLeft.default;
exports.arrowRightIcon = arrowRight.default;
exports.magicIcon = magic.default;
exports.commentIcon = comment.default;
exports.userIcon = user.default;
exports.calendarIcon = calendar.default;
exports.folderOpenedIcon = folderOpened.default;
exports.collectionTagIcon = collectionTag.default;
exports.editPenIcon = editPen.default;
exports.githubIcon = github.default;
exports.copyIcon = copy.default;
exports.codeIcon = code.default;
exports.caretTopIcon = caretTop.default;
exports.weChatPayIcon = weChatPay.default;
exports.shareIcon = share.default;
exports.thumbsIcon = thumbs.default;
exports.externalLinkIcon = externalLink.default;
exports.dArrowRightIcon = dArrowRight.default;
exports.dArrowLeftIcon = dArrowLeft.default;
exports.moreFilledIcon = moreFilled.default;
exports.fullscreenIcon = fullscreen.default;
exports.refreshLeftIcon = refreshLeft.default;
exports.refreshRightIcon = refreshRight.default;
exports.scaleToOriginalIcon = scaleToOriginal.default;
exports.zoomInIcon = zoomIn.default;
exports.zoomOutIcon = zoomOut.default;
exports.circleCloseFilledIcon = circleCloseFilled.default;
exports.infoFilledIcon = infoFilled.default;
exports.successFilledIcon = successFilled.default;
exports.warningFilledIcon = warningFilled.default;
exports.questionFilledIcon = questionFilled.default;
exports.topIcon = top.default;
exports.lockIcon = lock.default;
exports.autoWidthIcon = autoWidth.default;
exports.clickIcon = click.default;
exports.fullScreenOneIcon = fullScreenOne.default;
exports.fullscreenTwoIcon = fullscreenTwo.default;
exports.layoutIcon = layout.default;
exports.overallReductionIcon = overallReduction.default;
exports.alignLeftIcon = alignLeft.default;
exports.alignTextLeftIcon = alignTextLeft.default;
exports.scaleIcon = scale.default;
exports.waterIcon = water.default;
exports.onClickOutside = onClickOutside.onClickOutside;
exports.useAnchorScroll = useAnchorScroll.useAnchorScroll;
exports.callBusuanzi = useUvPv.callBusuanzi;
exports.callVercount = useUvPv.callVercount;
exports.createScript = useUvPv.createScript;
exports.useUvPv = useUvPv.useUvPv;
exports.useClipboard = useClipboard.useClipboard;
exports.useCopyBanner = useCopyBanner.useCopyBanner;
exports.useDebounce = useDebounce.useDebounce;
exports.useElementHover = useElementHover.useElementHover;
exports.useEventListener = useEventListener.useEventListener;
exports.localeContextKey = useLocale.localeContextKey;
exports.useLocale = useLocale.useLocale;
exports.useMediaQuery = useMediaQuery.useMediaQuery;
exports.useMounted = useMounted.useMounted;
exports.useNamespace = useNamespace.useNamespace;
exports.usePopoverSize = usePopoverSize.usePopoverSize;
exports.useScopeDispose = useScopeDispose.useScopeDispose;
exports.useScrollData = useScrollData.useScrollData;
exports.StorageSerializers = useStorage.StorageSerializers;
exports.guessSerializerType = useStorage.guessSerializerType;
exports.useStorage = useStorage.useStorage;
exports.useSwitchData = useSwitchData.useSwitchData;
exports.useTextTypes = useTextTypes.useTextTypes;
exports.useThemeColor = useThemeColor.useThemeColor;
exports.varNameList = useThemeColor.varNameList;
exports.useViewTransition = useViewTransition.useViewTransition;
exports.useVpRouter = useVpRouter.useVpRouter;
exports.useWindowSize = useWindowSize.useWindowSize;
exports.useIntersectionObserver = useWindowTransition.useIntersectionObserver;
exports.useWindowTransition = useWindowTransition.useWindowTransition;
exports.Z_INDEX_INJECTION_KEY = useZIndex.Z_INDEX_INJECTION_KEY;
exports.defaultInitialZIndex = useZIndex.defaultInitialZIndex;
exports.useZIndex = useZIndex.useZIndex;
exports.zIndexContextKey = useZIndex.zIndexContextKey;
exports.zhCn = zhCn.default;
exports.en = en.default;
exports.createContainerThenGet = simpleContainer.createContainerThenGet;
exports.createContainerThenUse = simpleContainer.createContainerThenUse;
exports.createContainersThenGet = simpleContainer.createContainersThenGet;
exports.createContainersThenUse = simpleContainer.createContainersThenUse;
exports.createCardContainer = cardContainer.createCardContainer;
exports.createCardContainers = cardContainer.createCardContainers;
exports.version = version.version;
exports.TeekConfigProvider = index$3.TeekConfigProvider;
exports.postsContext = index$3.postsContext;
exports.teekConfigContext = index$3.teekConfigContext;
exports.useAllPosts = index$3.useAllPosts;
exports.useCommon = index$3.useCommon;
exports.usePagePath = index$3.usePagePath;
exports.usePageState = index$3.usePageState;
exports.usePosts = index$3.usePosts;
exports.useSidebar = index$3.useSidebar;
exports.useTagColor = index$3.useTagColor;
exports.useTeekConfig = index$3.useTeekConfig;
exports.useWindowTransitionConfig = index$3.useWindowTransitionConfig;
exports.TkCataloguePage = index_vue_vue_type_script_setup_true_name_CataloguePage_lang.default;
exports.TkCatalogueItem = CatalogueItem_vue_vue_type_script_setup_true_name_CatalogueItem_lang.default;
exports.TkArchivesPage = index_vue_vue_type_script_setup_true_name_ArchivesPage_lang.default;
exports.TkArticleOverviewPage = index_vue_vue_type_script_setup_true_name_ArticleOverviewPage_lang.default;
exports.TkLoginPage = index_vue_vue_type_script_setup_true_name_LoginPage_lang.default;
exports.defaultPrivateConfig = login.defaultPrivateConfig;
exports.getLoginStorageKey = login.getLoginStorageKey;
exports.loginUrlKeyMap = login.loginUrlKeyMap;
exports.verifyModeMap = login.verifyModeMap;
exports.useWatchLogin = useWatchLogin.useWatchLogin;
exports.TkRiskLinkPage = index_vue_vue_type_script_setup_true_name_RiskLinkPage_lang.default;
exports.useRiskLink = useRiskLink.useRiskLink;
exports.TkDemoCode = index_vue_vue_type_script_setup_true_name_DemoCode_lang.default;
exports.TkTitleTag = index_vue_vue_type_script_setup_true_name_TitleTag_lang.default;
exports.TkIcon = index_vue_vue_type_script_setup_true_name_Icon_lang.default;
exports.WhatsApp = SocialIcons.WhatsApp;
exports.bilibili = SocialIcons.bilibili;
exports.email = SocialIcons.email;
exports.gitee = SocialIcons.gitee;
exports.github = SocialIcons.github;
exports.moblieQQ = SocialIcons.moblieQQ;
exports.music = SocialIcons.music;
exports.telegram = SocialIcons.telegram;
Object.defineProperty(exports, "addIcons", {
  enumerable: true,
  get: function () { return vue.addCollection; }
});
exports.TkLayout = index_vue_vue_type_script_setup_true_name_TeekLayout_lang.default;
exports.TkArticlePage = index_vue_vue_type_script_setup_true_name_ArticlePage_lang.default;
exports.TkDocAsideOutline = DocAsideOutline_vue_vue_type_script_setup_true_name_DocAsideOutline_lang.default;
exports.TkAvatar = index_vue_vue_type_script_setup_true_name_Avatar_lang.default;
exports.TkBreadcrumb = Breadcrumb_vue_vue_type_script_setup_true_name_Breadcrumb_lang.default;
exports.TkBreadcrumbItem = BreadcrumbItem_vue_vue_type_script_setup_true_name_BreadcrumbItem_lang.default;
exports.TkImageViewer = index_vue_vue_type_script_setup_true_name_ImageViewer_lang.default;
exports.TkInputSlide = index_vue_vue_type_script_setup_true_name_InputSlide_lang.default;
exports.TkMessage = method.default;
exports.TkPagination = index_vue_vue_type_script_setup_true_name_Pagination_lang.default;
exports.TkPopover = index_vue_vue_type_script_setup_true_name_Popover_lang.default;
exports.TkSegmented = Segmented_vue_vue_type_script_setup_true_name_Segmented_lang.default;
exports.TkSegmentedItem = SegmentedItem_vue_vue_type_script_setup_true_name_SegmentedItem_lang.default;
exports.TkTransitionCollapse = index_vue_vue_type_script_setup_true_name_TransitionCollapse_lang.default;
exports.TkVerifyCode = index_vue_vue_type_script_setup_true_lang.default;
exports.TkVpContainer = index_vue_vue_type_script_setup_true_name_VpContainer_lang.default;
exports.TkArticleAnalyze = index_vue_vue_type_script_setup_true_name_ArticleAnalyze_lang.default;
exports.TkAsideBottomAppreciation = AsideBottomAppreciation_vue_vue_type_script_setup_true_name_AsideBottomAppreciation_lang.default;
exports.TkDocAfterAppreciation = DocAfterAppreciation_vue_vue_type_script_setup_true_name_DocAfterAppreciation_lang.default;
exports.TkDocAfterAppreciationPopper = DocAfterAppreciationPopper_vue_vue_type_script_setup_true_name_DocAfterAppreciationPopper_lang.default;
exports.TkArticleBreadcrumb = index_vue_vue_type_script_setup_true_name_ArticleBreadcrumb_lang.default;
exports.TkArticleHeadingHighlight = index_vue_vue_type_script_setup_true_name_ArticleHeadingHighlight_lang.default;
exports.TkArticleImagePreview = index_vue_vue_type_script_setup_true_name_ArticleImagePreview_lang.default;
exports.TkArticleInfo = index_vue_vue_type_script_setup_true_name_ArticleInfo_lang.default;
exports.TkArticlePageStyle = index_vue_vue_type_script_setup_true_name_ArticlePageStyle_lang.default;
exports.TkArticleShare = index_vue_vue_type_script_setup_true_name_ArticleShare_lang.default;
exports.TkArticleTitle = index_vue_vue_type_script_setup_true_name_ArticleTitle_lang.default;
exports.TkArticleUpdate = index_vue_vue_type_script_setup_true_name_ArticleUpdate_lang.default;
exports.TkBodyBgImage = index_vue_vue_type_script_setup_true_name_BodyBgImage_lang.default;
exports.TkCodeBlockToggle = index_vue_vue_type_script_setup_true_name_CodeBlockToggle_lang.default;
exports.TkCommentArtalk = index_vue_vue_type_script_setup_true_name_CommentArtalk_lang.default;
exports.TkCommentGiscus = index_vue_vue_type_script_setup_true_name_CommentGiscus_lang.default;
exports.TkCommentTwikoo = index_vue_vue_type_script_setup_true_name_CommentTwikoo_lang.default;
exports.TkCommentWaline = index_vue_vue_type_script_setup_true_name_CommentWaline_lang.default;
exports.TkFooterGroup = index_vue_vue_type_script_setup_true_name_FooterGroup_lang.default;
exports.TkFooterInfo = index_vue_vue_type_script_setup_true_name_FooterInfo_lang.default;
exports.TkHome = index_vue_vue_type_script_setup_true_name_Home_lang.default;
exports.TkHomeBanner = index_vue_vue_type_script_setup_true_name_HomeBanner_lang.default;
exports.TkHomeBannerBgImage = HomeBannerBgImage_vue_vue_type_script_setup_true_name_HomeBannerBgImage_lang.default;
exports.TkHomeBannerBgPure = HomeBannerBgPure_vue_vue_type_script_setup_true_name_HomeBannerBgPure_lang.default;
exports.TkHomeBannerContent = HomeBannerContent_vue_vue_type_script_setup_true_name_HomeBannerContent_lang.default;
exports.TkHomeBannerFeature = HomeBannerFeature_vue_vue_type_script_setup_true_name_HomeBannerFeature_lang.default;
exports.TkHomeBannerWaves = HomeBannerWaves_vue_vue_type_script_setup_true_name_HomeBannerWaves_lang.default;
exports.TkPageCard = index_vue_vue_type_script_setup_true_name_PageCard_lang.default;
exports.TkHomeCategoryCard = index_vue_vue_type_script_setup_true_name_HomeCategoryCard_lang.default;
exports.TkHomeDocAnalysisCard = index_vue_vue_type_script_setup_true_name_HomeDocAnalysisCard_lang.default;
exports.TkHomeFeature = index_vue_vue_type_script_setup_true_lang$1.default;
exports.TkHomeFriendLinkCard = index_vue_vue_type_script_setup_true_name_HomeFriendLinkCard_lang.default;
exports.TkHomeFullscreenWallpaper = index_vue_vue_type_script_setup_true_name_HomeFullscreenWallpaper_lang.default;
exports.TkHomeCardList = index_vue_vue_type_script_setup_true_name_HomeCardList_lang.default;
exports.TkHomeMyCard = index_vue_vue_type_script_setup_true_name_HomeMyCard_lang.default;
exports.TkHomeMyCardScreen = HomeMyCardScreen_vue_vue_type_script_setup_true_name_HomeMyCardScreen_lang.default;
exports.TkHomePostList = index_vue_vue_type_script_setup_true_name_HomePostList_lang.default;
exports.TkHomePostItem = HomePostItem_vue_vue_type_script_setup_true_name_HomePostItem_lang.default;
exports.TkHomeTagCard = index_vue_vue_type_script_setup_true_name_HomeTagCard_lang.default;
exports.TkHomeTopArticleCard = index_vue_vue_type_script_setup_true_name_HomeTopArticleCard_lang.default;
exports.TkNotice = index_vue_vue_type_script_setup_true_name_Notice_lang.default;
exports.TkRightBottomButton = index_vue_vue_type_script_setup_true_name_RightBottomButton_lang.default;
exports.TkRouteLoading = index$4.default;
exports.TkSidebarTrigger = index_vue_vue_type_script_setup_true_name_SidebarTrigger_lang.default;
exports.TkThemeEnhance = index_vue_vue_type_script_setup_true_name_ThemeEnhance_lang.default;
exports.TkSwitch = Switch_vue_vue_type_script_setup_true_name_Switch_lang.default;
exports.TkThemeEnhanceBaseTemplate = BaseTemplate_vue_vue_type_script_setup_true_name_BaseTemplate_lang.default;
exports.TkFocusTrap = index$1.default;
exports.createImageViewer = index$2.createImageViewer;
exports.getInstance = instance.getInstance;
exports.getLastOffset = instance.getLastOffset;
exports.getOffsetOrSpace = instance.getOffsetOrSpace;
exports.instances = instance.instances;
exports.createDynamicComponent = compile.createDynamicComponent;
exports.artalkContext = artalk.artalkContext;
exports.giscusContext = giscus.giscusContext;
exports.twikooContext = twikoo.twikooContext;
exports.walineContext = waline.walineContext;
exports.postDataUpdateSymbol = home.postDataUpdateSymbol;
exports.pageNumKey = homePostList.pageNumKey;
exports.LayoutMode = themeEnhance.LayoutMode;
exports.SpotlightStyle = themeEnhance.SpotlightStyle;
exports.ThemeColorName = themeEnhance.ThemeColorName;
exports.activateMaxWidthSlideMedia = themeEnhance.activateMaxWidthSlideMedia;
exports.layoutModeAttribute = themeEnhance.layoutModeAttribute;
exports.mobileMaxWidthMedia = themeEnhance.mobileMaxWidthMedia;
exports.themeColorAttribute = themeEnhance.themeColorAttribute;
exports.themeColorList = themeEnhance.themeColorList;
exports.touchMedia = themeEnhance.touchMedia;
exports.docMaxWidthSlideStorageKey = namespace.docMaxWidthSlideStorageKey;
exports.docMaxWidthVar = namespace.docMaxWidthVar;
exports.layoutModeStorageKey = namespace.layoutModeStorageKey;
exports.ns = namespace.ns;
exports.pageMaxWidthSlideStorageKey = namespace.pageMaxWidthSlideStorageKey;
exports.pageMaxWidthVar = namespace.pageMaxWidthVar;
exports.spotlightStorageKey = namespace.spotlightStorageKey;
exports.spotlightStyleStorageKey = namespace.spotlightStyleStorageKey;
exports.themeBgColorStorageKey = namespace.themeBgColorStorageKey;
exports.themeColorStorageKey = namespace.themeColorStorageKey;
exports.transitionName = namespace.transitionName;
exports.useThemeColorList = useThemeColorList.useThemeColorList;
exports.default = index;
