// vitepress-theme-teek@1.5.1/es/components/theme/FooterInfo/src/index.vue2.mjs
import { defineComponent, computed, createElementBlock, createCommentVNode, unref, openBlock, normalizeClass, Fragment, renderList, createBlock, toDisplayString, createElementVNode } from 'vue';
import { withBase } from 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../composables/useLocale.mjs';
import '../../../../composables/useZIndex.mjs';
import copyrightIcon from '../../../../static/icons/copyright.mjs';
import icpRecordIcon from '../../../../static/icons/icpRecord.mjs';
import themeIcon from '../../../../static/icons/theme.mjs';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';
import _sfc_main$1 from '../../../common/Icon/src/index.vue2.mjs';
import '@iconify/vue';
import securityRecordImg from '../../../../static/img/securityRecord.png.mjs';

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["aria-label"];
const _hoisted_3 = ["href", "title", "aria-label"];
const _hoisted_4 = { key: 1 };
const _hoisted_5 = ["innerHTML"];
const _hoisted_6 = ["aria-label"];
const _hoisted_7 = ["href", "aria-label"];
const _hoisted_8 = { key: 2 };
const _hoisted_9 = ["innerHTML"];
const _hoisted_10 = ["innerHTML"];
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "FooterInfo" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace("footer-info");
    const { t } = useLocale();
    const { getTeekConfigRef } = useTeekConfig();
    const footerInfo = getTeekConfigRef("footerInfo", {});
    const social = getTeekConfigRef("social", []);
    const footerData = computed(() => {
      const { theme = {}, copyright = {}, icpRecord, securityRecord } = footerInfo.value || {};
      const data = [];
      if (theme.show !== false) {
        data.push({
          name: "Theme By Teek",
          icon: themeIcon,
          link: "https://github.com/Kele-Bingtang/vitepress-theme-teek",
          // 可覆盖上面的配置项
          ...theme
        });
      }
      const { show = true, createYear = "", suffix = "" } = copyright;
      if (show) {
        data.push({
          name: `Copyright ${createYear ? `${createYear}-` : ""}${(/* @__PURE__ */ new Date()).getFullYear()} ${suffix}`,
          icon: copyrightIcon,
          ...copyright
        });
      }
      if (icpRecord) data.push({ icon: icpRecordIcon, ...icpRecord });
      if (securityRecord) {
        data.push({ icon: securityRecordImg, iconType: "img", imgAlt: "Security Record", ...securityRecord });
      }
      return data;
    });
    return (_ctx, _cache) => {
      return unref(footerInfo) || unref(social).length ? (openBlock(), createElementBlock("div", {
        key: 0,
        class: normalizeClass([unref(ns).b(), unref(ns).join("wallpaper-outside")]),
        role: "contentinfo",
        "aria-label": unref(t)("tk.footerInfo.label")
      }, [
        unref(social).length ? (openBlock(), createElementBlock("div", {
          key: 0,
          class: normalizeClass(`${unref(ns).e("icons")} flx-center`),
          role: "group",
          "aria-label": unref(t)("tk.footerInfo.socialLabel")
        }, [
          (openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(unref(social), (item, index) => {
              return openBlock(), createElementBlock("a", {
                key: index,
                href: item.link && unref(withBase)(item.link),
                title: item.name,
                target: "_blank",
                "aria-label": item.name
              }, [
                item.icon ? (openBlock(), createBlock(unref(_sfc_main$1), {
                  key: 0,
                  iconType: item.iconType,
                  icon: item.icon,
                  size: "20px",
                  color: "var(--vp-c-text-2)",
                  hover: "",
                  imgAlt: item.imgAlt,
                  "aria-hidden": "true"
                }, null, 8, ["iconType", "icon", "imgAlt"])) : item.name ? (openBlock(), createElementBlock(
                  "span",
                  _hoisted_4,
                  toDisplayString(item.name),
                  1
                  /* TEXT */
                )) : createCommentVNode("v-if", true)
              ], 8, _hoisted_3);
            }),
            128
            /* KEYED_FRAGMENT */
          ))
        ], 10, _hoisted_2)) : createCommentVNode("v-if", true),
        unref(footerInfo) ? (openBlock(), createElementBlock(
          Fragment,
          { key: 1 },
          [
            (openBlock(true), createElementBlock(
              Fragment,
              null,
              renderList([unref(footerInfo).topMessage || []].flat(), (message, index) => {
                return openBlock(), createElementBlock("p", {
                  key: index,
                  innerHTML: message,
                  class: "flx-wrap-justify-center"
                }, null, 8, _hoisted_5);
              }),
              128
              /* KEYED_FRAGMENT */
            )),
            createElementVNode("div", {
              class: normalizeClass(`${unref(ns).e("list")} flx-wrap-justify-center`),
              role: "list",
              "aria-label": unref(t)("tk.footerInfo.infoLabel")
            }, [
              (openBlock(true), createElementBlock(
                Fragment,
                null,
                renderList(footerData.value, (item) => {
                  return openBlock(), createElementBlock(
                    "div",
                    {
                      key: item.name,
                      class: normalizeClass(`${unref(ns).e("list__item")} flx-align-center`),
                      role: "listitem"
                    },
                    [
                      item.icon ? (openBlock(), createBlock(unref(_sfc_main$1), {
                        key: 0,
                        iconType: item.iconType,
                        icon: item.icon,
                        size: "16px",
                        color: "var(--vp-c-text-2)",
                        imgAlt: item.imgAlt,
                        "aria-hidden": "true"
                      }, null, 8, ["iconType", "icon", "imgAlt"])) : createCommentVNode("v-if", true),
                      item.link ? (openBlock(), createElementBlock("a", {
                        key: 1,
                        href: unref(withBase)(item.link),
                        target: "_blank",
                        "aria-label": item.name
                      }, toDisplayString(item.name), 9, _hoisted_7)) : (openBlock(), createElementBlock(
                        "span",
                        _hoisted_8,
                        toDisplayString(item.name),
                        1
                        /* TEXT */
                      ))
                    ],
                    2
                    /* CLASS */
                  );
                }),
                128
                /* KEYED_FRAGMENT */
              )),
              unref(footerInfo).customHtml ? (openBlock(), createElementBlock("span", {
                key: 0,
                innerHTML: unref(footerInfo).customHtml
              }, null, 8, _hoisted_9)) : createCommentVNode("v-if", true)
            ], 10, _hoisted_6),
            (openBlock(true), createElementBlock(
              Fragment,
              null,
              renderList([unref(footerInfo).bottomMessage || []].flat(), (message, index) => {
                return openBlock(), createElementBlock("p", {
                  key: index,
                  innerHTML: message,
                  class: "flx-wrap-justify-center"
                }, null, 8, _hoisted_10);
              }),
              128
              /* KEYED_FRAGMENT */
            ))
          ],
          64
          /* STABLE_FRAGMENT */
        )) : createCommentVNode("v-if", true)
      ], 10, _hoisted_1)) : createCommentVNode("v-if", true);
    };
  }
});

export { _sfc_main as default };
