// vitepress-theme-teek/es/components/theme/HomeBanner/src/index.vue2.mjs
import { defineComponent, computed, ref, watch, onMounted, onUnmounted, createElementBlock, openBlock, Fragment, renderSlot, createElementVNode, unref, normalizeStyle, normalizeClass, createBlock, createCommentVNode, resolveDynamicComponent, normalizeProps, guardReactiveProps, withCtx, createVNode, createSlots, renderList, nextTick } from 'vue';
import { useData } from 'vitepress';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';
import { upperFirst } from '../../../../helper/dist/index.mjs';
import { useEventListener } from '../../../../composables/useEventListener.mjs';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../composables/useLocale.mjs';
import '../../../../composables/useZIndex.mjs';
import _sfc_main$2 from './HomeBannerBgPure.vue2.mjs';
import _sfc_main$1 from './HomeBannerBgImage.vue2.mjs';
import _sfc_main$3 from './HomeBannerContent.vue2.mjs';
import _sfc_main$4 from './HomeBannerFeature.vue2.mjs';
import _sfc_main$5 from './HomeBannerWaves.vue2.mjs';

const _hoisted_1 = ["aria-label"];
const fullImgNavBarKey = "full-img-nav-bar";
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "HomeBanner" },
  __name: "index",
  props: {
    disabled: { type: Boolean, default: false }
  },
  setup(__props) {
    const ns = useNamespace("banner");
    const { t } = useLocale();
    const { getTeekConfigRef } = useTeekConfig();
    const { frontmatter } = useData();
    const bannerConfig = getTeekConfigRef("banner", {
      bgStyle: "pure",
      imgWaves: true,
      textColor: "#ffffff",
      titleFontSize: "3.2rem",
      descFontSize: "1.4rem",
      features: frontmatter.value.tk?.features || []
    });
    const bodyBgImgConfig = getTeekConfigRef("bodyBgImg", {
      imgSrc: "",
      bannerStyle: "full"
    });
    const currentBgStyle = computed(() => {
      const { bgStyle } = bannerConfig.value;
      const { imgSrc, bannerStyle } = bodyBgImgConfig.value;
      const isBannerPureBgStyle = bgStyle === "pure";
      const isBannerPartImgBgStyle = bgStyle === "partImg";
      const isBannerFullImgBgStyle = bgStyle === "fullImg";
      const isBodyImgBgStyle = !!imgSrc;
      const isBodyPartImgBgStyle = isBodyImgBgStyle && bannerStyle === "part";
      const isBodyFullImgBgStyle = isBodyImgBgStyle && bannerStyle === "full";
      return {
        isBannerPureBgStyle,
        isBannerPartImgBgStyle,
        isBannerFullImgBgStyle,
        isBodyImgBgStyle,
        isBodyPartImgBgStyle,
        isBodyFullImgBgStyle
      };
    });
    const getStyle = () => {
      const titleTextVar = ns.cssVarName("banner-title-text");
      const descTextVar = ns.cssVarName("banner-desc-text");
      const textColorVar = ns.cssVarName("banner-text-color");
      const { titleFontSize, descFontSize, textColor } = bannerConfig.value;
      return { [titleTextVar]: titleFontSize, [descTextVar]: descFontSize, [textColorVar]: textColor };
    };
    const bannerRef = ref(null);
    const toggleFullImgNavBarClass = (add = true) => {
      const vPNavDom = document.querySelector(".VPNavBar");
      if (add) vPNavDom?.classList.add(fullImgNavBarKey);
      else vPNavDom?.classList.remove(fullImgNavBarKey);
    };
    const toggleClass = async () => {
      if (!bannerRef.value || __props.disabled) return;
      const current = currentBgStyle.value;
      if (!current.isBannerFullImgBgStyle && !current.isBodyFullImgBgStyle) return;
      await nextTick();
      const offset = current.isBodyImgBgStyle ? 0 : 100;
      const windowInBanner = document.documentElement.scrollTop + offset < document.documentElement.clientHeight;
      toggleFullImgNavBarClass(windowInBanner);
    };
    watch(
      () => __props.disabled,
      () => toggleFullImgNavBarClass(!__props.disabled)
    );
    onMounted(() => {
      if (!__props.disabled && currentBgStyle.value.isBannerFullImgBgStyle || currentBgStyle.value.isBodyFullImgBgStyle) {
        toggleClass();
        useEventListener(window, "scroll", toggleClass);
      }
    });
    onUnmounted(() => {
      toggleFullImgNavBarClass(false);
    });
    const className = computed(() => {
      const {
        isBannerPureBgStyle,
        isBannerPartImgBgStyle,
        isBannerFullImgBgStyle,
        isBodyPartImgBgStyle,
        isBodyFullImgBgStyle
      } = currentBgStyle.value;
      if (isBodyPartImgBgStyle) return ns.is("part-img");
      if (isBodyFullImgBgStyle) return ns.is("full-img");
      if (isBannerPureBgStyle) return ns.is("pure");
      if (isBannerPartImgBgStyle) return ns.is("part-img");
      if (isBannerFullImgBgStyle) return ns.is("full-img");
      return "";
    });
    const styleComponentMap = {
      bodyPart: { el: "div", props: { class: `body-pure` } },
      bodyFull: { el: "div", props: { class: `body-full` } },
      bannerPure: { el: _sfc_main$2 },
      bannerPartImg: { el: _sfc_main$1 },
      bannerFullImg: { el: _sfc_main$1 }
    };
    const styleComponent = computed(() => {
      const { isBodyImgBgStyle } = currentBgStyle.value;
      const { bgStyle } = bannerConfig.value;
      const { bannerStyle } = bodyBgImgConfig.value;
      const currentStyle = isBodyImgBgStyle ? `body${upperFirst(bannerStyle)}` : `banner${upperFirst(bgStyle)}`;
      return styleComponentMap[currentStyle];
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        Fragment,
        null,
        [
          renderSlot(_ctx.$slots, "teek-home-banner-before"),
          createElementVNode("div", {
            ref_key: "bannerRef",
            ref: bannerRef,
            class: normalizeClass([unref(ns).b(), className.value]),
            style: normalizeStyle(getStyle()),
            "aria-label": unref(t)("tk.homeBanner.label")
          }, [
            (openBlock(), createBlock(
              resolveDynamicComponent(styleComponent.value.el),
              normalizeProps(guardReactiveProps(styleComponent.value.props)),
              {
                default: withCtx(() => [
                  createElementVNode(
                    "div",
                    {
                      class: normalizeClass([unref(ns).e("content"), { "no-feature": !unref(bannerConfig).features.length }])
                    },
                    [
                      renderSlot(_ctx.$slots, "teek-home-banner-content-before"),
                      createVNode(
                        _sfc_main$3,
                        null,
                        createSlots({
                          _: 2
                          /* DYNAMIC */
                        }, [
                          renderList(_ctx.$slots, (_, name) => {
                            return {
                              name,
                              fn: withCtx((scope) => [
                                renderSlot(_ctx.$slots, name, normalizeProps(guardReactiveProps(scope)))
                              ])
                            };
                          })
                        ]),
                        1024
                        /* DYNAMIC_SLOTS */
                      ),
                      renderSlot(_ctx.$slots, "teek-home-banner-content-after"),
                      renderSlot(_ctx.$slots, "teek-home-banner-feature-before"),
                      createVNode(_sfc_main$4),
                      renderSlot(_ctx.$slots, "teek-home-banner-feature-after")
                    ],
                    2
                    /* CLASS */
                  )
                ]),
                _: 3
                /* FORWARDED */
              },
              16
              /* FULL_PROPS */
            )),
            unref(bannerConfig).imgWaves && currentBgStyle.value.isBannerFullImgBgStyle && !currentBgStyle.value.isBodyImgBgStyle ? (openBlock(), createBlock(_sfc_main$5, {
              key: 0,
              "aria-label": unref(t)("tk.homeBanner.wavesLabel")
            }, null, 8, ["aria-label"])) : createCommentVNode("v-if", true)
          ], 14, _hoisted_1),
          renderSlot(_ctx.$slots, "teek-home-banner-after")
        ],
        64
        /* STABLE_FRAGMENT */
      );
    };
  }
});

export { _sfc_main as default };
