// vitepress-theme-teek@1.5.1/lib/components/theme/HomeBanner/src/HomeBannerWaves.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
require('../../../../composables/useZIndex.js');

var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "HomeBannerWaves" },
  __name: "HomeBannerWaves",
  setup(__props) {
    const ns = useNamespace.useNamespace("banner-waves");
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock(
        "svg",
        {
          class: Vue.normalizeClass(Vue.unref(ns).b()),
          xmlns: "http://www.w3.org/2000/svg",
          "xmlns:xlink": "http://www.w3.org/1999/xlink",
          viewBox: "0 24 150 28",
          preserveAspectRatio: "none",
          "shape-rendering": "auto"
        },
        [
          Vue.createCommentVNode("\u5F62\u72B6\u5BB9\u5668"),
          _cache[0] || (_cache[0] = Vue.createElementVNode(
            "defs",
            null,
            [
              Vue.createElementVNode("path", {
                id: "gentle-wave",
                d: "M-160 44c30 0 58-18 88-18s 58 18 88 18 58-18 88-18 58 18 88 18 v44h-352z"
              })
            ],
            -1
            /* CACHED */
          )),
          Vue.createCommentVNode("\u7EC4\u5408\u6CE2\u6D6A"),
          _cache[1] || (_cache[1] = Vue.createStaticVNode('<g class="parallax"><use class="use" xlink:href="#gentle-wave" x="48" y="0"></use><use class="use dark:fill-black" xlink:href="#gentle-wave" x="48" y="3"></use><use class="use dark:fill-black" xlink:href="#gentle-wave" x="48" y="5"></use><use class="use dark:fill-black" xlink:href="#gentle-wave" x="48" y="7"></use></g>', 1))
        ],
        2
        /* CLASS */
      );
    };
  }
});

exports.default = _sfc_main;
