// vitepress-theme-teek@1.5.1/es/components/theme/HomeDocAnalysisCard/src/index.vue2.mjs
import { defineComponent, computed, watch, nextTick, createBlock, openBlock, unref, normalizeClass, withCtx, createElementBlock, Fragment, renderList, createCommentVNode, createElementVNode } from 'vue';
import { useData } from 'vitepress';
import { isFunction, formatDiffDateToDay, getNowDate, formatDiffDate } from '../../../../helper/dist/index.mjs';
import { useUvPv } from '../../../../composables/useUvPv.mjs';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../composables/useLocale.mjs';
import { useVpRouter } from '../../../../composables/useVpRouter.mjs';
import '../../../../composables/useZIndex.mjs';
import docAnalysisIcon from '../../../../static/icons/docAnalysis.mjs';
import { useTeekConfig, usePosts } from '../../ConfigProvider/index.mjs';
import _sfc_main$1 from '../../../common/PageCard/src/index.vue2.mjs';

const _hoisted_1 = ["innerHTML"];
const _hoisted_2 = ["innerHTML"];
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "HomeDocAnalysisCard" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace("doc-analysis");
    const { t } = useLocale();
    const { getTeekConfigRef } = useTeekConfig();
    const { theme } = useData();
    const vpRouter = useVpRouter();
    const docAnalysisConfig = getTeekConfigRef("docAnalysis", {
      createTime: void 0,
      title: t("tk.docAnalysisCard.title", { icon: docAnalysisIcon }),
      statistics: {},
      overrideInfo: [],
      appendInfo: []
    });
    const docAnalysisInfo = computed(() => theme.value.docAnalysisInfo || {});
    const finalTitle = computed(() => {
      const { title } = docAnalysisConfig.value;
      if (isFunction(title)) return title(docAnalysisIcon);
      return title;
    });
    const createToNowDay = computed(() => formatDiffDateToDay(docAnalysisConfig.value.createTime || getNowDate()));
    const posts = usePosts();
    const postAddNum = computed(() => {
      const sortPostsByDate = posts.value.sortPostsByDate;
      let weekAddNum = 0;
      let monthAddNum = 0;
      const currentDate = new Date(getNowDate());
      for (const item of sortPostsByDate) {
        if (!item.date) continue;
        const postDate = new Date(item.date);
        if (postDate.getTime() > currentDate.getTime() - 7 * 24 * 60 * 60 * 1e3) weekAddNum++;
        if (postDate.getTime() > currentDate.getTime() - 30 * 24 * 60 * 60 * 1e3) monthAddNum++;
        else return { weekAddNum, monthAddNum };
      }
      return { weekAddNum, monthAddNum };
    });
    const formatWordCount = (wordCount) => {
      if (wordCount < 1e3) return wordCount + "";
      if (wordCount < 1e6) return Math.round(wordCount / 100) / 10 + "k";
      return Math.round(wordCount / 1e4) / 10 + "w";
    };
    const statisticsConfig = computed(() => ({
      url: "",
      provider: "",
      siteView: true,
      iteration: false,
      pageIteration: 2e3,
      permalink: true,
      ...docAnalysisConfig.value.statistics
    }));
    const useSiteView = computed(() => !!statisticsConfig.value.provider && statisticsConfig.value.siteView);
    const { router } = vpRouter;
    const { sitePv, siteUv, isGet, request } = useUvPv(false, statisticsConfig.value);
    watch(useSiteView, (newVal) => {
      if (newVal) request();
    });
    watch(
      router.route,
      () => {
        if (useSiteView.value) {
          if (statisticsConfig.value.permalink && router.state?.permalinkPlugin) {
            nextTick(request);
          } else request();
        }
      },
      { immediate: true }
    );
    const appendInfo = computed(() => {
      const { appendInfo: appendInfo2 } = docAnalysisConfig.value;
      return isFunction(appendInfo2) ? appendInfo2() : appendInfo2;
    });
    const docAnalysisList = computed(() => {
      const { createTime, overrideInfo } = docAnalysisConfig.value;
      const { fileList = [], totalFileWords, lastCommitTime } = docAnalysisInfo.value;
      const list = [
        {
          key: "totalPosts",
          label: t("tk.docAnalysisCard.totalPosts"),
          originValue: fileList.length,
          value: `${fileList.length} ${t("tk.docAnalysisCard.fileUnit")}`
        },
        {
          key: "weekAddNum",
          label: t("tk.docAnalysisCard.weekAddNum"),
          originValue: postAddNum.value?.weekAddNum,
          value: `${postAddNum.value?.weekAddNum} ${t("tk.docAnalysisCard.fileUnit")}`
        },
        {
          key: "monthAddNum",
          label: t("tk.docAnalysisCard.monthAddNum"),
          originValue: postAddNum.value?.monthAddNum,
          value: `${postAddNum.value?.monthAddNum} ${t("tk.docAnalysisCard.fileUnit")}`
        },
        {
          key: "runtime",
          label: t("tk.docAnalysisCard.runtime"),
          originValue: createTime,
          value: `${createToNowDay.value === 0 ? t("tk.docAnalysisCard.runtimeLess") : `${createToNowDay.value} ${t("tk.docAnalysisCard.runtimeUnit")}`}`
        },
        {
          key: "totalWordCount",
          label: t("tk.docAnalysisCard.totalWordCount"),
          originValue: totalFileWords,
          value: `${formatWordCount(totalFileWords)} ${t("tk.docAnalysisCard.wordCountUnit")}`
        },
        {
          key: "lastActiveTime",
          label: t("tk.docAnalysisCard.lastActiveTime"),
          originValue: lastCommitTime,
          value: formatDiffDate(lastCommitTime)
        },
        {
          key: "viewCount",
          label: t("tk.docAnalysisCard.viewCount"),
          originValue: sitePv.value,
          value: isGet.value ? `${sitePv.value} ${t("tk.docAnalysisCard.viewCountUnit")}` : "Get...",
          show: useSiteView.value
        },
        {
          key: "visitCount",
          label: t("tk.docAnalysisCard.visitCount"),
          originValue: siteUv.value,
          value: isGet.value ? `${siteUv.value} ${t("tk.docAnalysisCard.visitCountUnit")}` : "Get...",
          show: useSiteView.value
        },
        ...appendInfo.value
      ];
      if (overrideInfo.length) {
        list.forEach((item) => {
          const override = overrideInfo.find((overrideItem) => overrideItem.key === item.key);
          if (override) {
            item.label = override.label || item.label;
            item.value = override.value ? override.value(item.originValue || "", item.value) : item.value;
            item.show = override.show !== false;
          }
        });
      }
      return list;
    });
    return (_ctx, _cache) => {
      return openBlock(), createBlock(unref(_sfc_main$1), {
        title: finalTitle.value,
        class: normalizeClass(unref(ns).b()),
        "aria-label": unref(t)("tk.docAnalysisCard.label")
      }, {
        default: withCtx(() => [
          (openBlock(true), createElementBlock(
            Fragment,
            null,
            renderList(docAnalysisList.value, (item) => {
              return openBlock(), createElementBlock(
                Fragment,
                {
                  key: item.key
                },
                [
                  item.show !== false ? (openBlock(), createElementBlock(
                    "div",
                    {
                      key: 0,
                      class: normalizeClass(unref(ns).e("item"))
                    },
                    [
                      createElementVNode("span", {
                        innerHTML: item.label
                      }, null, 8, _hoisted_1),
                      createElementVNode("span", {
                        innerHTML: item.value
                      }, null, 8, _hoisted_2)
                    ],
                    2
                    /* CLASS */
                  )) : createCommentVNode("v-if", true)
                ],
                64
                /* STABLE_FRAGMENT */
              );
            }),
            128
            /* KEYED_FRAGMENT */
          ))
        ]),
        _: 1
        /* STABLE */
      }, 8, ["title", "class", "aria-label"]);
    };
  }
});

export { _sfc_main as default };
