// vitepress-theme-teek@1.5.1/lib/components/theme/Notice/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var index$1 = require('../../../../helper/dist/index.js');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
var useVpRouter = require('../../../../composables/useVpRouter.js');
require('../../../../composables/useZIndex.js');
var close = require('../../../../static/icons/close.js');
var notice = require('../../../../static/icons/notice.js');
var index = require('../../ConfigProvider/index.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../../common/Icon/src/index.vue2.js');
require('@iconify/vue');

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["aria-label"];
const _hoisted_3 = ["aria-label"];
const _hoisted_4 = { class: "flx-align-center" };
const _hoisted_5 = {
  id: "notice-title",
  class: "title sle"
};
const _hoisted_6 = ["aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "Notice" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace.useNamespace("notice");
    const { t } = useLocale.useLocale();
    const { getTeekConfigRef } = index.useTeekConfig();
    const vpRouter = useVpRouter.useVpRouter();
    const { localeIndex } = vitepress.useData();
    const noticeConfig = getTeekConfigRef("notice", {
      noticeStyle: void 0,
      iconStyle: {},
      popoverStyle: {},
      title: t("tk.notice.title"),
      initOpen: true,
      duration: 0,
      mobileMinify: false,
      reopen: true,
      useStorage: true,
      twinkle: false,
      position: "top",
      noticeIcon: notice.default,
      closeIcon: close.default,
      onAfterRouteChange: void 0
    });
    const destroyNoticeIcon = Vue.ref(false);
    const showNoticeIcon = Vue.computed(() => !showPopover.value && !destroyNoticeIcon.value);
    const showPopover = Vue.ref(noticeConfig.value.initOpen);
    Vue.watch(
      () => noticeConfig.value.initOpen,
      (newValue) => showPopover.value = newValue
    );
    const styleObj = Vue.computed(() => {
      const { noticeStyle } = noticeConfig.value;
      if (!noticeStyle) return "";
      return noticeStyle.trim().startsWith(`.${ns.b()}`) ? noticeStyle : `.${ns.b()} { ${noticeStyle} }`;
    });
    const noticeTitle = Vue.computed(() => {
      const { title } = noticeConfig.value;
      if (index$1.isString(title)) return title;
      return title(localeIndex.value);
    });
    const { isMobile } = index.useCommon();
    Vue.watch(
      () => noticeConfig.value.mobileMinify,
      (val) => {
        if (isMobile.value) destroyNoticeIcon.value = val;
      },
      { immediate: true }
    );
    let timer;
    const closePopoverWhenTimeout = () => {
      const { duration } = noticeConfig.value;
      if (showPopover.value && duration > 0) {
        if (timer) {
          clearTimeout(timer);
          timer = null;
        }
        timer = setTimeout(handleClosePopover, duration);
      }
    };
    Vue.onMounted(() => {
      vpRouter.bindAfterRouteChange(
        ns.join("notice"),
        () => noticeConfig.value.onAfterRouteChange?.(vpRouter.route, showNoticeIcon.value, showPopover.value)
      );
      closePopoverWhenTimeout();
    });
    const openOrDisableScroll = (action) => {
      if (!index$1.isClient) return;
      if (noticeConfig.value.position !== "center") return;
      const actions = {
        open: "remove",
        disable: "add"
      };
      document.documentElement.classList[actions[action]]("disable-scroll");
    };
    const storageKey = Vue.computed(() => ns.storageKey("notice", localeIndex.value, index$1.isClient ? window.location.hostname : ""));
    if (noticeConfig.value.useStorage) {
      Vue.watch(
        localeIndex,
        () => {
          if (!noticeConfig.value.useStorage) return;
          const oldValue = localStorage.getItem(storageKey.value);
          if (oldValue) {
            const isShowPopover = oldValue === "true";
            showPopover.value = isShowPopover;
            if (isShowPopover) openOrDisableScroll("disable");
          }
        },
        { immediate: true }
      );
    }
    const handleOpenPopover = () => {
      showPopover.value = true;
      storagePopoverState("true");
      closePopoverWhenTimeout();
      openOrDisableScroll("disable");
    };
    const handleClosePopover = () => {
      showPopover.value = false;
      storagePopoverState("false");
      if (!noticeConfig.value.reopen) destroyNoticeIcon.value = true;
      if (timer) clearTimeout(timer);
      openOrDisableScroll("open");
    };
    const storagePopoverState = (state) => {
      if (noticeConfig.value.useStorage) localStorage.setItem(storageKey.value, state);
    };
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("div", {
        class: Vue.normalizeClass([Vue.unref(ns).b(), Vue.unref(ns).join("wallpaper-outside")]),
        "aria-label": Vue.unref(t)("tk.notice.label")
      }, [
        styleObj.value ? (Vue.openBlock(), Vue.createBlock(Vue.resolveDynamicComponent("style"), { key: 0 }, {
          default: Vue.withCtx(() => [
            Vue.createTextVNode(
              Vue.toDisplayString(styleObj.value),
              1
              /* TEXT */
            )
          ]),
          _: 1
          /* STABLE */
        })) : Vue.createCommentVNode("v-if", true),
        Vue.createCommentVNode(" \u516C\u544A\u56FE\u6807 "),
        !destroyNoticeIcon.value ? Vue.withDirectives((Vue.openBlock(), Vue.createElementBlock("div", {
          key: 1,
          class: Vue.normalizeClass([Vue.unref(ns).e("icon"), { twinkle: Vue.unref(noticeConfig).twinkle }, "flx"]),
          style: Vue.normalizeStyle(Vue.unref(noticeConfig).iconStyle),
          onClick: handleOpenPopover,
          role: "button",
          "aria-label": Vue.unref(t)("tk.notice.openLabel")
        }, [
          Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
            icon: Vue.unref(noticeConfig).noticeIcon,
            color: "#ffffff",
            size: "14px",
            "aria-hidden": "true"
          }, null, 8, ["icon"])
        ], 14, _hoisted_2)), [
          [Vue.vShow, showNoticeIcon.value]
        ]) : Vue.createCommentVNode("v-if", true),
        Vue.createCommentVNode(" \u516C\u544A\u5F39\u7A97 "),
        Vue.withDirectives(Vue.createElementVNode(
          "div",
          {
            class: Vue.normalizeClass([Vue.unref(ns).e("popover"), Vue.unref(ns).is(Vue.unref(noticeConfig).position)]),
            style: Vue.normalizeStyle(Vue.unref(noticeConfig).popoverStyle),
            role: "dialog",
            "aria-modal": true,
            "aria-labelledby": "notice-title"
          },
          [
            Vue.renderSlot(_ctx.$slots, "header", {}, () => [
              Vue.createElementVNode("div", {
                class: Vue.normalizeClass([Vue.unref(ns).e("popover__header"), "flx-justify-between"]),
                "aria-label": Vue.unref(t)("tk.notice.headLabel")
              }, [
                Vue.createElementVNode("div", _hoisted_4, [
                  Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                    icon: Vue.unref(noticeConfig).noticeIcon,
                    color: "#ffffff",
                    size: "20px",
                    "aria-hidden": "true"
                  }, null, 8, ["icon"]),
                  Vue.createElementVNode(
                    "span",
                    _hoisted_5,
                    Vue.toDisplayString(noticeTitle.value),
                    1
                    /* TEXT */
                  )
                ]),
                Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                  icon: Vue.unref(noticeConfig).closeIcon,
                  color: "#ffffff",
                  size: "20px",
                  class: Vue.normalizeClass(Vue.unref(ns).join("pointer")),
                  onClick: handleClosePopover,
                  role: "button",
                  "aria-label": Vue.unref(t)("tk.notice.closeLabel")
                }, null, 8, ["icon", "class", "aria-label"])
              ], 10, _hoisted_3)
            ]),
            Vue.createElementVNode("div", {
              class: Vue.normalizeClass(Vue.unref(ns).e("popover__content")),
              "aria-label": Vue.unref(t)("tk.notice.contentLabel")
            }, [
              Vue.renderSlot(_ctx.$slots, "teek-notice-content")
            ], 10, _hoisted_6)
          ],
          6
          /* CLASS, STYLE */
        ), [
          [Vue.vShow, showPopover.value]
        ]),
        Vue.createCommentVNode(" \u906E\u7F69\u5C42\uFF0C\u4E0E\u516C\u544A\u5F39\u7A97\u4E00\u8D77\u51FA\u73B0 "),
        Vue.withDirectives(Vue.createElementVNode(
          "div",
          {
            class: Vue.normalizeClass(Vue.unref(ns).e("mask")),
            role: "presentation"
          },
          null,
          2
          /* CLASS */
        ), [
          [Vue.vShow, showPopover.value && Vue.unref(noticeConfig).position === "center"]
        ])
      ], 10, _hoisted_1);
    };
  }
});

exports.default = _sfc_main;
