// vitepress-theme-teek/lib/config/index.js
'use strict';

var todo = require('../markdown/plugins/todo.js');
var container = require('../markdown/plugins/container.js');
var shareCard = require('../markdown/plugins/shareCard.js');
var imgCard = require('../markdown/plugins/imgCard.js');
var navCard = require('../markdown/plugins/navCard.js');
var demo = require('../markdown/plugins/demo.js');
var video = require('../markdown/plugins/video.js');
var simpleContainer = require('../markdown/helper/simpleContainer.js');
require('markdown-it-container');
require('js-yaml');
var vitePlugins = require('./vitePlugins.js');
var Permalink = require('vitepress-plugin-permalink');
var themeEnhance = require('../components/theme/ThemeEnhance/src/themeEnhance.js');

const defineTeekConfig = (config = {}) => {
  const { vitePlugins: vitePlugins$1, markdown = {}, ...teekConfig } = config;
  const plugins = vitePlugins.registerPluginAndGet(vitePlugins$1, teekConfig.teekTheme);
  const head = [];
  if (teekConfig.docAnalysis?.statistics?.provider === "busuanzi") {
    head.push(["meta", { name: "referrer", content: "no-referrer-when-downgrade" }]);
  }
  return {
    // 使用永久链接插件需要忽略死链提醒
    ignoreDeadLinks: true,
    metaChunk: true,
    head,
    vite: {
      plugins,
      ssr: { noExternal: ["vitepress-theme-teek"] },
      // 解决项目启动后终端打印 Scss 的废弃警告：The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.
      css: { preprocessorOptions: { scss: { api: "modern" } } }
    },
    markdown: {
      config: (md) => {
        [todo.default, shareCard.default, imgCard.default, navCard.default, video.default].forEach((plugin) => md.use(plugin));
        const { container: container$1 = {}, demo: demo$1, config: config2 } = markdown;
        if (!demo$1?.disabled) md.use(demo.default, demo$1).use(container.default, container$1.label);
        simpleContainer.createContainersThenUse(md, container$1.config?.() || []);
        config2?.(md);
      }
    },
    themeConfig: teekConfig
  };
};

Object.defineProperty(exports, "createRewrites", {
  enumerable: true,
  get: function () { return Permalink.createRewrites; }
});
exports.LayoutMode = themeEnhance.LayoutMode;
exports.SpotlightStyle = themeEnhance.SpotlightStyle;
exports.ThemeColorName = themeEnhance.ThemeColorName;
exports.defineTeekConfig = defineTeekConfig;
