// vitepress-theme-teek@1.5.1/lib/components/theme/HomeTagCard/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var index$1 = require('../../../../helper/dist/index.js');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var tag = require('../../../../static/icons/tag.js');
var homePostList = require('../../HomePostList/src/homePostList.js');
var index = require('../../ConfigProvider/index.js');
var home = require('../../Home/src/home.js');
var index_vue_vue_type_script_setup_true_name_PageCard_lang = require('../../../common/PageCard/src/index.vue2.js');

const _hoisted_1 = ["onClick", "aria-label"];
const _hoisted_2 = { class: "num" };
const _hoisted_3 = ["href", "aria-label"];
const _hoisted_4 = ["aria-label"];
const tagKey = "tag";
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "HomeTagCard" },
  __name: "index",
  props: {
    tagsPage: { type: Boolean, default: false }
  },
  setup(__props) {
    const ns = useNamespace.useNamespace("tag");
    const { t } = useLocale.useLocale();
    const { getTeekConfigRef } = index.useTeekConfig();
    const pageNum = Vue.ref(1);
    const tagConfig = getTeekConfigRef("tag", {
      pageTitle: t("tk.tagCard.pageTitle", { icon: tag.default }),
      homeTitle: t("tk.tagCard.homeTitle", { icon: tag.default }),
      emptyLabel: t("tk.tagCard.emptyLabel"),
      moreLabel: t("tk.tagCard.moreLabel"),
      limit: 21,
      autoPage: false,
      pageSpeed: 4e3,
      bgColor: ""
    });
    const { tagPath } = index.usePagePath();
    const posts = index.usePosts();
    const tagColor = index.useTagColor();
    const tags = Vue.computed(() => posts.value.groupCards.tags);
    const currentTags = Vue.computed(() => {
      const { limit } = tagConfig.value;
      const t2 = tags.value;
      const p = pageNum.value;
      return __props.tagsPage ? t2 : t2.slice((p - 1) * limit, p * limit);
    });
    const finalTitle = Vue.computed(() => {
      const { pageTitle, homeTitle } = tagConfig.value;
      const pt = index$1.isFunction(pageTitle) ? pageTitle(tag.default) : pageTitle;
      const ht = index$1.isFunction(homeTitle) ? homeTitle(tag.default) : homeTitle;
      return { pt, ht };
    });
    const getTagStyle = (index) => {
      const tagColorConst = tagColor.value;
      const color = tagColorConst[index % tagColorConst.length];
      return {
        [ns.cssVarName("home-tag-bg-color")]: color.bg,
        backgroundColor: color.bg,
        color: color.text,
        borderColor: color.border
      };
    };
    const updatePostListData = Vue.inject(home.postDataUpdateSymbol, () => {
    });
    const router = vitepress.useRouter();
    const selectedTag = Vue.ref("");
    const handleSwitchTag = (tag = "") => {
      const { pathname, searchParams } = new URL(window.location.href);
      const categoriesPageLinkConst = vitepress.withBase(tagPath.value);
      const inCategoriesPage = categoriesPageLinkConst === pathname;
      searchParams.delete(homePostList.pageNumKey);
      searchParams.append(homePostList.pageNumKey, "1");
      searchParams.delete(tagKey);
      if (tag) searchParams.append(tagKey, tag);
      const searchParamsStr = tag ? `?${searchParams.toString()}` : "";
      if (inCategoriesPage && selectedTag.value === tag) return;
      selectedTag.value = tag;
      if (!inCategoriesPage) return router.go(categoriesPageLinkConst + searchParamsStr);
      window.history.pushState({}, "", pathname + searchParamsStr);
      updatePostListData();
    };
    Vue.onMounted(() => {
      const { searchParams } = new URL(window.location.href);
      const tag = searchParams.get(tagKey);
      if (tag) selectedTag.value = tag;
    });
    Vue.watch(
      () => __props.tagsPage,
      () => {
        if (!__props.tagsPage) {
          selectedTag.value = "";
          return;
        }
        const { searchParams } = new URL(window.location.href);
        const tag = searchParams.get(tagKey);
        if (tag && selectedTag.value !== tag) selectedTag.value = tag;
      }
    );
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_PageCard_lang.default), {
        page: !__props.tagsPage,
        modelValue: pageNum.value,
        "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => pageNum.value = $event),
        pageSize: Vue.unref(tagConfig).limit,
        total: tags.value.length,
        title: finalTitle.value[__props.tagsPage ? "pt" : "ht"],
        titleClick: handleSwitchTag,
        autoPage: Vue.unref(tagConfig).autoPage,
        pageSpeed: Vue.unref(tagConfig).pageSpeed,
        class: Vue.normalizeClass([Vue.unref(ns).b(), Vue.unref(ns).is("page", __props.tagsPage)]),
        "aria-label": Vue.unref(t)("tk.tagCard.label")
      }, {
        default: Vue.withCtx(({ transitionName }) => [
          tags.value.length ? (Vue.openBlock(), Vue.createBlock(Vue.TransitionGroup, {
            key: 0,
            name: transitionName,
            tag: "div",
            mode: "out-in",
            class: Vue.normalizeClass(Vue.unref(ns).e("list")),
            "aria-label": Vue.unref(t)("tk.tagCard.listLabel")
          }, {
            default: Vue.withCtx(() => [
              (Vue.openBlock(true), Vue.createElementBlock(
                Vue.Fragment,
                null,
                Vue.renderList(currentTags.value, (item, index) => {
                  return Vue.openBlock(), Vue.createElementBlock("a", {
                    key: item.name,
                    style: Vue.normalizeStyle(getTagStyle(index)),
                    onClick: ($event) => handleSwitchTag(item.name),
                    class: Vue.normalizeClass([{ active: item.name === selectedTag.value }, Vue.unref(ns).join("pointer")]),
                    "aria-label": item.name
                  }, [
                    Vue.createElementVNode(
                      "span",
                      null,
                      Vue.toDisplayString(item.name),
                      1
                      /* TEXT */
                    ),
                    Vue.createElementVNode(
                      "span",
                      _hoisted_2,
                      Vue.toDisplayString(item.length),
                      1
                      /* TEXT */
                    )
                  ], 14, _hoisted_1);
                }),
                128
                /* KEYED_FRAGMENT */
              )),
              !__props.tagsPage && Vue.unref(tagConfig).limit < tags.value.length ? (Vue.openBlock(), Vue.createElementBlock("a", {
                key: 0,
                href: Vue.unref(vitepress.withBase)(Vue.unref(tagPath)),
                class: "more",
                "aria-label": Vue.unref(tagConfig).moreLabel
              }, Vue.toDisplayString(Vue.unref(tagConfig).moreLabel), 9, _hoisted_3)) : Vue.createCommentVNode("v-if", true)
            ]),
            _: 1
            /* STABLE */
          }, 8, ["name", "class", "aria-label"])) : (Vue.openBlock(), Vue.createElementBlock("div", {
            key: 1,
            class: Vue.normalizeClass(Vue.unref(ns).m("empty")),
            "aria-label": Vue.unref(tagConfig).emptyLabel
          }, Vue.toDisplayString(Vue.unref(tagConfig).emptyLabel), 11, _hoisted_4))
        ]),
        _: 1
        /* STABLE */
      }, 8, ["page", "modelValue", "pageSize", "total", "title", "autoPage", "pageSpeed", "class", "aria-label"]);
    };
  }
});

exports.default = _sfc_main;
