// vitepress-theme-teek@1.5.1/es/components/theme/ArchivesPage/src/index.vue2.mjs
import { defineComponent, computed, ref, onMounted, resolveComponent, createBlock, openBlock, unref, normalizeClass, withCtx, renderSlot, createElementVNode, toDisplayString, createVNode, createElementBlock, Fragment, renderList } from 'vue';
import { useData, withBase } from 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useLocale } from '../../../../composables/useLocale.mjs';
import { useWindowTransition } from '../../../../composables/useWindowTransition.mjs';
import '../../../../composables/useZIndex.mjs';
import { usePosts, useWindowTransitionConfig } from '../../ConfigProvider/index.mjs';
import _sfc_main$1 from '../../../common/ArticlePage/src/index.vue2.mjs';
import _sfc_main$2 from '../../ArticleTitle/src/index.vue2.mjs';

const _hoisted_1 = { class: "count" };
const _hoisted_2 = { class: "vp-doc" };
const _hoisted_3 = { class: "year" };
const _hoisted_4 = { class: "count" };
const _hoisted_5 = { class: "month" };
const _hoisted_6 = { class: "count" };
const _hoisted_7 = ["href", "aria-label"];
const _hoisted_8 = { class: "date" };
var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "ArchivesPage" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace("archives");
    const { t } = useLocale();
    const { frontmatter } = useData();
    const posts = usePosts();
    const defaultLabel = computed(() => {
      const frontmatterConst = frontmatter.value;
      return {
        title: frontmatterConst.title ?? t("tk.archives.title"),
        totalCount: frontmatterConst.totalCount ?? t("tk.archives.totalCount"),
        year: frontmatterConst.year ?? t("tk.archives.year"),
        month: frontmatterConst.month ?? t("tk.archives.month"),
        count: frontmatterConst.count ?? t("tk.archives.count"),
        notFound: frontmatterConst.notFound ?? t("tk.archives.notFound")
      };
    });
    const windowTransition = useWindowTransitionConfig((config) => config.archives);
    const timelineItemListInstance = ref(null);
    const { start } = useWindowTransition(timelineItemListInstance, false);
    onMounted(() => {
      windowTransition.value && start();
    });
    return (_ctx, _cache) => {
      const _component_Content = resolveComponent("Content");
      return openBlock(), createBlock(unref(_sfc_main$1), {
        class: normalizeClass(unref(ns).b()),
        "aria-label": unref(t)("tk.archives.label")
      }, {
        default: withCtx(() => [
          renderSlot(_ctx.$slots, "teek-archives-top-before"),
          createElementVNode(
            "div",
            {
              class: normalizeClass(`${unref(ns).e("header")} flx-justify-between`)
            },
            [
              createElementVNode(
                "h1",
                null,
                toDisplayString(defaultLabel.value.title),
                1
                /* TEXT */
              ),
              createElementVNode(
                "div",
                _hoisted_1,
                toDisplayString(defaultLabel.value.totalCount.replace("{count}", unref(posts).sortPostsByDate.length)),
                1
                /* TEXT */
              )
            ],
            2
            /* CLASS */
          ),
          renderSlot(_ctx.$slots, "teek-archives-top-after"),
          createElementVNode("div", _hoisted_2, [
            createVNode(_component_Content)
          ]),
          createElementVNode(
            "div",
            {
              class: normalizeClass(unref(ns).e("timeline"))
            },
            [
              (openBlock(true), createElementBlock(
                Fragment,
                null,
                renderList(unref(posts).groupPostsByYearMonth, (monthPosts, year) => {
                  return openBlock(), createElementBlock(
                    Fragment,
                    { key: year },
                    [
                      createElementVNode(
                        "div",
                        {
                          class: normalizeClass(`${unref(ns).em("timeline", "year")} flx-justify-between`)
                        },
                        [
                          createElementVNode(
                            "div",
                            _hoisted_3,
                            toDisplayString(String(year).trim() === "NaN" ? defaultLabel.value.notFound : String(year).trim() + defaultLabel.value.year),
                            1
                            /* TEXT */
                          ),
                          createElementVNode(
                            "div",
                            _hoisted_4,
                            toDisplayString(unref(posts).groupPostsByYear[year].length + defaultLabel.value.count),
                            1
                            /* TEXT */
                          )
                        ],
                        2
                        /* CLASS */
                      ),
                      createElementVNode(
                        "div",
                        {
                          class: normalizeClass(unref(ns).e("timeline__m"))
                        },
                        [
                          (openBlock(true), createElementBlock(
                            Fragment,
                            null,
                            renderList(monthPosts, (p, month) => {
                              return openBlock(), createElementBlock(
                                Fragment,
                                { key: month },
                                [
                                  createElementVNode(
                                    "div",
                                    {
                                      class: normalizeClass(`${unref(ns).em("timeline__m", "month")} flx-justify-between`)
                                    },
                                    [
                                      createElementVNode(
                                        "div",
                                        _hoisted_5,
                                        toDisplayString(String(month) === "NaN" ? defaultLabel.value.notFound : month + defaultLabel.value.month),
                                        1
                                        /* TEXT */
                                      ),
                                      createElementVNode(
                                        "div",
                                        _hoisted_6,
                                        toDisplayString(p.length + defaultLabel.value.count),
                                        1
                                        /* TEXT */
                                      )
                                    ],
                                    2
                                    /* CLASS */
                                  ),
                                  createElementVNode("ul", null, [
                                    (openBlock(true), createElementBlock(
                                      Fragment,
                                      null,
                                      renderList(p, (item) => {
                                        return openBlock(), createElementBlock("li", {
                                          ref_for: true,
                                          ref_key: "timelineItemListInstance",
                                          ref: timelineItemListInstance,
                                          key: item.url
                                        }, [
                                          createElementVNode("a", {
                                            href: item.url && unref(withBase)(item.url),
                                            "aria-label": `${item.title}`
                                          }, [
                                            createElementVNode(
                                              "span",
                                              _hoisted_8,
                                              toDisplayString(item.date?.slice(5, 10)),
                                              1
                                              /* TEXT */
                                            ),
                                            createVNode(unref(_sfc_main$2), {
                                              post: item,
                                              "title-tag-props": { position: "right", size: "small" }
                                            }, null, 8, ["post"])
                                          ], 8, _hoisted_7)
                                        ]);
                                      }),
                                      128
                                      /* KEYED_FRAGMENT */
                                    ))
                                  ])
                                ],
                                64
                                /* STABLE_FRAGMENT */
                              );
                            }),
                            128
                            /* KEYED_FRAGMENT */
                          ))
                        ],
                        2
                        /* CLASS */
                      )
                    ],
                    64
                    /* STABLE_FRAGMENT */
                  );
                }),
                128
                /* KEYED_FRAGMENT */
              ))
            ],
            2
            /* CLASS */
          )
        ]),
        _: 3
        /* FORWARDED */
      }, 8, ["class", "aria-label"]);
    };
  }
});

export { _sfc_main as default };
