// vitepress-theme-teek@1.5.1/lib/components/common/TitleTag/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
require('../../../../composables/useZIndex.js');

var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "TitleTag" },
  __name: "index",
  props: {
    text: {},
    type: {},
    position: {},
    size: {}
  },
  setup(__props) {
    const ns = useNamespace.useNamespace("title-tag");
    return (_ctx, _cache) => {
      return __props.text || _ctx.$slots.default ? (Vue.openBlock(), Vue.createElementBlock(
        "span",
        {
          key: 0,
          class: Vue.normalizeClass([Vue.unref(ns).b(), __props.type && Vue.unref(ns).m(__props.type), __props.position && Vue.unref(ns).m(__props.position), __props.size && Vue.unref(ns).m(__props.size)])
        },
        [
          Vue.renderSlot(_ctx.$slots, "default", {}, () => [
            Vue.createTextVNode(
              Vue.toDisplayString(__props.text),
              1
              /* TEXT */
            )
          ])
        ],
        2
        /* CLASS */
      )) : Vue.createCommentVNode("v-if", true);
    };
  }
});

exports.default = _sfc_main;
