// vitepress-theme-teek@1.5.1/es/components/theme/ThemeEnhance/src/index.vue2.mjs
import { defineComponent, computed, createBlock, createCommentVNode, unref, openBlock, normalizeClass, withCtx, createElementVNode, renderSlot, createElementBlock, Fragment, createVNode } from 'vue';
import readingIcon from '../../../../static/icons/reading.mjs';
import 'vitepress';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import { useMediaQuery } from '../../../../composables/useMediaQuery.mjs';
import '../../../../composables/useZIndex.mjs';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';
import _sfc_main$8 from '../../../common/Icon/src/index.vue2.mjs';
import '@iconify/vue';
import _sfc_main$1 from '../../../common/Popover/src/index.vue2.mjs';
import { ns } from './namespace.mjs';
import { mobileMaxWidthMedia } from './themeEnhance.mjs';
import _sfc_main$2 from './LayoutSwitch.vue2.mjs';
import _sfc_main$3 from './LayoutPageWidthSlide.vue2.mjs';
import _sfc_main$4 from './LayoutDocWidthSlide.vue2.mjs';
import _sfc_main$5 from './ThemeColor.vue2.mjs';
import _sfc_main$6 from './Spotlight.vue2.mjs';
import _sfc_main$7 from './SpotlightStyle.vue2.mjs';

var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "ThemeEnhance" },
  __name: "index",
  setup(__props) {
    const { getTeekConfigRef } = useTeekConfig();
    const themeEnhanceConfig = getTeekConfigRef("themeEnhance", { position: "top" });
    const isMobile = useMediaQuery(mobileMaxWidthMedia);
    const disabledList = computed(() => {
      return {
        layoutSwitch: themeEnhanceConfig.value.layoutSwitch?.disabled ?? false,
        themeColor: themeEnhanceConfig.value.themeColor?.disabled ?? false,
        spotlight: themeEnhanceConfig.value.spotlight?.disabled ?? false
      };
    });
    return (_ctx, _cache) => {
      return !unref(isMobile) && unref(themeEnhanceConfig).position === "top" ? (openBlock(), createBlock(unref(_sfc_main$1), {
        key: 0,
        class: normalizeClass([unref(ns).b(), "flx-align-center"]),
        "popper-class": unref(ns).e("popover"),
        "y-offset": -15
      }, {
        reference: withCtx(() => [
          createVNode(unref(_sfc_main$8), {
            icon: unref(readingIcon),
            size: 20
          }, null, 8, ["icon"])
        ]),
        default: withCtx(() => [
          createElementVNode(
            "div",
            {
              class: normalizeClass(unref(ns).e("content"))
            },
            [
              renderSlot(_ctx.$slots, "teek-theme-enhance-top"),
              !disabledList.value.layoutSwitch ? (openBlock(), createElementBlock(
                Fragment,
                { key: 0 },
                [
                  createVNode(_sfc_main$2),
                  createVNode(_sfc_main$3),
                  createVNode(_sfc_main$4)
                ],
                64
                /* STABLE_FRAGMENT */
              )) : createCommentVNode("v-if", true),
              !disabledList.value.themeColor ? (openBlock(), createBlock(_sfc_main$5, { key: 1 })) : createCommentVNode("v-if", true),
              !disabledList.value.spotlight ? (openBlock(), createElementBlock(
                Fragment,
                { key: 2 },
                [
                  createVNode(_sfc_main$6),
                  createVNode(_sfc_main$7)
                ],
                64
                /* STABLE_FRAGMENT */
              )) : createCommentVNode("v-if", true),
              renderSlot(_ctx.$slots, "teek-theme-enhance-bottom")
            ],
            2
            /* CLASS */
          )
        ]),
        _: 3
        /* FORWARDED */
      }, 8, ["class", "popper-class"])) : createCommentVNode("v-if", true);
    };
  }
});

export { _sfc_main as default };
