// vitepress-theme-teek@1.5.1/es/components/theme/ThemeEnhance/src/themeEnhance.mjs
var LayoutMode = /* @__PURE__ */ ((LayoutMode2) => {
  LayoutMode2["FullWidth"] = "fullWidth";
  LayoutMode2["SidebarWidthAdjustableOnly"] = "sidebarWidthAdjustableOnly";
  LayoutMode2["BothWidthAdjustable"] = "bothWidthAdjustable";
  LayoutMode2["Original"] = "original";
  return LayoutMode2;
})(LayoutMode || {});
var SpotlightStyle = /* @__PURE__ */ ((SpotlightStyle2) => {
  SpotlightStyle2["Under"] = "under";
  SpotlightStyle2["Aside"] = "aside";
  return SpotlightStyle2;
})(SpotlightStyle || {});
var ThemeColorName = /* @__PURE__ */ ((ThemeColorName2) => {
  ThemeColorName2["vpDefault"] = "vp-default";
  ThemeColorName2["vpGreen"] = "vp-green";
  ThemeColorName2["vpYellow"] = "vp-yellow";
  ThemeColorName2["vpRed"] = "vp-red";
  ThemeColorName2["epBlue"] = "ep-blue";
  ThemeColorName2["epGreen"] = "ep-green";
  ThemeColorName2["epYellow"] = "ep-yellow";
  ThemeColorName2["epRed"] = "ep-red";
  return ThemeColorName2;
})(ThemeColorName || {});
const themeColorList = [
  "vp-default" /* vpDefault */,
  "vp-green" /* vpGreen */,
  "vp-yellow" /* vpYellow */,
  "vp-red" /* vpRed */,
  "ep-blue" /* epBlue */,
  "ep-green" /* epGreen */,
  "ep-yellow" /* epYellow */,
  "ep-red" /* epRed */
];
const mobileMaxWidthMedia = "(max-width: 768px)";
const activateMaxWidthSlideMedia = "(min-width: 1440px)";
const touchMedia = "(pointer: coarse)";
const layoutModeAttribute = "layout-mode";
const themeColorAttribute = "theme-color";

export { LayoutMode, SpotlightStyle, ThemeColorName, activateMaxWidthSlideMedia, layoutModeAttribute, mobileMaxWidthMedia, themeColorAttribute, themeColorList, touchMedia };
