// vitepress-theme-teek/lib/components/theme/HomeBanner/src/HomeBannerContent.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
var useSwitchData = require('../../../../composables/useSwitchData.js');
var useTextTypes = require('../../../../composables/useTextTypes.js');
require('../../../../composables/useZIndex.js');
var index = require('../../ConfigProvider/index.js');

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["aria-label"];
const _hoisted_3 = ["aria-label"];
const _hoisted_4 = { key: 0 };
const _hoisted_5 = ["aria-label"];
const _hoisted_6 = ["aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "HomeBannerContent" },
  __name: "HomeBannerContent",
  setup(__props) {
    const ns = useNamespace.useNamespace("banner-content");
    const { t } = useLocale.useLocale();
    const { getTeekConfigRef } = index.useTeekConfig();
    const { site, frontmatter } = vitepress.useData();
    const bannerConfig = getTeekConfigRef("banner", {
      name: frontmatter.value.tk?.name || site.value.title || "",
      descStyle: "default",
      description: [],
      switchTime: 4e3,
      switchShuffle: false,
      typesInTime: 200,
      typesOutTime: 100,
      typesNextTime: 800,
      typesShuffle: false
    });
    const descArray = Vue.computed(() => [
      ...new Set(
        [frontmatter.value.tk?.description || bannerConfig.value.description || []].flat()?.filter((v) => !!v)
      )
    ]);
    const isDefaultDescStyle = Vue.computed(() => bannerConfig.value.descStyle === "default");
    const isTypesDescStyle = Vue.computed(() => bannerConfig.value.descStyle === "types");
    const isSwitchDescStyle = Vue.computed(() => bannerConfig.value.descStyle === "switch");
    const {
      text: typesText,
      isFinished,
      start: startTypes
    } = useTextTypes.useTextTypes(descArray, {
      inputTime: bannerConfig.value.typesInTime,
      outputTime: bannerConfig.value.typesOutTime,
      nextTime: bannerConfig.value.typesNextTime,
      shuffle: bannerConfig.value.typesShuffle
    });
    const { data: text, start: startAutoSwitch } = useSwitchData.useSwitchData(descArray, {
      timeout: bannerConfig.value.switchTime,
      shuffle: bannerConfig.value.switchShuffle,
      onUpdate: (data, newValue) => {
        data.value = "";
        setTimeout(() => {
          data.value = newValue;
        }, 20);
      }
    });
    Vue.onMounted(() => {
      if (isTypesDescStyle.value) startTypes();
      if (isSwitchDescStyle.value) startAutoSwitch();
    });
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("div", {
        class: Vue.normalizeClass(Vue.unref(ns).b()),
        "aria-label": Vue.unref(t)("tk.homeBanner.contentLabel")
      }, [
        Vue.createCommentVNode(" \u9996\u9875 banner \u56FE\u6587\u5B57\u63D2\u69FD "),
        _ctx.$slots["teek-home-banner-name"] ? Vue.renderSlot(_ctx.$slots, "teek-home-banner-name", Vue.normalizeProps(Vue.mergeProps({ key: 0 }, { name: Vue.unref(bannerConfig).name }))) : (Vue.openBlock(), Vue.createElementBlock(
          Vue.Fragment,
          { key: 1 },
          [
            Vue.createCommentVNode(" \u5982\u679C\u6CA1\u6709\u4F20\u5165\u63D2\u69FD\uFF0C\u5219\u6E32\u67D3\u9ED8\u8BA4\u7684 h1 \u6807\u7B7E\u53CA\u5185\u5BB9 "),
            Vue.createElementVNode("h1", {
              class: Vue.normalizeClass(Vue.unref(ns).e("title")),
              "aria-label": Vue.unref(t)("tk.homeBanner.titleLabel")
            }, Vue.toDisplayString(Vue.unref(bannerConfig).name), 11, _hoisted_2)
          ],
          2112
          /* STABLE_FRAGMENT, DEV_ROOT_FRAGMENT */
        )),
        Vue.createElementVNode("p", {
          class: Vue.normalizeClass(Vue.unref(ns).e("desc")),
          "aria-label": Vue.unref(t)("tk.homeBanner.descLabel")
        }, [
          isDefaultDescStyle.value ? (Vue.openBlock(), Vue.createElementBlock(
            "span",
            _hoisted_4,
            Vue.toDisplayString(descArray.value[0]),
            1
            /* TEXT */
          )) : isSwitchDescStyle.value ? (Vue.openBlock(), Vue.createBlock(Vue.Transition, {
            key: 1,
            name: Vue.unref(ns).join("fade"),
            mode: "out-in"
          }, {
            default: Vue.withCtx(() => [
              (Vue.openBlock(), Vue.createElementBlock("span", {
                key: Vue.unref(text) || descArray.value[0],
                onClick: _cache[0] || (_cache[0] = //@ts-ignore
                (...args) => Vue.unref(startAutoSwitch) && Vue.unref(startAutoSwitch)(...args)),
                class: "switch",
                "aria-label": Vue.unref(t)("tk.homeBanner.descSwitchLabel")
              }, Vue.toDisplayString(Vue.unref(text) || descArray.value[0]), 9, _hoisted_5))
            ]),
            _: 1
            /* STABLE */
          }, 8, ["name"])) : isTypesDescStyle.value && descArray.value.length ? (Vue.openBlock(), Vue.createElementBlock(
            Vue.Fragment,
            { key: 2 },
            [
              Vue.createElementVNode("span", {
                "aria-label": Vue.unref(t)("tk.homeBanner.descTypedLabel")
              }, Vue.toDisplayString(Vue.unref(typesText)), 9, _hoisted_6),
              Vue.createElementVNode(
                "span",
                {
                  class: Vue.normalizeClass(["typed", { "is-animation": Vue.unref(isFinished) }])
                },
                "|",
                2
                /* CLASS */
              )
            ],
            64
            /* STABLE_FRAGMENT */
          )) : Vue.createCommentVNode("v-if", true)
        ], 10, _hoisted_3)
      ], 10, _hoisted_1);
    };
  }
});

exports.default = _sfc_main;
