// vitepress-theme-teek/lib/components/common/ArticlePage/src/components/DocAsideOutlineItem.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
require('vitepress');
var useNamespace = require('../../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
require('../../../../../composables/useZIndex.js');

const _hoisted_1 = ["href", "title"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "DocAsideOutlineItem" },
  __name: "DocAsideOutlineItem",
  props: {
    headers: {},
    root: { type: Boolean }
  },
  setup(__props) {
    const ns = useNamespace.useNamespace("aside-outline-item");
    function onClick({ target: el }) {
      const id = el.href.split("#")[1];
      const heading = document.getElementById(decodeURIComponent(id));
      heading?.focus({ preventScroll: true });
    }
    return (_ctx, _cache) => {
      const _component_DocAsideOutlineItem = Vue.resolveComponent("DocAsideOutlineItem", true);
      return Vue.openBlock(), Vue.createElementBlock(
        "ul",
        {
          class: Vue.normalizeClass([Vue.unref(ns).b(), __props.root ? Vue.unref(ns).is("root") : Vue.unref(ns).is("nested")])
        },
        [
          (Vue.openBlock(true), Vue.createElementBlock(
            Vue.Fragment,
            null,
            Vue.renderList(__props.headers, ({ children, link, title }) => {
              return Vue.openBlock(), Vue.createElementBlock("li", { key: link }, [
                Vue.createElementVNode("a", {
                  class: Vue.normalizeClass(Vue.unref(ns).m("link")),
                  href: link,
                  onClick,
                  title
                }, Vue.toDisplayString(title), 11, _hoisted_1),
                children?.length ? (Vue.openBlock(), Vue.createBlock(_component_DocAsideOutlineItem, {
                  key: 0,
                  headers: children
                }, null, 8, ["headers"])) : Vue.createCommentVNode("v-if", true)
              ]);
            }),
            128
            /* KEYED_FRAGMENT */
          ))
        ],
        2
        /* CLASS */
      );
    };
  }
});

exports.default = _sfc_main;
