// vitepress-theme-teek@1.5.1/lib/components/theme/LoginPage/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var index$1 = require('../../../../helper/dist/index.js');
var useNamespace = require('../../../../composables/useNamespace.js');
require('vitepress-theme-teek/theme-chalk/tk-copy-banner.css');
var useLocale = require('../../../../composables/useLocale.js');
require('../../../../composables/useZIndex.js');
var user = require('../../../../static/icons/user.js');
var refreshRight = require('../../../../static/icons/refreshRight.js');
var successFilled = require('../../../../static/icons/successFilled.js');
var warningFilled = require('../../../../static/icons/warningFilled.js');
var lock = require('../../../../static/icons/lock.js');
var index = require('../../ConfigProvider/index.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../../common/Icon/src/index.vue2.js');
require('@iconify/vue');
var method = require('../../../common/Message/src/method.js');
var index_vue_vue_type_script_setup_true_lang = require('../../../common/VerifyCode/src/index.vue2.js');
var login = require('./login.js');

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["src"];
const _hoisted_3 = ["src"];
const _hoisted_4 = { class: "title" };
const _hoisted_5 = { class: "flx-space-y-20 login-form" };
const _hoisted_6 = ["for"];
const _hoisted_7 = ["onUpdate:modelValue", "type", "placeholder", "onFocus", "onBlur"];
const _hoisted_8 = ["aria-label"];
const _hoisted_9 = ["aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "LoginPage" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace.useNamespace("login");
    const router = vitepress.useRouter();
    const { frontmatter } = vitepress.useData();
    const posts = index.usePosts();
    const { t } = useLocale.useLocale();
    const { getTeekConfigRef } = index.useTeekConfig();
    const privateConfig = getTeekConfigRef("private", login.defaultPrivateConfig);
    const { siteLoginKey, pagesLoginKey, pageLoginKey, realmLoginKey } = login.getLoginStorageKey();
    const imgCode = Vue.ref("");
    const loginForm = Vue.reactive({
      username: {
        model: "",
        focusModel: false,
        errorModel: false,
        icon: user.default,
        placeholder: t("tk.login.usernamePlaceholder"),
        type: "text"
      },
      password: {
        model: "",
        focusModel: false,
        errorModel: false,
        icon: lock.default,
        placeholder: t("tk.login.passwordPlaceholder"),
        type: "password"
      },
      verifyCode: {
        model: "",
        focusModel: false,
        errorModel: false,
        icon: warningFilled.default,
        placeholder: t("tk.login.verifyCodePlaceholder"),
        type: "text",
        append: Vue.markRaw(index_vue_vue_type_script_setup_true_lang.default),
        appendModel: imgCode
      }
    });
    const checkLoginForm = () => {
      if (loginForm.verifyCode.model === "") {
        loginForm.verifyCode.errorModel = true;
        method.default.warning({ message: t("tk.login.verifyCodeNonNull"), plain: true });
        return false;
      }
      if (loginForm.verifyCode.model !== imgCode.value) {
        loginForm.verifyCode.errorModel = true;
        method.default.error({ message: t("tk.login.verifyCodeError"), plain: true });
        return false;
      }
      if (loginForm.username.model === "" || loginForm.password.model === "") {
        loginForm.username.errorModel = true;
        method.default.warning({ message: t("tk.login.loginInfoNonNull"), plain: true });
        return false;
      }
      if (loginForm.password.model === "") {
        loginForm.password.errorModel = true;
        method.default.warning({ message: t("tk.login.loginInfoNonNull"), plain: true });
        return false;
      }
      return true;
    };
    const getExpire = (expire) => {
      if (!expire) return 864e5;
      if (expire.indexOf("d") !== -1) return parseInt(expire.replace("d", "")) * 24 * 60 * 60 * 1e3;
      if (expire.indexOf("h") !== -1) return parseInt(expire.replace("h", "")) * 60 * 60 * 1e3;
      return parseInt(expire) * 1e3;
    };
    const resetForm = () => {
      Object.values(loginForm).forEach((form) => {
        form.model = "";
        form.focusModel = false;
      });
    };
    const getLoginHandler = () => {
      const { searchParams } = new URL(window.location.href);
      const verifyModeValue = searchParams.get(login.loginUrlKeyMap.verifyMode);
      const toPath = searchParams.get(login.loginUrlKeyMap.toPath);
      const realmValue = searchParams.get(login.loginUrlKeyMap.realm);
      const { site = [], pages = [], realm = {} } = privateConfig.value;
      return [
        {
          // 单页面级别登录
          condition: () => verifyModeValue === login.verifyModeMap.page && toPath,
          handle: () => execLogin([], pageLoginKey, { toPath })
        },
        {
          // 单页面级别登录
          condition: () => verifyModeValue === login.verifyModeMap.realm && realmValue,
          handle: () => execLogin(realm[realmValue] || [], realmLoginKey, { isRealm: true, realm: realmValue })
        },
        {
          // 全局页面级别登录
          condition: () => verifyModeValue === login.verifyModeMap.pages,
          handle: () => execLogin(pages, pagesLoginKey)
        },
        {
          // 站点级别登录
          condition: () => verifyModeValue === login.verifyModeMap.site,
          handle: () => execLogin(site, siteLoginKey, { isSite: true })
        }
      ].find((item) => item.condition());
    };
    const login$1 = () => {
      if (!index$1.isClient) return;
      const { enabled = false } = privateConfig.value;
      if (!enabled) {
        method.default.success({ message: t("tk.login.loginSuccess"), plain: true });
        return router.go("/");
      }
      if (!checkLoginForm()) return;
      const { searchParams } = new URL(window.location.href);
      const toPath = searchParams.get(login.loginUrlKeyMap.toPath);
      let isLogin = false;
      const handler = getLoginHandler();
      if (handler) {
        const { doLogin } = privateConfig.value;
        const loginInfo = { username: loginForm.username.model, password: loginForm.password.model };
        const nativeLogin = handler.handle;
        isLogin = doLogin ? doLogin(loginInfo, "page", nativeLogin) : nativeLogin();
      }
      if (isLogin === void 0) return;
      if (isLogin) {
        method.default.success({ message: t("tk.login.loginSuccess"), plain: true });
        if (toPath) router.go(toPath);
      } else method.default.error({ message: t("tk.login.loginError"), plain: true });
    };
    const execLogin = (loginInfo, storageKey, options = {}) => {
      const { toPath } = options;
      if (toPath) return execSinglePageLogin(toPath, storageKey);
      const credential = loginInfo.find(
        (item) => item.username === loginForm.username.model && item.password === loginForm.password.model
      );
      if (!credential) return false;
      return storeLoginInfo(credential, storageKey, options);
    };
    const execSinglePageLogin = (toPath, storageKey) => {
      const post = posts.value.originPosts.find((post2) => [post2.frontmatter.permalink, post2.url].includes(toPath));
      if (!post) return false;
      const { username, password, session, expire, strategy, realm } = post.frontmatter || {};
      const loginInfoList = [
        ...post.frontmatter.loginInfo || [],
        { username, password, session, expire, strategy, realm }
      ].filter((item) => ![void 0, ""].includes(item.username) && ![void 0, ""].includes(item.password));
      const loginInfo = loginInfoList.find(
        (item) => item.username === loginForm.username.model && item.password === loginForm.password.model
      );
      if (loginInfo) {
        return storeLoginInfo({ ...loginInfo}, storageKey + post.url);
      }
      if (realm && privateConfig.value.realm) {
        const nativeLogin = () => execLogin(privateConfig.value.realm[realm], realmLoginKey, { isRealm: true, realm });
        return privateConfig.value.doLogin ? privateConfig.value.doLogin({ username, password }, "realm", nativeLogin) : nativeLogin();
      }
      return false;
    };
    const storeLoginInfo = (loginInfo, storageKey, options = {}) => {
      const { session, expire, strategy = "once", role = "common" } = loginInfo;
      const { isSite = false, isRealm = false, realm } = options;
      const storage = session || privateConfig.value.session ? sessionStorage : localStorage;
      const key = isRealm ? `${storageKey}${realm}` : storageKey;
      const encrypt = privateConfig.value.encrypt;
      try {
        storage.setItem(
          key,
          JSON.stringify({
            username: loginForm.username.model,
            password: encrypt ? encrypt(loginForm.password.model, frontmatter) : loginForm.password.model,
            loginTime: (/* @__PURE__ */ new Date()).getTime(),
            expire: strategy === "always" ? (/* @__PURE__ */ new Date()).getTime() + 30 * 1e3 : getExpire(expire || privateConfig.value.expire),
            strategy: strategy || "once",
            ...isSite && { role }
            // 站点级别登录信息需要存储角色，如果为 admin，代表后续有所有的文章页面权限
          })
        );
        return true;
      } catch (error) {
        console.error("[Teek Error] Failed to store credentials:", error);
        return false;
      }
    };
    const handleFocus = (item, formName) => {
      item.focusModel = true;
      item.errorModel = false;
      privateConfig.value.onFocus?.(item.model, formName);
    };
    const handleBlur = (item, formName) => {
      item.focusModel = false;
      if (item.model === "") item.errorModel = true;
      privateConfig.value.onBlur?.(item.model, formName);
    };
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("div", {
        class: Vue.normalizeClass(Vue.unref(ns).b()),
        "aria-label": Vue.unref(t)("tk.login.label")
      }, [
        Vue.createElementVNode(
          "div",
          {
            class: Vue.normalizeClass(Vue.unref(ns).e("wrapper"))
          },
          [
            Vue.unref(frontmatter).leftImg ? (Vue.openBlock(), Vue.createElementBlock(
              "div",
              {
                key: 0,
                class: Vue.normalizeClass(Vue.unref(ns).e("left"))
              },
              [
                Vue.createElementVNode("img", {
                  src: Vue.unref(vitepress.withBase)(Vue.unref(frontmatter).leftImg),
                  alt: "login"
                }, null, 8, _hoisted_2)
              ],
              2
              /* CLASS */
            )) : Vue.createCommentVNode("v-if", true),
            Vue.createElementVNode(
              "div",
              {
                class: Vue.normalizeClass(Vue.unref(ns).e("right"))
              },
              [
                Vue.createElementVNode(
                  "div",
                  {
                    class: Vue.normalizeClass([Vue.unref(ns).e("right__header"), "flx-center"])
                  },
                  [
                    Vue.unref(frontmatter).logo ? (Vue.openBlock(), Vue.createElementBlock("img", {
                      key: 0,
                      src: Vue.unref(frontmatter).logo,
                      alt: "logo"
                    }, null, 8, _hoisted_3)) : Vue.createCommentVNode("v-if", true),
                    Vue.createElementVNode(
                      "span",
                      _hoisted_4,
                      Vue.toDisplayString(Vue.unref(frontmatter).name ?? "VitePress Theme Teek"),
                      1
                      /* TEXT */
                    )
                  ],
                  2
                  /* CLASS */
                ),
                Vue.createElementVNode("form", _hoisted_5, [
                  (Vue.openBlock(true), Vue.createElementBlock(
                    Vue.Fragment,
                    null,
                    Vue.renderList(loginForm, (item, key) => {
                      return Vue.openBlock(), Vue.createElementBlock("div", {
                        key,
                        class: "flx login-form-item"
                      }, [
                        Vue.createElementVNode(
                          "div",
                          {
                            class: Vue.normalizeClass([Vue.unref(ns).e("right__form"), Vue.unref(ns).is("focus", item.focusModel), Vue.unref(ns).is("error", item.errorModel)])
                          },
                          [
                            Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                              icon: item.icon
                            }, null, 8, ["icon"]),
                            Vue.createElementVNode("label", {
                              for: "input-" + key,
                              class: "sr-only"
                            }, Vue.toDisplayString(item.placeholder), 9, _hoisted_6),
                            Vue.withDirectives(Vue.createElementVNode("input", {
                              "onUpdate:modelValue": ($event) => item.model = $event,
                              type: item.type,
                              class: Vue.normalizeClass(Vue.unref(ns).em("right__form", "control")),
                              placeholder: item.placeholder,
                              onFocus: ($event) => handleFocus(item, key),
                              onBlur: ($event) => handleBlur(item, key),
                              onKeydown: Vue.withKeys(login$1, ["enter"])
                            }, null, 42, _hoisted_7), [
                              [Vue.vModelDynamic, item.model]
                            ])
                          ],
                          2
                          /* CLASS */
                        ),
                        item.append ? (Vue.openBlock(), Vue.createBlock(Vue.resolveDynamicComponent(item.append), {
                          key: 0,
                          modelValue: item.appendModel,
                          "onUpdate:modelValue": ($event) => item.appendModel = $event
                        }, null, 8, ["modelValue", "onUpdate:modelValue"])) : Vue.createCommentVNode("v-if", true)
                      ]);
                    }),
                    128
                    /* KEYED_FRAGMENT */
                  )),
                  Vue.createElementVNode(
                    "div",
                    {
                      class: Vue.normalizeClass(Vue.unref(ns).e("right__form__btn"))
                    },
                    [
                      Vue.createElementVNode("button", {
                        type: "button",
                        onClick: _cache[0] || (_cache[0] = ($event) => resetForm()),
                        class: "flx-center",
                        "aria-label": Vue.unref(t)("tk.login.reset")
                      }, [
                        Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), { icon: Vue.unref(refreshRight.default) }, null, 8, ["icon"]),
                        Vue.createElementVNode(
                          "span",
                          null,
                          Vue.toDisplayString(Vue.unref(t)("tk.login.reset")),
                          1
                          /* TEXT */
                        )
                      ], 8, _hoisted_8),
                      Vue.createElementVNode("button", {
                        type: "button",
                        onClick: _cache[1] || (_cache[1] = ($event) => login$1()),
                        class: "flx-center primary",
                        "aria-label": Vue.unref(t)("tk.login.login")
                      }, [
                        Vue.createVNode(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), { icon: Vue.unref(successFilled.default) }, null, 8, ["icon"]),
                        Vue.createElementVNode(
                          "span",
                          null,
                          Vue.toDisplayString(Vue.unref(t)("tk.login.login")),
                          1
                          /* TEXT */
                        )
                      ], 8, _hoisted_9)
                    ],
                    2
                    /* CLASS */
                  )
                ])
              ],
              2
              /* CLASS */
            )
          ],
          2
          /* CLASS */
        )
      ], 10, _hoisted_1);
    };
  }
});

exports.default = _sfc_main;
