// vitepress-theme-teek/es/components/theme/HomeFullscreenWallpaper/src/index.vue2.mjs
import { defineComponent, ref, createElementBlock, openBlock, normalizeClass, unref } from 'vue';
import { useEventListener } from '../../../../composables/useEventListener.mjs';
import 'vitepress';
import { useNamespace } from '../../../../composables/useNamespace.mjs';
import 'vitepress-theme-teek/theme-chalk/tk-copy-banner.css';
import '../../../../composables/useZIndex.mjs';
import { useTeekConfig } from '../../ConfigProvider/index.mjs';

var _sfc_main = /* @__PURE__ */ defineComponent({
  ...{ name: "HomeFullscreenWallpaper" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace("fullscreen-wallpaper");
    const { getTeekConfigRef } = useTeekConfig();
    const isFullscreen = ref(false);
    const wallpaperConfig = getTeekConfigRef("wallpaper", {
      hideBanner: false,
      hideMask: false
    });
    const handleKeyDown = (event) => {
      if (event.key === "F12" && isFullscreen.value) return event.preventDefault();
      if (event.key === "I" && event.ctrlKey && event.shiftKey && isFullscreen.value) return event.preventDefault();
      if (event.key === "F11") {
        event.preventDefault();
        if (document.documentElement.requestFullscreen) {
          document.documentElement.requestFullscreen();
        } else if (document.documentElement.mozRequestFullScreen) {
          document.documentElement.mozRequestFullScreen();
        } else if (document.documentElement.webkitRequestFullscreen) {
          document.documentElement.webkitRequestFullscreen();
        } else if (document.documentElement.msRequestFullscreen) {
          document.documentElement.msRequestFullscreen();
        }
      }
    };
    const handleFullscreenChange = () => {
      const htmlDom = document.documentElement;
      if (htmlDom.scrollTop !== 0) return;
      const vpDocDom = document.querySelector(".vp-doc");
      const bannerContentDom = document.querySelector(`.${ns.join("banner__content")}`);
      const wavesDom = document.querySelector(`.${ns.join("banner-waves")}`);
      const bodyBgImageMaskDom = document.querySelector(`.${ns.join("body-bg-image")} .mask`);
      const bannerMaskDom = document.querySelector(`.${ns.join("banner-bg-image")} .mask`);
      isFullscreen.value = !!document.fullscreenElement;
      const { hideBanner, hideMask } = wallpaperConfig.value;
      const options = [
        { el: htmlDom, executeClass: ns.b() },
        { el: vpDocDom, executeClass: "display-none" },
        { el: bannerContentDom, executeClass: "no-feature" },
        {
          el: bannerContentDom,
          executeClass: "display-none",
          execute: hideBanner
        },
        { el: wavesDom, executeClass: "display-none", execute: hideBanner },
        { el: bodyBgImageMaskDom, executeClass: "display-none", execute: hideMask },
        { el: bannerMaskDom, executeClass: "display-none", execute: hideMask }
      ];
      addOrRemoveClass(!!document.fullscreenElement, options);
    };
    const addOrRemoveClass = (add, options) => {
      if (add) {
        options.forEach((item) => {
          if (item.execute !== false) item.executeClass && item.el?.classList.add(item.executeClass);
          else item.notExecuteClass && item.el?.classList.add(item.notExecuteClass);
        });
        return;
      }
      options.forEach((item) => {
        if (item.execute !== false) item.executeClass && item.el?.classList.remove(item.executeClass);
        else item.notExecuteClass && item.el?.classList.remove(item.notExecuteClass);
      });
    };
    const handleContextMenu = (event) => {
      if (isFullscreen.value) event.preventDefault();
    };
    const getDocument = () => document;
    useEventListener(getDocument, "keydown", handleKeyDown);
    useEventListener(getDocument, "fullscreenchange", handleFullscreenChange);
    useEventListener(getDocument, "contextmenu", handleContextMenu);
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(
        "div",
        {
          class: normalizeClass(unref(ns).b())
        },
        null,
        2
        /* CLASS */
      );
    };
  }
});

export { _sfc_main as default };
